// @ciscodesignsystems/cds-react-heading/injected/heading.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var cdsComponentTypes = require('@ciscodesignsystems/cds-component-types');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var index = require('./types/index.cjs.js');
var heading_module = require('./heading.module.cjs.js');

const CDSHeading = /*#__PURE__*/react.forwardRef((props, ref) => {
  const {
    as,
    children,
    className,
    margin,
    size = 'section',
    ...rest
  } = props;
  const marginDataAttr = cdsComponentTypes.handleMarginInput(margin);
  const classNames = cdsComponentUtilities.cx(heading_module["default"], 'cds-heading', `cds-heading--${index.CDSHeadingSizeMap[size]}`, `cds-heading__margin-top--${marginDataAttr.top}`, `cds-heading__margin-bottom--${marginDataAttr.bottom}`, `cds-heading__margin-right--${marginDataAttr.right}`, `cds-heading__margin-left--${marginDataAttr.left}`, className);
  const Component = as || index.CDSHeadingSizeTagMap[index.CDSHeadingSizeMap[size]];
  return jsxRuntime.jsx(Component, {
    className: classNames,
    role: "heading",
    ref: ref,
    ...rest,
    children: children
  });
});
CDSHeading.displayName = 'CDSHeading';
CDSHeading.defaultProps = {
  margin: 0
};

exports.CDSHeading = CDSHeading;
exports["default"] = CDSHeading;
