// @ciscodesignsystems/cds-react-heading/injected/heading.esm.js
import { forwardRef } from 'react';
import { handleMarginInput } from '@ciscodesignsystems/cds-component-types';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { CDSHeadingSizeMap, CDSHeadingSizeTagMap } from './types/index.esm.js';
import styles from './heading.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const CDSHeading = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    as,
    children,
    className,
    margin,
    size = 'section',
    ...rest
  } = props;
  const marginDataAttr = handleMarginInput(margin);
  const classNames = cx(styles, 'cds-heading', `cds-heading--${CDSHeadingSizeMap[size]}`, `cds-heading__margin-top--${marginDataAttr.top}`, `cds-heading__margin-bottom--${marginDataAttr.bottom}`, `cds-heading__margin-right--${marginDataAttr.right}`, `cds-heading__margin-left--${marginDataAttr.left}`, className);
  const Component = as || CDSHeadingSizeTagMap[CDSHeadingSizeMap[size]];
  return /*#__PURE__*/jsx(Component, {
    className: classNames,
    role: "heading",
    ref: ref,
    ...rest,
    children: children
  });
});
CDSHeading.displayName = 'CDSHeading';
CDSHeading.defaultProps = {
  margin: 0
};

export { CDSHeading, CDSHeading as default };
