// @ciscodesignsystems/cds-react-table/components/table-cell/table-cell.esm.js
import { flexRender } from '@tanstack/react-table';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { getPinnedStatus } from '../../util/getPinnedStatus.esm.js';
import styles from './table-cell.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const TableCell = ({
  cell,
  pinnedPosition = undefined,
  children = undefined
}) => {
  const align = cell.column.columnDef.meta?.align ?? 'left';
  const {
    isPinned,
    pinnedClasses
  } = getPinnedStatus(cell.column);
  const isSelected = cell.row.getIsSelected();
  const cellClassNames = cx(styles, 'cds-table__cell', {
    'cds-table__cell--selected': isSelected
  }, pinnedClasses);
  return /*#__PURE__*/jsx("td", {
    className: cellClassNames,
    align: align,
    style: {
      left: isPinned === 'left' ? `${pinnedPosition}px` : undefined,
      right: isPinned === 'right' ? `${pinnedPosition}px` : undefined
    },
    children: children ?? flexRender(cell.column.columnDef.cell, cell.getContext())
  });
};

export { TableCell };
