// @ciscodesignsystems/cds-react-dropdown/util/hooks/use-menu.esm.js
import { useState, useRef, useEffect } from 'react';
import { useFloatingTree, useFloatingNodeId, useFloatingParentNodeId } from '@floating-ui/react';
import { useControllableState } from '@ciscodesignsystems/cds-component-utilities';
import { useMenuInteractions } from './use-menu-interactions.esm.js';

const useMenu = ({
  open,
  onChange,
  defaultOpen
}) => {
  const [isOpen, setIsOpen] = useControllableState({
    value: open,
    onChange: onChange,
    defaultValue: defaultOpen
  });
  const [activeIndex, setActiveIndex] = useState(null);
  const [allowHover, setAllowHover] = useState(false);
  const listItemsRef = useRef([]);
  const tree = useFloatingTree();
  const nodeId = useFloatingNodeId();
  const parentId = useFloatingParentNodeId();
  const isNested = parentId !== null;
  const {
    context,
    getFloatingProps,
    getItemProps,
    getReferenceProps,
    refs,
    strategy,
    x,
    y
  } = useMenuInteractions({
    activeIndex,
    allowHover,
    isNested,
    isOpen,
    listItemsRef,
    nodeId,
    setActiveIndex,
    setIsOpen
  });

  // Event emitter allows you to communicate across tree components.
  // This effect closes all menus when an item gets clicked anywhere
  // in the tree.
  useEffect(() => {
    if (!tree) {
      return;
    }
    function handleTreeClick() {
      setIsOpen(false);
    }
    function onSubMenuOpen(event) {
      if (event.nodeId !== nodeId && event.parentId === parentId) {
        setIsOpen(false);
      }
    }
    tree.events.on('click', handleTreeClick);
    tree.events.on('menuopen', onSubMenuOpen);
    return () => {
      tree.events.off('click', handleTreeClick);
      tree.events.off('menuopen', onSubMenuOpen);
    };
  }, [tree, nodeId, parentId, setIsOpen]);
  useEffect(() => {
    if (isOpen && tree) {
      tree.events.emit('menuopen', {
        parentId,
        nodeId
      });
    }
  }, [tree, isOpen, nodeId, parentId]);

  // Determine if "hover" logic can run based on the modality of input. This
  // prevents unwanted focus synchronization as menus open and close with
  // keyboard navigation and the cursor is resting on the menu.
  useEffect(() => {
    function onPointerMove({
      pointerType
    }) {
      if (pointerType !== 'touch') {
        setAllowHover(true);
      }
    }
    function onKeyDown() {
      setAllowHover(false);
    }
    window.addEventListener('pointermove', onPointerMove, {
      once: true,
      capture: true
    });
    window.addEventListener('keydown', onKeyDown, true);
    return () => {
      window.removeEventListener('pointermove', onPointerMove, {
        capture: true
      });
      window.removeEventListener('keydown', onKeyDown, true);
    };
  }, [allowHover]);
  return {
    isOpen,
    activeIndex,
    setActiveIndex,
    allowHover,
    setAllowHover,
    listItemsRef,
    tree,
    nodeId,
    isNested,
    context,
    refs,
    strategy,
    x,
    y,
    getFloatingProps,
    getItemProps,
    getReferenceProps
  };
};

export { useMenu };
