// @ciscodesignsystems/cds-react-dropdown/injected/dropdown.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var react$1 = require('@floating-ui/react');
var phosphorReact = require('phosphor-react');
var index$1 = require('./components/item/index.cjs.js');
var index = require('./components/root/index.cjs.js');
var index$3 = require('./components/section/index.cjs.js');
var index$2 = require('./components/submenu/index.cjs.js');
var context = require('./util/context.cjs.js');
var withSubComponet = require('./util/with-sub-componet.cjs.js');

/**
 * This is where you want to insert `CDSDropdown.`-prefixed contextual components in order to create your dropdown menu.
 *
 * Usage example:
 *
 * ```tsx
 * <CDSDropdown label="Label">
 *   <CDSDropdown.Section label="Section title" divider={true}>
 *     <CDSDropdown.Item label="Item 1" />
 *   </CDSDropdown.Section>
 *   <CDSDropdown.Section label="Section title" />
 *     <CDSDropdown.Item label="Item 2" />
 *     <CDSDropdown.Submenu label="Submenu">
 *       <CDSDropdown.Item label="Submenu item" />
 *     </CDSDropdown.Submenu>
 *   </CDSDropdown.Section>
 * </CDSDropdown>
 * ```
 */
const CDSDropdown = /*#__PURE__*/react.forwardRef(({
  placement = 'left',
  size = 'md',
  className,
  ...rest
}, ref) => {
  const iconContext = react.useMemo(() => ({
    size: size === 'sm' ? 18 : 20,
    weight: 'bold'
  }), [size]);
  const dropdownContextValue = react.useMemo(() => ({
    placement,
    size,
    className
  }), [className, placement, size]);
  return jsxRuntime.jsx(phosphorReact.IconContext.Provider, {
    value: iconContext,
    children: jsxRuntime.jsx(context.CDSDropdownContext.Provider, {
      value: dropdownContextValue,
      children: jsxRuntime.jsx(react$1.FloatingTree, {
        children: jsxRuntime.jsx(index.CDSDropdownRoot, {
          className: className,
          ref: ref,
          ...rest
        })
      })
    })
  });
});
CDSDropdown.displayName = 'CDSDropdown';
CDSDropdown.defaultProps = {
  label: '',
  open: undefined,
  onChange: undefined,
  defaultOpen: false,
  root: undefined
};
const CDSDropdownNamespace = withSubComponet.withSubComponent(CDSDropdown, {
  /**
   * The basic building block of dropdown menus, this creates an item that can be clicked to trigger an action.
   *
   * Usage example:
   *
   * ```tsx
   * <CDSDropdown label="Label">
   *   <CDSDropdown.Item label="Option 1" onClick={handleOnFoo} />
   *   <CDSDropdown.Item label="Option 2" onClick={handleOnBar} />
   * </CDSDropdown>
   * ```
   */
  Item: index$1.CDSDropdownItem,
  /**
   * Allows for the creation of cascading or sub-menus.
   *
   * Usage example:
   *
   * ```tsx
   * <CDSDropdown label="Label">
   *   <CDSDropdown.Item label="Option 1" />
   *   <CDSDropdown.Item label="Option 2" />
   *   <CDSDropdown.Submenu label="Submenu">
   *     <CDSDropdown.Item label="Option 3" />
   *     <CDSDropdown.Item label="Option 4" />
   *   </CDSDropdown.Submenu>
   * </CDSDropdown>
   * ```
   */
  Submenu: index$2.CDSDropdownSubmenu,
  /**
   * Allows for the creation of sections within the menus.
   *
   * Usage example:
   *
   * ```tsx
   * <CDSDropdown label="Label">
   *   <CDSDropdown.Section label="Section" divider={true}>
   *     <CDSDropdown.Item label="Item" />
   *   </CDSDropdown.Section>
   *   <CDSDropdown.Section label="Section">
   *     <CDSDropdown.Item label="Item" />
   *   </CDSDropdown.Section>
   * </CDSDropdown>
   * ```
   */
  Section: index$3.CDSDropdownSection
});
CDSDropdownNamespace.displayName = 'CDSDropdown';

exports.CDSDropdown = CDSDropdownNamespace;
exports["default"] = CDSDropdownNamespace;
