// @ciscodesignsystems/cds-react-header/components/header-badge/header-badge.esm.js
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { CDSHeaderBadgeSizes } from './types/index.esm.js';
import styles from './header-badge.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';

/* eslint-disable @nx/enforce-module-boundaries */
const CDSHeaderBadge = ({
  className = '',
  children = undefined,
  content = undefined,
  size = 'md',
  sentiment,
  prominence = '',
  childAlign = 'inner-left'
}) => {
  const getDivPadding = () => {
    if (children && content) {
      return {
        left: '0 0.25em 0 0',
        right: '0 0 0 0.25em',
        inner_right: '0',
        inner_left: '0'
      }[childAlign.replace('-', '_')];
    }
    return '0';
  };
  const getSpanPadding = () => {
    if (children) {
      return {
        left: '0.25em 0.5em',
        right: '0.25em 0.5em',
        inner_right: '0.25em 0 0.25em 0.5em',
        inner_left: '0.25em 0.5em 0.25em 0'
      }[childAlign.replace('-', '_')];
    }
    return '0.25em 0.5em';
  };
  const showchildAlignSpan = checkAlign => childAlign === checkAlign && children;
  return /*#__PURE__*/jsxs("div", {
    className: cx(styles, 'cds-header-badge-wrapper', className),
    style: {
      padding: getDivPadding()
    },
    children: [showchildAlignSpan('left') ? /*#__PURE__*/jsx("span", {
      className: cx(styles, 'cds-header-badge-align-left'),
      children: children
    }) : null, /*#__PURE__*/jsxs("div", {
      className: cx(styles, 'cds-header-badge-content-wrapper'),
      "data-cds-sentiment": sentiment,
      "data-cds-prominence": prominence,
      style: {
        fontSize: CDSHeaderBadgeSizes[size] + 'px'
      },
      children: [showchildAlignSpan('inner-left') ? /*#__PURE__*/jsx("span", {
        className: cx(styles, 'cds-header-badge-align-inner-left'),
        children: children
      }) : null, content ? /*#__PURE__*/jsx("span", {
        "aria-live": "polite",
        className: cx(styles, 'cds-header-badge-content'),
        style: {
          padding: getSpanPadding()
        },
        children: content
      }) : !!childAlign && /*#__PURE__*/jsx("span", {
        className: cx(styles, 'cds-header-badge-content-empty')
      }), showchildAlignSpan('inner-right') ? /*#__PURE__*/jsx("span", {
        className: cx(styles, 'cds-header-badge-align-inner-right'),
        children: children
      }) : null]
    }), showchildAlignSpan('right') ? /*#__PURE__*/jsx("span", {
      className: cx(styles, 'cds-header-badge-align-right'),
      children: children
    }) : null]
  });
};

export { CDSHeaderBadge, CDSHeaderBadge as default };
