// @ciscodesignsystems/cds-react-header/injected/components/alerts-menu/alerts-menu.esm.js
import { forwardRef } from 'react';
import { Bell } from 'phosphor-react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { CDSButton } from '@ciscodesignsystems/cds-react-button/injected';
import styles from './alerts-menu.module.esm.js';
import { jsx } from 'react/jsx-runtime';
import { CDSHeaderMenuRoot, CDSHeaderMenuContent, CDSHeaderMenuRow } from '../header-menu/header-menu.esm.js';
import { CDSHeaderBadge } from '../header-badge/header-badge.esm.js';

/**
 * The most-parent element of the alerts widget
 */
const CDSAlertsMenuRoot = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    sentiment,
    variant,
    children,
    contentRef,
    alertCount,
    alertNumber,
    alertThreshold,
    alertThresholdSymbol,
    triggerRef
  } = props;
  const getBadgeString = () => {
    if (alertNumber && alertThreshold && alertNumber > alertThreshold) {
      return `${alertThreshold}${alertThresholdSymbol}`;
    } else {
      if (alertNumber) return alertNumber;
      return undefined;
    }
  };
  const showHeaderBadge = alertCount && getBadgeString() !== undefined;
  return /*#__PURE__*/jsx(CDSHeaderMenuRoot, {
    ref: ref,
    id: "headerAlertsMenu",
    "aria-label": "Alerts menu",
    trigger: /*#__PURE__*/jsx(CDSButton, {
      ref: triggerRef,
      sentiment: sentiment,
      variant: variant,
      icon: /*#__PURE__*/jsx(Bell, {
        weight: "bold",
        size: "24px"
      }),
      "aria-label": "Alerts content",
      className: cx(styles, 'cds-alerts-menu-trigger'),
      children: showHeaderBadge && /*#__PURE__*/jsx(CDSHeaderBadge, {
        content: getBadgeString(),
        sentiment: "negative"
      })
    }),
    children: /*#__PURE__*/jsx(CDSHeaderMenuContent, {
      className: cx(styles, 'cds-alerts-menu'),
      align: "left",
      ref: contentRef,
      children: children
    })
  });
});
CDSAlertsMenuRoot.displayName = 'CDSAlertsMenuRoot';

/**
 * Divider that goes between rows of content
 */
const CDSAlertsMenuDivider = /*#__PURE__*/forwardRef((props, ref) => {
  return /*#__PURE__*/jsx("hr", {
    className: cx(styles, 'cds-alerts-menu__divider'),
    ref: ref,
    ...props
  });
});
CDSAlertsMenuDivider.displayName = 'CDSAlertsMenuDivider';

/**
 * Content that goes at the bottom of the Alerts Menu
 */
const CDSAlertsMenuFooter = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return /*#__PURE__*/jsx(CDSHeaderMenuRow, {
    ref: ref,
    className: cx(styles, 'cds-alerts-menu__footer'),
    ...rest,
    children: children
  });
});
CDSAlertsMenuFooter.displayName = 'CDSAlertsMenuFooter';
CDSAlertsMenuRoot.defaultProps = {
  sentiment: 'inverse',
  variant: 'ghost',
  alertCount: true,
  alertNumber: undefined,
  alertThreshold: 99,
  alertThresholdSymbol: '+',
  contentRef: undefined,
  triggerRef: undefined
};

export { CDSAlertsMenuDivider, CDSAlertsMenuFooter, CDSAlertsMenuRoot, CDSAlertsMenuRoot as default };
