// @ciscodesignsystems/cds-react-select/components/add-item/add-item.esm.js
import { Plus } from 'phosphor-react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './add-item.module.esm.js';
import { jsx, jsxs } from 'react/jsx-runtime';

const CDSSelectAddItem = ({
  inputValue,
  handleAddClick,
  size
}) => {
  return /*#__PURE__*/jsx("div", {
    className: cx(styles, 'cds-select__item', 'cds-select__item--add', {
      [`cds-select__item--${size}`]: size
    }),
    children: /*#__PURE__*/jsxs("div", {
      className: cx(styles, 'cds-select__item-content'),
      children: [/*#__PURE__*/jsx("span", {
        className: cx(styles, 'cds-select__item__add-value'),
        "data-testid": 'cds-select-item-add-value',
        children: inputValue
      }), /*#__PURE__*/jsxs("button", {
        className: cx(styles, 'cds-select__item', 'cds-select__item__add-btn'),
        "data-testid": 'cds-select-item-add-btn',
        type: 'button',
        onClick: handleAddClick,
        children: [/*#__PURE__*/jsx(Plus, {
          size: 16,
          weight: 'bold'
        }), "Add"]
      })]
    })
  });
};

export { CDSSelectAddItem };
