// @ciscodesignsystems/cds-react-text/injected/text.esm.js
import { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { CDSTextSizeMap } from './types/index.esm.js';
import styles from './text.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const CDSText = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    as,
    align = 'left',
    className,
    size = 'p3',
    weight = 'regular',
    monospace,
    compact,
    ellipsis,
    width,
    style,
    color = 'regular',
    ...rest
  } = props;
  const Component = as || 'p';
  const applyPresetColor = ['light', 'regular'].includes(color);
  const classNames = cx(styles, 'cds-text', `cds-text--${CDSTextSizeMap[size]}`, `cds-text__weight--${weight}`, {
    'cds-text--monospace': monospace,
    'cds-text--compact': compact,
    'cds-text--ellipsis': ellipsis,
    [`cds-text__color--${color}`]: applyPresetColor
  }, className);
  const computedStyles = {
    ...style,
    ...(align && {
      textAlign: align
    }),
    ...(color && !applyPresetColor && {
      color
    }),
    ...(width && {
      width
    })
  };
  return /*#__PURE__*/jsx(Component, {
    className: classNames,
    ref: ref,
    style: computedStyles,
    ...rest
  });
});
CDSText.displayName = 'CDSText';
CDSText.defaultProps = {
  compact: false,
  ellipsis: false,
  monospace: false,
  width: undefined
};

export { CDSText, CDSText as default };
