// @ciscodesignsystems/cds-react-header/injected/components/skip-to-main/skip-to-main.esm.js
import { forwardRef, useState, useRef, useCallback, useEffect } from 'react';
import { useMergeRefs } from '@ciscodesignsystems/cds-component-types';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './skip-to-main.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const CDSSkipToMain = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    as: Component = 'a',
    label = 'Skip to main content',
    inputRef,
    ...rest
  } = props;
  const [isVisible, setIsVisible] = useState(false);
  const [isClicked, setIsClicked] = useState(false);
  const skipLinkRef = useRef(null);
  const mergedRef = useMergeRefs([skipLinkRef, ref]);
  const handleKeyUp = useCallback(event => {
    if (event.key === 'Tab' && !isClicked) {
      setIsVisible(true);
    }
  }, [isClicked]);
  useEffect(() => {
    document.addEventListener('keyup', handleKeyUp);
    return () => {
      document.removeEventListener('keyup', handleKeyUp);
    };
  }, [handleKeyUp]);
  useEffect(() => {
    const handleFirstTabPress = event => {
      if (event.key === 'Tab' && !isClicked) {
        if (skipLinkRef.current) {
          skipLinkRef.current.focus();
        }
        setIsVisible(true);
        window.removeEventListener('keydown', handleFirstTabPress);
      }
    };
    window.addEventListener('keydown', handleFirstTabPress);
    return () => {
      window.removeEventListener('keydown', handleFirstTabPress);
    };
  }, [isClicked, skipLinkRef]);
  const handleSkipToMainClick = () => {
    setIsClicked(true);
    setIsVisible(false);
  };
  const handleEnterClick = event => {
    if (event.key === 'Enter') {
      setIsClicked(true);
      setIsVisible(false);
    }
  };
  return isVisible ? /*#__PURE__*/jsx("div", {
    className: cx(styles, 'cds-skip-to-main'),
    children: /*#__PURE__*/jsx(Component, {
      ...rest,
      className: cx(styles, 'cds-skip-to-main__link'),
      onClick: handleSkipToMainClick,
      onKeyDown: handleEnterClick,
      ref: mergedRef,
      children: label
    })
  }) : null;
});
CDSSkipToMain.displayName = 'CDSSkipToMain';

export { CDSSkipToMain, CDSSkipToMain as default };
