// @ciscodesignsystems/cds-react-checkbox/checkbox.esm.js
import { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './checkbox.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';

const CDSCheckboxSVG = ({
  indeterminate = false
}) => {
  return /*#__PURE__*/jsxs("svg", {
    fill: "none",
    viewBox: "0 0 18 18",
    className: cx(styles, 'cds-checkbox__svg'),
    children: [/*#__PURE__*/jsx("rect", {
      height: "100%",
      width: "100%"
    }), indeterminate ? /*#__PURE__*/jsx("line", {
      strokeWidth: "2",
      x1: "25%",
      x2: "75%",
      y1: "50%",
      y2: "50%"
    }) : /*#__PURE__*/jsx("polyline", {
      points: "14 5 8 12 4 8",
      strokeWidth: "2"
    }), /*#__PURE__*/jsx("line", {})]
  });
};
const CDSCheckbox = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    id,
    checked,
    disabled,
    indeterminate,
    size,
    className,
    children,
    ...rest
  } = props;
  const classes = cx(styles, 'cds-checkbox', `cds-checkbox--${size}`, {
    'cds-checkbox--disabled': disabled,
    'cds-checkbox--indeterminate': indeterminate,
    'cds-checkbox--checked': checked
  }, className);
  return /*#__PURE__*/jsxs("label", {
    htmlFor: id,
    className: classes,
    children: [/*#__PURE__*/jsx("input", {
      id: id,
      type: "checkbox",
      checked: checked,
      disabled: disabled,
      ref: ref,
      ...rest
    }), /*#__PURE__*/jsx(CDSCheckboxSVG, {
      indeterminate: indeterminate
    }), children]
  });
});
CDSCheckbox.displayName = 'CDSCheckbox';
CDSCheckbox.defaultProps = {
  children: undefined,
  indeterminate: false,
  checked: false,
  size: 'md'
};

export { CDSCheckbox, CDSCheckbox as default };
