// @ciscodesignsystems/cds-react-checkbox/checkbox.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var checkbox_module = require('./checkbox.module.cjs.js');

const CDSCheckboxSVG = ({
  indeterminate = false
}) => {
  return jsxRuntime.jsxs("svg", {
    fill: "none",
    viewBox: "0 0 18 18",
    className: cdsComponentUtilities.cx(checkbox_module["default"], 'cds-checkbox__svg'),
    children: [jsxRuntime.jsx("rect", {
      height: "100%",
      width: "100%"
    }), indeterminate ? jsxRuntime.jsx("line", {
      strokeWidth: "2",
      x1: "25%",
      x2: "75%",
      y1: "50%",
      y2: "50%"
    }) : jsxRuntime.jsx("polyline", {
      points: "14 5 8 12 4 8",
      strokeWidth: "2"
    }), jsxRuntime.jsx("line", {})]
  });
};
const CDSCheckbox = /*#__PURE__*/react.forwardRef((props, ref) => {
  const {
    id,
    checked,
    disabled,
    indeterminate,
    size,
    className,
    children,
    ...rest
  } = props;
  const classes = cdsComponentUtilities.cx(checkbox_module["default"], 'cds-checkbox', `cds-checkbox--${size}`, {
    'cds-checkbox--disabled': disabled,
    'cds-checkbox--indeterminate': indeterminate,
    'cds-checkbox--checked': checked
  }, className);
  return jsxRuntime.jsxs("label", {
    htmlFor: id,
    className: classes,
    children: [jsxRuntime.jsx("input", {
      id: id,
      type: "checkbox",
      checked: checked,
      disabled: disabled,
      ref: ref,
      ...rest
    }), jsxRuntime.jsx(CDSCheckboxSVG, {
      indeterminate: indeterminate
    }), children]
  });
});
CDSCheckbox.displayName = 'CDSCheckbox';
CDSCheckbox.defaultProps = {
  children: undefined,
  indeterminate: false,
  checked: false,
  size: 'md'
};

exports.CDSCheckbox = CDSCheckbox;
exports["default"] = CDSCheckbox;
