// @ciscodesignsystems/cds-react-header/header.esm.js
import { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './header.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { CDSHeaderLogoLink } from './components/header-logo-link/header-logo-link.esm.js';

const CDSHeader = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    as,
    sentiment,
    gradient,
    title,
    logo,
    logoLinkProps,
    logoLink,
    logoLinkTarget,
    children,
    minWidth,
    className,
    ...rest
  } = props;
  const Component = as || 'header';
  return /*#__PURE__*/jsxs(Component, {
    ref: ref,
    className: cx(styles, 'cds-header', className),
    style: {
      minWidth: minWidth,
      background: gradient === true ? `linear-gradient(90deg, var(--${sentiment}-bg-gradient-start), var(--${sentiment}-bg-gradient-end))` : `var(--${sentiment}-bg-base`
    },
    ...rest,
    children: [/*#__PURE__*/jsx("div", {
      className: cx(styles, 'cds-header__logo'),
      children: /*#__PURE__*/jsxs(CDSHeaderLogoLink, {
        ...(logoLinkProps ?? {
          href: logoLink,
          target: logoLinkTarget
        }),
        children: [/*#__PURE__*/jsx("span", {
          "data-testid": "headerLogo",
          className: cx(styles, 'cds-header__logoImage'),
          children: logo
        }), /*#__PURE__*/jsx("span", {
          className: cx(styles, 'cds-header__title'),
          "data-testid": "headerTitle",
          style: {
            color: `var(--${sentiment}-text-default)`
          },
          children: title
        })]
      })
    }), children]
  });
});
CDSHeader.displayName = 'CDSHeader';
CDSHeader.defaultProps = {
  as: undefined,
  sentiment: 'inverse',
  gradient: true,
  title: 'Product Name',
  logoLink: '#0',
  logoLinkTarget: '_self',
  minWidth: '880px',
  className: '',
  logoLinkProps: undefined
};

export { CDSHeader };
