// @ciscodesignsystems/cds-react-header/components/menu/components/menu-item/menu-item.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var menuItem_module = require('./menu-item.module.cjs.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const getTitle = node => {
  if (typeof node === 'string') {
    return node;
  }
  if (Array.isArray(node)) {
    return node.map(child => getTitle(child)).join('');
  }
  if ( /*#__PURE__*/React__default["default"].isValidElement(node) && node.props && node.props.children) {
    return getTitle(node.props.children);
  }
  return '';
};
const CDSMenuItem = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    label,
    title,
    subtitle,
    children,
    icon,
    action,
    truncate,
    className,
    ...rest
  } = props;
  const classes = cdsComponentUtilities.cx(menuItem_module["default"], 'cds-menu__item', className);
  const contentClasses = cdsComponentUtilities.cx(menuItem_module["default"], 'cds-menu__item__content', {
    'cds-menu__item__content--truncated': truncate
  });
  return jsxRuntime.jsxs("li", {
    ref: ref,
    className: classes,
    ...rest,
    children: [icon ? jsxRuntime.jsx("div", {
      className: cdsComponentUtilities.cx(menuItem_module["default"], 'cds-menu__item__affix', 'cds-menu__item__affix--left'),
      children: icon
    }) : null, jsxRuntime.jsxs("div", {
      className: contentClasses,
      children: [label !== '' ? jsxRuntime.jsx("div", {
        className: cdsComponentUtilities.cx(menuItem_module["default"], 'cds-menu__item__content__label'),
        title: getTitle(label),
        children: label
      }) : null, jsxRuntime.jsx("div", {
        className: cdsComponentUtilities.cx(menuItem_module["default"], 'cds-menu__item__content__title'),
        title: getTitle(title),
        children: title
      }), subtitle !== '' ? jsxRuntime.jsx("div", {
        className: cdsComponentUtilities.cx(menuItem_module["default"], 'cds-menu__item__content__subtitle'),
        title: getTitle(subtitle),
        children: subtitle
      }) : null]
    }), jsxRuntime.jsx("div", {
      className: cdsComponentUtilities.cx(menuItem_module["default"], 'cds-menu__item__affix', 'cds-menu__item__affix--right'),
      children: action
    })]
  });
});
CDSMenuItem.displayName = 'CDSMenuItem';
CDSMenuItem.defaultProps = {
  icon: false,
  label: '',
  subtitle: '',
  truncate: false
};

exports.CDSMenuItem = CDSMenuItem;
exports["default"] = CDSMenuItem;
exports.getTitle = getTitle;
