// @ciscodesignsystems/cds-react-breadcrumb/components/breadcrumb-link.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var phosphorReact = require('phosphor-react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var cdsReactTooltip = require('@ciscodesignsystems/cds-react-tooltip');
var breadcrumb_module = require('../breadcrumb.module.cjs.js');

const CDSBreadcrumbLink = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    children,
    className,
    withArrow = false,
    as,
    ...rest
  } = props;
  const classes = cdsComponentUtilities.cx(breadcrumb_module["default"], 'cds-breadcrumb__link', className);
  const Component = as || 'a';
  return children.length > 50 ? jsxRuntime.jsx(cdsReactTooltip.CDSTooltip, {
    placement: "bottom",
    title: children,
    children: jsxRuntime.jsxs(Component, {
      className: classes,
      ref: ref,
      ...rest,
      children: [withArrow ? jsxRuntime.jsx(phosphorReact.ArrowLeft, {
        className: cdsComponentUtilities.cx(breadcrumb_module["default"], 'cds-breadcrumb__arrow'),
        weight: "bold"
      }) : '', children.slice(0, 50), "..."]
    })
  }) : jsxRuntime.jsxs(Component, {
    className: classes,
    ref: ref,
    ...rest,
    children: [withArrow ? jsxRuntime.jsx(phosphorReact.ArrowLeft, {
      className: cdsComponentUtilities.cx(breadcrumb_module["default"], 'cds-breadcrumb__arrow'),
      weight: "bold"
    }) : '', children]
  });
});
CDSBreadcrumbLink.displayName = 'CDSBreadcrumb.Link';

exports.CDSBreadcrumbLink = CDSBreadcrumbLink;
exports["default"] = CDSBreadcrumbLink;
