// @ciscodesignsystems/cds-react-select/injected/components/add-item/add-item.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var phosphorReact = require('phosphor-react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var addItem_module = require('./add-item.module.cjs.js');

const CDSSelectAddItem = ({
  inputValue,
  handleAddClick,
  size
}) => {
  return jsxRuntime.jsx("div", {
    className: cdsComponentUtilities.cx(addItem_module["default"], 'cds-select__item', 'cds-select__item--add', {
      [`cds-select__item--${size}`]: size
    }),
    children: jsxRuntime.jsxs("div", {
      className: cdsComponentUtilities.cx(addItem_module["default"], 'cds-select__item-content'),
      children: [jsxRuntime.jsx("span", {
        className: cdsComponentUtilities.cx(addItem_module["default"], 'cds-select__item__add-value'),
        "data-testid": 'cds-select-item-add-value',
        children: inputValue
      }), jsxRuntime.jsxs("button", {
        className: cdsComponentUtilities.cx(addItem_module["default"], 'cds-select__item', 'cds-select__item__add-btn'),
        "data-testid": 'cds-select-item-add-btn',
        type: 'button',
        onClick: handleAddClick,
        children: [jsxRuntime.jsx(phosphorReact.Plus, {
          size: 16,
          weight: 'bold'
        }), "Add"]
      })]
    })
  });
};

exports.CDSSelectAddItem = CDSSelectAddItem;
