// @ciscodesignsystems/cds-react-breadcrumb/injected/components/breadcrumb-multiple.esm.js
import { CaretRight } from 'phosphor-react';
import { useOverflow, cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from '../breadcrumb.module.esm.js';
import { jsx, jsxs } from 'react/jsx-runtime';

const CDSBreadcrumbMultiple = ({
  children,
  className,
  ...props
}) => {
  const {
    childArray: [first, ...rest],
    ref,
    visibleItems
  } = useOverflow(children, '.cds-breadcrumb__item', 0);
  return /*#__PURE__*/jsx("nav", {
    className: cx(styles, 'cds-breadcrumb', className),
    ...props,
    ref: ref,
    children: /*#__PURE__*/jsxs("ul", {
      children: [/*#__PURE__*/jsx("li", {
        className: cx(styles, 'cds-breadcrumb__item'),
        children: first
      }), visibleItems < rest.length && /*#__PURE__*/jsxs("div", {
        className: cx(styles, 'cds-breadcrumb__item'),
        children: [/*#__PURE__*/jsx(CaretRight, {
          size: 16,
          weight: "bold"
        }), "..."]
      }), rest.slice(-visibleItems).map(current => /*#__PURE__*/jsxs("li", {
        className: cx(styles, 'cds-breadcrumb__item'),
        children: [/*#__PURE__*/jsx(CaretRight, {
          size: 16,
          weight: "bold"
        }), current]
      }, typeof current !== 'string' && typeof current !== 'number' && 'key' in current ? current.key : current.toString()))]
    })
  });
};

export { CDSBreadcrumbMultiple };
