// @ciscodesignsystems/cds-react-breadcrumb/injected/components/breadcrumb-link.esm.js
import { forwardRef } from 'react';
import { ArrowLeft } from 'phosphor-react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { CDSTooltip } from '@ciscodesignsystems/cds-react-tooltip/injected';
import styles from '../breadcrumb.module.esm.js';
import { jsx, jsxs } from 'react/jsx-runtime';

const CDSBreadcrumbLink = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    className,
    withArrow = false,
    as,
    ...rest
  } = props;
  const classes = cx(styles, 'cds-breadcrumb__link', className);
  const Component = as || 'a';
  return children.length > 50 ? /*#__PURE__*/jsx(CDSTooltip, {
    placement: "bottom",
    title: children,
    children: /*#__PURE__*/jsxs(Component, {
      className: classes,
      ref: ref,
      ...rest,
      children: [withArrow ? /*#__PURE__*/jsx(ArrowLeft, {
        className: cx(styles, 'cds-breadcrumb__arrow'),
        weight: "bold"
      }) : '', children.slice(0, 50), "..."]
    })
  }) : /*#__PURE__*/jsxs(Component, {
    className: classes,
    ref: ref,
    ...rest,
    children: [withArrow ? /*#__PURE__*/jsx(ArrowLeft, {
      className: cx(styles, 'cds-breadcrumb__arrow'),
      weight: "bold"
    }) : '', children]
  });
});
CDSBreadcrumbLink.displayName = 'CDSBreadcrumb.Link';

export { CDSBreadcrumbLink, CDSBreadcrumbLink as default };
