// @ciscodesignsystems/cds-react-header/injected/components/header-badge/header-badge.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var index = require('./types/index.cjs.js');
var headerBadge_module = require('./header-badge.module.cjs.js');

const CDSHeaderBadge = ({
  className = '',
  children = undefined,
  content = undefined,
  size = 'md',
  sentiment,
  prominence = '',
  childAlign = 'inner-left'
}) => {
  const getDivPadding = () => {
    if (children && content) {
      return {
        left: '0 0.25em 0 0',
        right: '0 0 0 0.25em',
        inner_right: '0',
        inner_left: '0'
      }[childAlign.replace('-', '_')];
    }
    return '0';
  };
  const getSpanPadding = () => {
    if (children) {
      return {
        left: '0.25em 0.5em',
        right: '0.25em 0.5em',
        inner_right: '0.25em 0 0.25em 0.5em',
        inner_left: '0.25em 0.5em 0.25em 0'
      }[childAlign.replace('-', '_')];
    }
    return '0.25em 0.5em';
  };
  const showchildAlignSpan = checkAlign => childAlign === checkAlign && children;
  return jsxRuntime.jsxs("div", {
    className: cdsComponentUtilities.cx(headerBadge_module["default"], 'cds-header-badge-wrapper', className),
    style: {
      padding: getDivPadding()
    },
    children: [showchildAlignSpan('left') ? jsxRuntime.jsx("span", {
      className: cdsComponentUtilities.cx(headerBadge_module["default"], 'cds-header-badge-align-left'),
      children: children
    }) : null, jsxRuntime.jsxs("div", {
      className: cdsComponentUtilities.cx(headerBadge_module["default"], 'cds-header-badge-content-wrapper'),
      "data-cds-sentiment": sentiment,
      "data-cds-prominence": prominence,
      style: {
        fontSize: index.CDSHeaderBadgeSizes[size] + 'px'
      },
      children: [showchildAlignSpan('inner-left') ? jsxRuntime.jsx("span", {
        className: cdsComponentUtilities.cx(headerBadge_module["default"], 'cds-header-badge-align-inner-left'),
        children: children
      }) : null, content ? jsxRuntime.jsx("span", {
        "aria-live": "polite",
        className: cdsComponentUtilities.cx(headerBadge_module["default"], 'cds-header-badge-content'),
        style: {
          padding: getSpanPadding()
        },
        children: content
      }) : !!childAlign && jsxRuntime.jsx("span", {
        className: cdsComponentUtilities.cx(headerBadge_module["default"], 'cds-header-badge-content-empty')
      }), showchildAlignSpan('inner-right') ? jsxRuntime.jsx("span", {
        className: cdsComponentUtilities.cx(headerBadge_module["default"], 'cds-header-badge-align-inner-right'),
        children: children
      }) : null]
    }), showchildAlignSpan('right') ? jsxRuntime.jsx("span", {
      className: cdsComponentUtilities.cx(headerBadge_module["default"], 'cds-header-badge-align-right'),
      children: children
    }) : null]
  });
};

exports.CDSHeaderBadge = CDSHeaderBadge;
exports["default"] = CDSHeaderBadge;
