// @ciscodesignsystems/cds-react-menu/menu.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var reactDropdownMenu = require('@radix-ui/react-dropdown-menu');

/**
 * MenuContent
 */
const CDSMenuContent = /*#__PURE__*/react.forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return jsxRuntime.jsxs("div", {
    children: [jsxRuntime.jsx("h3", {
      children: "Do not implement - this is for example purposes only"
    }), jsxRuntime.jsx(reactDropdownMenu.DropdownMenuPortal, {
      children: jsxRuntime.jsx(reactDropdownMenu.DropdownMenuContent, {
        ref: ref,
        "data-cds-menu-content": true,
        ...rest,
        children: children
      })
    })]
  });
});
CDSMenuContent.displayName = 'CDSMenuContent';
/**
 * MenuItem
 */
const CDSMenuItem = /*#__PURE__*/react.forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return jsxRuntime.jsx(reactDropdownMenu.DropdownMenuItem, {
    ref: ref,
    "data-cds-menu-item": true,
    ...rest,
    children: children
  });
});
CDSMenuItem.displayName = 'CDSMenuItem';
/**
 * MenuTrigger
 */
const CDSMenuTrigger = props => {
  const {
    children,
    ...rest
  } = props;
  return jsxRuntime.jsx(reactDropdownMenu.DropdownMenuTrigger, {
    asChild: true,
    ...rest,
    children: children
  });
};
/**
 * SubMenu
 */
const CDSSubMenuRoot = ({
  children,
  ...rest
}) => {
  return jsxRuntime.jsx(reactDropdownMenu.DropdownMenuSub, {
    ...rest,
    children: children
  });
};
/**
 * SubMenuTrigger
 * TODO @bla - add a caret icon prop
 */
const CDSSubMenuTrigger = /*#__PURE__*/react.forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return jsxRuntime.jsxs(reactDropdownMenu.DropdownMenuSubTrigger, {
    ref: ref,
    "data-cds-menu-item": true,
    "data-cds-sub-menu-item": true,
    ...rest,
    children: [children, jsxRuntime.jsx(ArrowRightIcon, {})]
  });
});
CDSSubMenuTrigger.displayName = 'CDSSubMenuTrigger';
/**
 * SubMenuContent
 */
const CDSSubMenuContent = /*#__PURE__*/react.forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return jsxRuntime.jsx(reactDropdownMenu.DropdownMenuPortal, {
    children: jsxRuntime.jsx(reactDropdownMenu.DropdownMenuSubContent, {
      ref: ref,
      "data-cds-menu-content": true,
      ...rest,
      children: children
    })
  });
});
CDSSubMenuContent.displayName = 'CDSSubMenuContent';
const ArrowRightIcon = () => {
  return jsxRuntime.jsxs("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    width: "16",
    height: "16",
    fill: "#000000",
    viewBox: "0 0 256 256",
    children: [jsxRuntime.jsx("rect", {
      width: "256",
      height: "256",
      fill: "none"
    }), jsxRuntime.jsx("line", {
      x1: "40",
      y1: "128",
      x2: "216",
      y2: "128",
      fill: "none",
      stroke: "#000000",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "16"
    }), jsxRuntime.jsx("polyline", {
      points: "144 56 216 128 144 200",
      fill: "none",
      stroke: "#000000",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "16"
    })]
  });
};

Object.defineProperty(exports, 'CDSMenuRoot', {
  enumerable: true,
  get: function () { return reactDropdownMenu.DropdownMenu; }
});
exports.CDSMenuContent = CDSMenuContent;
exports.CDSMenuItem = CDSMenuItem;
exports.CDSMenuTrigger = CDSMenuTrigger;
exports.CDSSubMenuContent = CDSSubMenuContent;
exports.CDSSubMenuRoot = CDSSubMenuRoot;
exports.CDSSubMenuTrigger = CDSSubMenuTrigger;
