// @ciscodesignsystems/cds-react-tag/injected/components/add-tag/index.esm.js
import { forwardRef, useState } from 'react';
import { Plus } from 'phosphor-react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './add-tag.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';

const CDSTagAdd = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children = 'Add',
    onSubmit,
    className,
    size = 'md',
    ...rest
  } = props;
  const [value, setValue] = useState('');
  const handleOnSubmit = () => {
    if (value.trim().length === 0) return;
    onSubmit(value);
    setValue('');
  };
  const handleKeyDown = event => {
    if (event.key === 'Enter') {
      event.preventDefault();
      handleOnSubmit();
    }
  };
  const handleOnBlur = () => {
    if (value.trim().length === 0) {
      setValue('');
    }
  };
  const containerClasses = cx(styles, 'cds-tag__add-container', {
    [`cds-tag__add-container--${size}`]: size
  }, className);
  return /*#__PURE__*/jsxs("div", {
    className: containerClasses,
    ref: ref,
    ...rest,
    children: [/*#__PURE__*/jsx("input", {
      className: cx(styles, 'cds-tag__add-input'),
      value: value,
      onChange: event => setValue(event.target.value),
      placeholder: children,
      onKeyDown: handleKeyDown,
      onBlur: handleOnBlur
    }), /*#__PURE__*/jsx(Plus, {
      size: 14,
      weight: "bold",
      className: cx(styles, 'cds-tag__add-icon')
    }), /*#__PURE__*/jsx("div", {
      "aria-hidden": true,
      className: cx(styles, 'cds-tag__add-hidden-text'),
      children: value || children
    })]
  });
});
CDSTagAdd.displayName = 'CDSTag.Add';

export { CDSTagAdd };
