// @ciscodesignsystems/cds-magnetic-theme-web/js/dist/token-theme-classic-light-rgb-es6.js
/**
 * This file is auto-generated. Do not edit it directly.
 */

export const BRAND_BG_DEFAULT = "55,60,66,1";
export const BRAND_BG_HOVER = "55,60,66,1";
export const BRAND_BG_FOCUS = "70,76,84,1";
export const BRAND_BG_ACTIVE = "70,76,84,1";
export const BRAND_BG_DISABLED = "225,228,232,1";
export const BRAND_BG_GRADIENT_START = "55,60,66,1";
export const BRAND_BG_GRADIENT_END = "15,18,20,1";
export const BRAND_BG_STRONG_DEFAULT = "55,60,66,1";
export const BRAND_BG_STRONG_HOVER = "55,60,66,1";
export const BRAND_BG_STRONG_FOCUS = "55,60,66,1";
export const BRAND_BG_STRONG_ACTIVE = "55,60,66,1";
export const BRAND_TEXT_DEFAULT = "247,247,247,1";
export const BRAND_TEXT_HOVER = "247,247,247,1";
export const BRAND_TEXT_ACTIVE = "247,247,247,1";
export const BRAND_TEXT_WEAK_DEFAULT = "136,144,153,1";
export const BRAND_TEXT_WEAK_HOVER = "247,247,247,1";
export const BRAND_TEXT_WEAK_FOCUS = "247,247,247,1";
export const BRAND_TEXT_WEAK_ACTIVE = "247,247,247,1";
export const BRAND_TEXT_STRONG_DEFAULT = "240,241,242,1";
export const BRAND_ICON_DEFAULT = "247,247,247,1";
export const BRAND_ICON_HOVER = "255,255,255,1";
export const BRAND_ICON_ACTIVE = "255,255,255,1";
export const BRAND_ICON_WEAK_DEFAULT = "247,247,247,1";
export const BRAND_ICON_WEAK_HOVER = "247,247,247,1";
export const BRAND_ICON_WEAK_FOCUS = "247,247,247,1";
export const BRAND_ICON_WEAK_ACTIVE = "247,247,247,1";
export const BRAND_BORDER_ACTIVE = "204,225,255,1";
export const BRAND_ACCENT_ICON_DEFAULT = "126,134,143,1";
export const BRAND_ACCENT_ICON_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_ICON_ACTIVE = "69,153,31,1";
export const BRAND_ACCENT_ICON_DISABLED = "208,212,217,1";
export const BRAND_ACCENT_ICON_WEAK_DEFAULT = "193,198,204,1";
export const BRAND_ACCENT_ICON_WEAK_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_ICON_WEAK_ACTIVE = "107,191,65,1";
export const BRAND_ACCENT_ICON_WEAK_DISABLED = "208,212,217,1";
export const BRAND_ACCENT_ICON_MEDIUM_DEFAULT = "167,173,181,1";
export const BRAND_ACCENT_ICON_MEDIUM_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_ICON_MEDIUM_ACTIVE = "82,166,43,1";
export const BRAND_ACCENT_ICON_MEDIUM_DISABLED = "208,212,217,1";
export const BRAND_ACCENT_BORDER_DEFAULT = "82,166,43,1";
export const BRAND_ACCENT_BORDER_HOVER = "225,228,232,1";
export const BRAND_ACCENT_BORDER_FOCUS = "69,153,31,1";
export const BRAND_ACCENT_BORDER_ACTIVE = "69,153,31,1";
export const BRAND_ACCENT_BORDER_WEAK_DEFAULT = "208,212,217,1";
export const BRAND_ACCENT_BORDER_WEAK_HOVER = "255,255,255,1";
export const BRAND_ACCENT_BORDER_WEAK_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_BORDER_WEAK_ACTIVE = "107,191,65,1";
export const BRAND_ACCENT_BORDER_WEAK_DISABLED = "255,255,255,1";
export const BRAND_ACCENT_BORDER_MEDIUM_DEFAULT = "240,241,242,1";
export const BRAND_ACCENT_BORDER_MEDIUM_HOVER = "255,255,255,1";
export const BRAND_ACCENT_BORDER_MEDIUM_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_BORDER_MEDIUM_ACTIVE = "224,245,213,1";
export const BRAND_ACCENT_BORDER_MEDIUM_DISABLED = "255,255,255,1";
export const BRAND_ACCENT_BORDER_STRONG_DEFAULT = "247,247,247,1";
export const BRAND_ACCENT_BORDER_STRONG_HOVER = "247,247,247,1";
export const BRAND_ACCENT_BORDER_STRONG_FOCUS = "247,247,247,1";
export const BRAND_ACCENT_BORDER_STRONG_ACTIVE = "247,247,247,1";
export const BRAND_ACCENT_BORDER_STRONG_DISABLED = "255,255,255,1";
export const BASE_BG_DEFAULT = "247,247,247,1";
export const BASE_BG_HOVER = "247,247,247,1";
export const BASE_BG_ACTIVE = "247,247,247,1";
export const BASE_BG_DISABLED = "240,241,242,1";
export const BASE_BG_WEAK_DEFAULT = "255,255,255,1";
export const BASE_BG_WEAK_HOVER = "225,228,232,1";
export const BASE_BG_WEAK_ACTIVE = "240,241,242,1";
export const BASE_BG_WEAK_DISABLED = "240,241,242,1";
export const BASE_BG_MEDIUM_DEFAULT = "247,247,247,1";
export const BASE_BG_MEDIUM_HOVER = "208,212,217,1";
export const BASE_BG_MEDIUM_ACTIVE = "225,228,232,1";
export const BASE_BG_MEDIUM_DISABLED = "247,247,247,1";
export const BASE_BG_STRONG_DEFAULT = "101,108,117,1";
export const BASE_BG_STRONG_HOVER = "35,40,46,1";
export const BASE_BG_STRONG_FOCUS = "35,40,46,1";
export const BASE_BG_STRONG_ACTIVE = "35,40,46,1";
export const BASE_BG_STRONG_DISABLED = "208,212,217,1";
export const BASE_ICON_IN_DEFAULT = "255,255,255,1";
export const BASE_ICON_IN_HOVER = "255,255,255,1";
export const BASE_ICON_IN_FOCUS = "255,255,255,1";
export const BASE_ICON_IN_ACTIVE = "255,255,255,1";
export const BASE_ICON_IN_DISABLED = "255,255,255,1";
export const BASE_TEXT_DEFAULT = "35,40,46,1";
export const BASE_TEXT_DISABLED = "193,198,204,1";
export const BASE_TEXT_WEAK_DEFAULT = "136,144,153,1";
export const BASE_TEXT_WEAK_DISABLED = "193,198,204,1";
export const BASE_TEXT_MEDIUM_DEFAULT = "89,96,105,1";
export const BASE_TEXT_MEDIUM_DISABLED = "193,198,204,1";
export const BASE_TEXT_STRONG_DEFAULT = "55,60,66,1";
export const BASE_TEXT_STRONG_DISABLED = "193,198,204,1";
export const BASE_TEXT_IN_DEFAULT = "255,255,255,1";
export const BASE_TEXT_IN_HOVER = "255,255,255,1";
export const BASE_TEXT_IN_FOCUS = "255,255,255,1";
export const BASE_TEXT_IN_ACTIVE = "255,255,255,1";
export const BASE_TEXT_IN_DISABLED = "255,255,255,1";
export const BASE_TEXT_IN_WEAK_DEFAULT = "247,247,247,1";
export const BASE_BORDER_DEFAULT = "225,228,232,1";
export const BASE_BORDER_WEAK_DEFAULT = "255,255,255,1";
export const BASE_BORDER_MEDIUM_DEFAULT = "225,228,232,1";
export const BASE_BORDER_STRONG_DEFAULT = "101,108,117,1";
export const INVERSE_BG_DEFAULT = "55,60,66,1";
export const INVERSE_BG_HOVER = "55,60,66,1";
export const INVERSE_BG_FOCUS = "70,76,84,1";
export const INVERSE_BG_ACTIVE = "70,76,84,1";
export const INVERSE_BG_DISABLED = "225,228,232,1";
export const INVERSE_BG_GRADIENT_START = "55,60,66,1";
export const INVERSE_BG_GRADIENT_END = "15,18,20,1";
export const INVERSE_BG_WEAK_DEFAULT = "55,60,66,1";
export const INVERSE_TEXT_DEFAULT = "247,247,247,1";
export const INVERSE_TEXT_HOVER = "247,247,247,1";
export const INVERSE_TEXT_FOCUS = "247,247,247,1";
export const INVERSE_TEXT_ACTIVE = "247,247,247,1";
export const INVERSE_TEXT_DISABLED = "101,108,117,1";
export const INVERSE_TEXT_MEDIUM_DEFAULT = "193,198,204,1";
export const INVERSE_TEXT_MEDIUM_DISABLED = "101,108,117,1";
export const INVERSE_ICON_DEFAULT = "247,247,247,1";
export const INVERSE_ICON_HOVER = "255,255,255,1";
export const INVERSE_ICON_FOCUS = "255,255,255,1";
export const INVERSE_ICON_ACTIVE = "255,255,255,1";
export const INVERSE_ICON_DISABLED = "111,118,128,1";
export const INVERSE_ICON_STRONG_DEFAULT = "255,255,255,1";
export const INVERSE_ICON_STRONG_HOVER = "255,255,255,1";
export const INVERSE_ICON_STRONG_FOCUS = "255,255,255,1";
export const INVERSE_ICON_STRONG_ACTIVE = "255,255,255,1";
export const INVERSE_ICON_STRONG_DISABLED = "255,255,255,1";
export const INVERSE_BORDER_DEFAULT = "111,118,128,1";
export const INVERSE_BORDER_HOVER = "100,158,245,1";
export const INVERSE_BORDER_FOCUS = "204,225,255,1";
export const INVERSE_BORDER_ACTIVE = "204,225,255,1";
export const INVERSE_BORDER_DISABLED = "225,228,232,1";
export const CONTROL_BG_DEFAULT = "255,255,255,1";
export const CONTROL_BG_HOVER = "227,238,255,1";
export const CONTROL_BG_FOCUS = "240,241,242,1";
export const CONTROL_BG_ACTIVE = "240,241,242,1";
export const CONTROL_BG_DISABLED = "247,247,247,1";
export const CONTROL_BG_WEAK_DEFAULT = "255,255,255,1";
export const CONTROL_BG_WEAK_HOVER = "204,225,255,1";
export const CONTROL_BG_WEAK_FOCUS = "227,238,255,1";
export const CONTROL_BG_WEAK_ACTIVE = "227,238,255,1";
export const CONTROL_BG_WEAK_DISABLED = "247,247,247,1";
export const CONTROL_BG_MEDIUM_DEFAULT = "255,255,255,1";
export const CONTROL_BG_MEDIUM_HOVER = "247,247,247,1";
export const CONTROL_BG_MEDIUM_FOCUS = "227,238,255,1";
export const CONTROL_BG_MEDIUM_ACTIVE = "227,238,255,1";
export const CONTROL_BG_MEDIUM_DISABLED = "225,228,232,1";
export const CONTROL_BG_STRONG_DEFAULT = "39,116,217,1";
export const CONTROL_BG_STRONG_HOVER = "13,92,189,1";
export const CONTROL_BG_STRONG_FOCUS = "13,92,189,1";
export const CONTROL_BG_STRONG_ACTIVE = "13,92,189,1";
export const CONTROL_BG_STRONG_DISABLED = "186,214,255,1";
export const CONTROL_ICON_DEFAULT = "29,105,204,1";
export const CONTROL_ICON_HOVER = "13,92,189,1";
export const CONTROL_ICON_FOCUS = "13,92,189,1";
export const CONTROL_ICON_ACTIVE = "13,92,189,1";
export const CONTROL_ICON_DISABLED = "163,200,255,1";
export const CONTROL_ICON_WEAK_DEFAULT = "126,134,143,1";
export const CONTROL_ICON_WEAK_HOVER = "39,116,217,1";
export const CONTROL_ICON_WEAK_FOCUS = "39,116,217,1";
export const CONTROL_ICON_WEAK_ACTIVE = "39,116,217,1";
export const CONTROL_ICON_WEAK_DISABLED = "208,212,217,1";
export const CONTROL_ICON_MEDIUM_DEFAULT = "39,116,217,1";
export const CONTROL_ICON_MEDIUM_HOVER = "13,92,189,1";
export const CONTROL_ICON_MEDIUM_FOCUS = "13,92,189,1";
export const CONTROL_ICON_MEDIUM_ACTIVE = "13,92,189,1";
export const CONTROL_ICON_MEDIUM_DISABLED = "163,200,255,1";
export const CONTROL_ICON_STRONG_DEFAULT = "101,108,117,1";
export const CONTROL_ICON_STRONG_HOVER = "70,76,84,1";
export const CONTROL_ICON_STRONG_FOCUS = "29,105,204,1";
export const CONTROL_ICON_STRONG_ACTIVE = "29,105,204,1";
export const CONTROL_ICON_STRONG_DISABLED = "193,198,204,1";
export const CONTROL_ICON_IN_DEFAULT = "255,255,255,1";
export const CONTROL_ICON_IN_HOVER = "255,255,255,1";
export const CONTROL_ICON_IN_FOCUS = "255,255,255,1";
export const CONTROL_ICON_IN_ACTIVE = "255,255,255,1";
export const CONTROL_ICON_IN_DISABLED = "193,198,204,1";
export const CONTROL_ICON_IN_STRONG_DEFAULT = "39,116,217,1";
export const CONTROL_ICON_IN_STRONG_HOVER = "39,116,217,1";
export const CONTROL_ICON_IN_STRONG_FOCUS = "39,116,217,1";
export const CONTROL_ICON_IN_STRONG_ACTIVE = "39,116,217,1";
export const CONTROL_ICON_IN_STRONG_DISABLED = "193,198,204,1";
export const CONTROL_BORDER_DEFAULT = "136,144,153,1";
export const CONTROL_BORDER_HOVER = "62,132,229,1";
export const CONTROL_BORDER_FOCUS = "62,132,229,1";
export const CONTROL_BORDER_ACTIVE = "62,132,229,1";
export const CONTROL_BORDER_DISABLED = "208,212,217,1";
export const CONTROL_BORDER_WEAK_DEFAULT = "225,228,232,1";
export const CONTROL_BORDER_WEAK_HOVER = "124,173,247,1";
export const CONTROL_BORDER_WEAK_FOCUS = "124,173,247,1";
export const CONTROL_BORDER_WEAK_ACTIVE = "124,173,247,1";
export const CONTROL_BORDER_WEAK_DISABLED = "208,212,217,1";
export const CONTROL_BORDER_MEDIUM_DEFAULT = "39,116,217,1";
export const CONTROL_BORDER_MEDIUM_HOVER = "186,214,255,1";
export const CONTROL_BORDER_MEDIUM_FOCUS = "39,116,217,1";
export const CONTROL_BORDER_MEDIUM_ACTIVE = "39,116,217,1";
export const CONTROL_BORDER_MEDIUM_DISABLED = "208,212,217,1";
export const CONTROL_BORDER_STRONG_DEFAULT = "101,108,117,1";
export const CONTROL_BORDER_STRONG_HOVER = "101,108,117,1";
export const CONTROL_BORDER_STRONG_FOCUS = "39,116,217,1";
export const CONTROL_BORDER_STRONG_ACTIVE = "39,116,217,1";
export const CONTROL_BORDER_STRONG_DISABLED = "208,212,217,1";
export const INTERACT_BG_DEFAULT = "29,105,204,1";
export const INTERACT_BG_HOVER = "13,92,189,1";
export const INTERACT_BG_FOCUS = "13,92,189,1";
export const INTERACT_BG_ACTIVE = "13,92,189,1";
export const INTERACT_BG_DISABLED = "163,200,255,1";
export const INTERACT_BG_WEAK_DEFAULT = "255,255,255,1";
export const INTERACT_BG_WEAK_HOVER = "204,225,255,1";
export const INTERACT_BG_WEAK_FOCUS = "204,225,255,1";
export const INTERACT_BG_WEAK_ACTIVE = "204,225,255,1";
export const INTERACT_BG_WEAK_DISABLED = "255,255,255,1";
export const INTERACT_BG_MEDIUM_DEFAULT = "39,116,217,1";
export const INTERACT_BG_MEDIUM_ACTIVE = "204,225,255,1";
export const INTERACT_BG_STRONG_ACTIVE = "247,247,247,1";
export const INTERACT_BG_STRONG_HOVER = "247,247,247,1";
export const INTERACT_TEXT_DEFAULT = "29,105,204,1";
export const INTERACT_TEXT_HOVER = "13,92,189,1";
export const INTERACT_TEXT_FOCUS = "13,92,189,1";
export const INTERACT_TEXT_ACTIVE = "13,92,189,1";
export const INTERACT_TEXT_DISABLED = "163,200,255,1";
export const INTERACT_TEXT_WEAK_DEFAULT = "101,108,117,1";
export const INTERACT_TEXT_WEAK_HOVER = "29,105,204,1";
export const INTERACT_TEXT_WEAK_FOCUS = "29,105,204,1";
export const INTERACT_TEXT_WEAK_ACTIVE = "29,105,204,1";
export const INTERACT_TEXT_WEAK_DISABLED = "193,198,204,1";
export const INTERACT_TEXT_IN_DEFAULT = "255,255,255,1";
export const INTERACT_TEXT_IN_HOVER = "255,255,255,1";
export const INTERACT_TEXT_IN_FOCUS = "255,255,255,1";
export const INTERACT_TEXT_IN_ACTIVE = "255,255,255,1";
export const INTERACT_TEXT_IN_DISABLED = "255,255,255,1";
export const INTERACT_ICON_DEFAULT = "29,105,204,1";
export const INTERACT_ICON_HOVER = "13,92,189,1";
export const INTERACT_ICON_FOCUS = "13,92,189,1";
export const INTERACT_ICON_ACTIVE = "13,92,189,1";
export const INTERACT_ICON_DISABLED = "163,200,255,1";
export const INTERACT_ICON_WEAK_DEFAULT = "126,134,143,1";
export const INTERACT_ICON_WEAK_HOVER = "111,118,128,1";
export const INTERACT_ICON_WEAK_FOCUS = "111,118,128,1";
export const INTERACT_ICON_WEAK_ACTIVE = "111,118,128,1";
export const INTERACT_ICON_WEAK_DISABLED = "208,212,217,1";
export const INTERACT_ICON_MEDIUM_DEFAULT = "126,134,143,1";
export const INTERACT_ICON_MEDIUM_HOVER = "39,116,217,1";
export const INTERACT_ICON_MEDIUM_FOCUS = "39,116,217,1";
export const INTERACT_ICON_MEDIUM_ACTIVE = "39,116,217,1";
export const INTERACT_ICON_MEDIUM_DISABLED = "193,198,204,1";
export const INTERACT_ICON_STRONG_DEFAULT = "101,108,117,1";
export const INTERACT_ICON_STRONG_HOVER = "70,76,84,1";
export const INTERACT_ICON_STRONG_FOCUS = "70,76,84,1";
export const INTERACT_ICON_STRONG_ACTIVE = "70,76,84,1";
export const INTERACT_ICON_STRONG_DISABLED = "167,173,181,1";
export const INTERACT_ICON_IN_DEFAULT = "255,255,255,1";
export const INTERACT_ICON_IN_HOVER = "255,255,255,1";
export const INTERACT_ICON_IN_FOCUS = "255,255,255,1";
export const INTERACT_ICON_IN_ACTIVE = "255,255,255,1";
export const INTERACT_ICON_IN_DISABLED = "255,255,255,1";
export const INTERACT_BORDER_DEFAULT = "29,105,204,1";
export const INTERACT_BORDER_HOVER = "13,92,189,1";
export const INTERACT_BORDER_FOCUS = "13,92,189,1";
export const INTERACT_BORDER_ACTIVE = "13,92,189,1";
export const INTERACT_BORDER_DISABLED = "163,200,255,1";
export const INTERACT_BORDER_WEAK_DEFAULT = "225,228,232,1";
export const INTERACT_BORDER_WEAK_HOVER = "204,225,255,1";
export const INTERACT_BORDER_WEAK_FOCUS = "225,228,232,1";
export const INTERACT_BORDER_WEAK_ACTIVE = "225,228,232,1";
export const INTERACT_BORDER_WEAK_DISABLED = "225,228,232,1";
export const INTERACT_BORDER_MEDIUM_DEFAULT = "39,116,217,1";
export const INTERACT_BORDER_MEDIUM_HOVER = "186,214,255,1";
export const INTERACT_BORDER_MEDIUM_FOCUS = "39,116,217,1";
export const INTERACT_BORDER_MEDIUM_ACTIVE = "39,116,217,1";
export const INTERACT_BORDER_MEDIUM_DISABLED = "225,228,232,1";
export const POSITIVE_BG_DEFAULT = "69,153,31,1";
export const POSITIVE_BG_HOVER = "69,153,31,1";
export const POSITIVE_BG_FOCUS = "69,153,31,1";
export const POSITIVE_BG_ACTIVE = "69,153,31,1";
export const POSITIVE_BG_DISABLED = "176,227,150,1";
export const POSITIVE_BG_WEAK_DEFAULT = "255,255,255,1";
export const POSITIVE_BG_WEAK_HOVER = "255,255,255,1";
export const POSITIVE_BG_WEAK_FOCUS = "255,255,255,1";
export const POSITIVE_BG_WEAK_ACTIVE = "255,255,255,1";
export const POSITIVE_BG_WEAK_DISABLED = "255,255,255,1";
export const POSITIVE_BG_MEDIUM_DEFAULT = "224,245,213,1";
export const POSITIVE_BG_MEDIUM_HOVER = "224,245,213,1";
export const POSITIVE_BG_MEDIUM_FOCUS = "224,245,213,1";
export const POSITIVE_BG_MEDIUM_ACTIVE = "224,245,213,1";
export const POSITIVE_BG_MEDIUM_DISABLED = "247,247,247,1";
export const POSITIVE_BG_STRONG_DEFAULT = "224,245,213,1";
export const POSITIVE_ICON_DEFAULT = "69,153,31,1";
export const POSITIVE_ICON_HOVER = "69,153,31,1";
export const POSITIVE_ICON_FOCUS = "69,153,31,1";
export const POSITIVE_ICON_ACTIVE = "69,153,31,1";
export const POSITIVE_ICON_DISABLED = "208,212,217,1";
export const POSITIVE_ICON_WEAK_DEFAULT = "176,227,150,1";
export const POSITIVE_ICON_WEAK_HOVER = "176,227,150,1";
export const POSITIVE_ICON_WEAK_FOCUS = "176,227,150,1";
export const POSITIVE_ICON_WEAK_ACTIVE = "176,227,150,1";
export const POSITIVE_ICON_WEAK_DISABLED = "176,227,150,1";
export const POSITIVE_ICON_STRONG_DEFAULT = "69,153,31,1";
export const POSITIVE_ICON_STRONG_HOVER = "69,153,31,1";
export const POSITIVE_ICON_STRONG_FOCUS = "69,153,31,1";
export const POSITIVE_ICON_STRONG_ACTIVE = "69,153,31,1";
export const POSITIVE_ICON_STRONG_DISABLED = "208,212,217,1";
export const POSITIVE_ICON_IN_DEFAULT = "255,255,255,1";
export const POSITIVE_ICON_IN_HOVER = "255,255,255,1";
export const POSITIVE_ICON_IN_FOCUS = "255,255,255,1";
export const POSITIVE_ICON_IN_ACTIVE = "255,255,255,1";
export const POSITIVE_ICON_IN_DISABLED = "255,255,255,1";
export const POSITIVE_BORDER_DEFAULT = "69,153,31,1";
export const POSITIVE_BORDER_HOVER = "69,153,31,1";
export const POSITIVE_BORDER_FOCUS = "69,153,31,1";
export const POSITIVE_BORDER_ACTIVE = "69,153,31,1";
export const POSITIVE_BORDER_DISABLED = "208,212,217,1";
export const POSITIVE_BORDER_WEAK_DEFAULT = "255,255,255,1";
export const POSITIVE_BORDER_WEAK_HOVER = "255,255,255,1";
export const POSITIVE_BORDER_WEAK_FOCUS = "255,255,255,1";
export const POSITIVE_BORDER_WEAK_ACTIVE = "255,255,255,1";
export const POSITIVE_BORDER_WEAK_DISABLED = "208,212,217,1";
export const POSITIVE_BORDER_STRONG_DEFAULT = "57,133,25,1";
export const POSITIVE_BORDER_STRONG_HOVER = "57,133,25,1";
export const POSITIVE_BORDER_STRONG_FOCUS = "57,133,25,1";
export const POSITIVE_BORDER_STRONG_ACTIVE = "57,133,25,1";
export const POSITIVE_BORDER_STRONG_DISABLED = "57,133,25,1";
export const POSITIVE_TEXT_IN_DEFAULT = "255,255,255,1";
export const POSITIVE_TEXT_IN_HOVER = "255,255,255,1";
export const POSITIVE_TEXT_IN_FOCUS = "255,255,255,1";
export const POSITIVE_TEXT_IN_ACTIVE = "255,255,255,1";
export const POSITIVE_TEXT_IN_DISABLED = "255,255,255,1";
export const WARNING_BG_DEFAULT = "204,134,4,1";
export const WARNING_BG_HOVER = "204,134,4,1";
export const WARNING_BG_FOCUS = "204,134,4,1";
export const WARNING_BG_ACTIVE = "204,134,4,1";
export const WARNING_BG_DISABLED = "242,210,104,1";
export const WARNING_BG_WEAK_DEFAULT = "255,255,255,1";
export const WARNING_BG_WEAK_HOVER = "255,255,255,1";
export const WARNING_BG_WEAK_FOCUS = "255,255,255,1";
export const WARNING_BG_WEAK_ACTIVE = "255,255,255,1";
export const WARNING_BG_WEAK_DISABLED = "255,255,255,1";
export const WARNING_BG_MEDIUM_DEFAULT = "250,239,185,1";
export const WARNING_BG_MEDIUM_HOVER = "250,239,185,1";
export const WARNING_BG_MEDIUM_FOCUS = "250,239,185,1";
export const WARNING_BG_MEDIUM_ACTIVE = "250,239,185,1";
export const WARNING_BG_MEDIUM_DISABLED = "247,247,247,1";
export const WARNING_BG_STRONG_DEFAULT = "250,239,185,1";
export const WARNING_ICON_DEFAULT = "204,134,4,1";
export const WARNING_ICON_HOVER = "204,134,4,1";
export const WARNING_ICON_FOCUS = "204,134,4,1";
export const WARNING_ICON_ACTIVE = "204,134,4,1";
export const WARNING_ICON_DISABLED = "208,212,217,1";
export const WARNING_ICON_STRONG_DEFAULT = "189,114,2,1";
export const WARNING_ICON_STRONG_HOVER = "189,114,2,1";
export const WARNING_ICON_STRONG_FOCUS = "189,114,2,1";
export const WARNING_ICON_STRONG_ACTIVE = "189,114,2,1";
export const WARNING_ICON_STRONG_DISABLED = "208,212,217,1";
export const WARNING_ICON_IN_DEFAULT = "255,255,255,1";
export const WARNING_ICON_IN_HOVER = "255,255,255,1";
export const WARNING_ICON_IN_FOCUS = "255,255,255,1";
export const WARNING_ICON_IN_ACTIVE = "255,255,255,1";
export const WARNING_ICON_IN_DISABLED = "255,255,255,1";
export const WARNING_BORDER_DEFAULT = "204,134,4,1";
export const WARNING_BORDER_HOVER = "204,134,4,1";
export const WARNING_BORDER_FOCUS = "204,134,4,1";
export const WARNING_BORDER_ACTIVE = "204,134,4,1";
export const WARNING_BORDER_DISABLED = "208,212,217,1";
export const WARNING_BORDER_WEAK_DEFAULT = "255,255,255,1";
export const WARNING_BORDER_WEAK_HOVER = "255,255,255,1";
export const WARNING_BORDER_WEAK_FOCUS = "255,255,255,1";
export const WARNING_BORDER_WEAK_ACTIVE = "255,255,255,1";
export const WARNING_BORDER_WEAK_DISABLED = "208,212,217,1";
export const WARNING_BORDER_STRONG_DEFAULT = "166,85,3,1";
export const WARNING_BORDER_STRONG_HOVER = "166,85,3,1";
export const WARNING_BORDER_STRONG_FOCUS = "166,85,3,1";
export const WARNING_BORDER_STRONG_ACTIVE = "166,85,3,1";
export const WARNING_BORDER_STRONG_DISABLED = "166,85,3,1";
export const WARNING_TEXT_IN_DEFAULT = "255,255,255,1";
export const WARNING_TEXT_IN_HOVER = "255,255,255,1";
export const WARNING_TEXT_IN_FOCUS = "255,255,255,1";
export const WARNING_TEXT_IN_ACTIVE = "255,255,255,1";
export const WARNING_TEXT_IN_DISABLED = "255,255,255,1";
export const SEVERE_WARNING_BG_DEFAULT = "242,103,34,1";
export const SEVERE_WARNING_BG_HOVER = "242,103,34,1";
export const SEVERE_WARNING_BG_FOCUS = "242,103,34,1";
export const SEVERE_WARNING_BG_ACTIVE = "242,103,34,1";
export const SEVERE_WARNING_BG_DISABLED = "252,201,167,1";
export const SEVERE_WARNING_BG_WEAK_DEFAULT = "255,255,255,1";
export const SEVERE_WARNING_BG_WEAK_HOVER = "255,255,255,1";
export const SEVERE_WARNING_BG_WEAK_FOCUS = "255,255,255,1";
export const SEVERE_WARNING_BG_WEAK_ACTIVE = "255,255,255,1";
export const SEVERE_WARNING_BG_WEAK_DISABLED = "255,255,255,1";
export const SEVERE_WARNING_BG_MEDIUM_DEFAULT = "255,234,219,1";
export const SEVERE_WARNING_BG_MEDIUM_HOVER = "255,234,219,1";
export const SEVERE_WARNING_BG_MEDIUM_FOCUS = "255,234,219,1";
export const SEVERE_WARNING_BG_MEDIUM_ACTIVE = "255,234,219,1";
export const SEVERE_WARNING_BG_MEDIUM_DISABLED = "247,247,247,1";
export const SEVERE_WARNING_BG_STRONG_DEFAULT = "255,234,219,1";
export const SEVERE_WARNING_ICON_DEFAULT = "242,103,34,1";
export const SEVERE_WARNING_ICON_HOVER = "217,90,26,1";
export const SEVERE_WARNING_ICON_FOCUS = "196,79,20,1";
export const SEVERE_WARNING_ICON_ACTIVE = "196,79,20,1";
export const SEVERE_WARNING_ICON_DISABLED = "208,212,217,1";
export const SEVERE_WARNING_ICON_STRONG_DEFAULT = "217,90,26,1";
export const SEVERE_WARNING_ICON_STRONG_HOVER = "217,90,26,1";
export const SEVERE_WARNING_ICON_STRONG_FOCUS = "217,90,26,1";
export const SEVERE_WARNING_ICON_STRONG_ACTIVE = "217,90,26,1";
export const SEVERE_WARNING_ICON_STRONG_DISABLED = "217,90,26,1";
export const SEVERE_WARNING_ICON_IN_DEFAULT = "255,255,255,1";
export const SEVERE_WARNING_ICON_IN_HOVER = "255,255,255,1";
export const SEVERE_WARNING_ICON_IN_FOCUS = "255,255,255,1";
export const SEVERE_WARNING_ICON_IN_ACTIVE = "255,255,255,1";
export const SEVERE_WARNING_ICON_IN_DISABLED = "255,255,255,1";
export const SEVERE_WARNING_BORDER_DEFAULT = "242,103,34,1";
export const SEVERE_WARNING_BORDER_HOVER = "242,103,34,1";
export const SEVERE_WARNING_BORDER_FOCUS = "242,103,34,1";
export const SEVERE_WARNING_BORDER_ACTIVE = "242,103,34,1";
export const SEVERE_WARNING_BORDER_DISABLED = "242,103,34,1";
export const SEVERE_WARNING_BORDER_WEAK_DEFAULT = "255,255,255,1";
export const SEVERE_WARNING_BORDER_WEAK_HOVER = "255,255,255,1";
export const SEVERE_WARNING_BORDER_WEAK_FOCUS = "255,255,255,1";
export const SEVERE_WARNING_BORDER_WEAK_ACTIVE = "255,255,255,1";
export const SEVERE_WARNING_BORDER_WEAK_DISABLED = "208,212,217,1";
export const NEGATIVE_BG_DEFAULT = "204,45,55,1";
export const NEGATIVE_BG_HOVER = "204,45,55,1";
export const NEGATIVE_BG_FOCUS = "204,45,55,1";
export const NEGATIVE_BG_ACTIVE = "204,45,55,1";
export const NEGATIVE_BG_DISABLED = "255,178,181,1";
export const NEGATIVE_BG_WEAK_DEFAULT = "255,255,255,1";
export const NEGATIVE_BG_WEAK_HOVER = "255,212,213,1";
export const NEGATIVE_BG_WEAK_FOCUS = "255,212,213,1";
export const NEGATIVE_BG_WEAK_ACTIVE = "255,212,213,1";
export const NEGATIVE_BG_WEAK_DISABLED = "255,255,255,1";
export const NEGATIVE_BG_MEDIUM_DEFAULT = "255,232,233,1";
export const NEGATIVE_BG_MEDIUM_HOVER = "255,232,233,1";
export const NEGATIVE_BG_MEDIUM_FOCUS = "255,232,233,1";
export const NEGATIVE_BG_MEDIUM_ACTIVE = "255,232,233,1";
export const NEGATIVE_BG_MEDIUM_DISABLED = "247,247,247,1";
export const NEGATIVE_BG_STRONG_DEFAULT = "255,232,233,1";
export const NEGATIVE_TEXT_DEFAULT = "204,45,55,1";
export const NEGATIVE_TEXT_HOVER = "178,36,45,1";
export const NEGATIVE_TEXT_FOCUS = "178,36,45,1";
export const NEGATIVE_TEXT_ACTIVE = "178,36,45,1";
export const NEGATIVE_TEXT_DISABLED = "255,178,181,1";
export const NEGATIVE_TEXT_IN_DEFAULT = "255,255,255,1";
export const NEGATIVE_TEXT_IN_HOVER = "255,255,255,1";
export const NEGATIVE_TEXT_IN_FOCUS = "255,255,255,1";
export const NEGATIVE_TEXT_IN_ACTIVE = "255,255,255,1";
export const NEGATIVE_TEXT_IN_DISABLED = "255,255,255,1";
export const NEGATIVE_ICON_DEFAULT = "204,45,55,1";
export const NEGATIVE_ICON_HOVER = "178,36,45,1";
export const NEGATIVE_ICON_FOCUS = "178,36,45,1";
export const NEGATIVE_ICON_ACTIVE = "178,36,45,1";
export const NEGATIVE_ICON_DISABLED = "255,178,181,1";
export const NEGATIVE_ICON_WEAK_DEFAULT = "255,199,201,1";
export const NEGATIVE_ICON_WEAK_HOVER = "255,199,201,1";
export const NEGATIVE_ICON_WEAK_FOCUS = "255,199,201,1";
export const NEGATIVE_ICON_WEAK_ACTIVE = "255,199,201,1";
export const NEGATIVE_ICON_WEAK_DISABLED = "208,212,217,1";
export const NEGATIVE_ICON_IN_DEFAULT = "255,255,255,1";
export const NEGATIVE_ICON_IN_HOVER = "255,255,255,1";
export const NEGATIVE_ICON_IN_FOCUS = "255,255,255,1";
export const NEGATIVE_ICON_IN_ACTIVE = "255,255,255,1";
export const NEGATIVE_ICON_IN_DISABLED = "255,255,255,1";
export const NEGATIVE_BORDER_DEFAULT = "204,45,55,1";
export const NEGATIVE_BORDER_HOVER = "178,36,45,1";
export const NEGATIVE_BORDER_FOCUS = "178,36,45,1";
export const NEGATIVE_BORDER_ACTIVE = "178,36,45,1";
export const NEGATIVE_BORDER_DISABLED = "255,178,181,1";
export const NEGATIVE_BORDER_WEAK_DEFAULT = "255,255,255,1";
export const NEGATIVE_BORDER_WEAK_HOVER = "255,255,255,1";
export const NEGATIVE_BORDER_WEAK_FOCUS = "255,255,255,1";
export const NEGATIVE_BORDER_WEAK_ACTIVE = "255,255,255,1";
export const NEGATIVE_BORDER_WEAK_DISABLED = "208,212,217,1";
export const NEGATIVE_BORDER_MEDIUM_DEFAULT = "217,56,67,1";
export const NEGATIVE_BORDER_MEDIUM_HOVER = "217,56,67,1";
export const NEGATIVE_BORDER_MEDIUM_FOCUS = "217,56,67,1";
export const NEGATIVE_BORDER_MEDIUM_ACTIVE = "217,56,67,1";
export const NEGATIVE_BORDER_MEDIUM_DISABLED = "255,178,181,1";
export const NEGATIVE_BORDER_STRONG_DEFAULT = "178,36,45,1";
export const NEGATIVE_BORDER_STRONG_HOVER = "178,36,45,1";
export const NEGATIVE_BORDER_STRONG_FOCUS = "178,36,45,1";
export const NEGATIVE_BORDER_STRONG_ACTIVE = "178,36,45,1";
export const NEGATIVE_BORDER_STRONG_DISABLED = "255,178,181,1";
export const INFO_BG_DEFAULT = "39,116,217,1";
export const INFO_BG_HOVER = "39,116,217,1";
export const INFO_BG_FOCUS = "39,116,217,1";
export const INFO_BG_ACTIVE = "39,116,217,1";
export const INFO_BG_DISABLED = "39,116,217,1";
export const INFO_BG_WEAK_DEFAULT = "255,255,255,1";
export const INFO_BG_WEAK_HOVER = "255,255,255,1";
export const INFO_BG_WEAK_FOCUS = "255,255,255,1";
export const INFO_BG_WEAK_ACTIVE = "255,255,255,1";
export const INFO_BG_WEAK_DISABLED = "255,255,255,1";
export const INFO_BG_MEDIUM_DEFAULT = "227,238,255,1";
export const INFO_BG_MEDIUM_HOVER = "227,238,255,1";
export const INFO_BG_MEDIUM_FOCUS = "227,238,255,1";
export const INFO_BG_MEDIUM_ACTIVE = "227,238,255,1";
export const INFO_BG_MEDIUM_DISABLED = "247,247,247,1";
export const INFO_BG_STRONG_DEFAULT = "227,238,255,1";
export const INFO_ICON_DEFAULT = "39,116,217,1";
export const INFO_ICON_HOVER = "39,116,217,1";
export const INFO_ICON_FOCUS = "39,116,217,1";
export const INFO_ICON_ACTIVE = "39,116,217,1";
export const INFO_ICON_DISABLED = "208,212,217,1";
export const INFO_ICON_WEAK_DEFAULT = "163,200,255,1";
export const INFO_ICON_WEAK_HOVER = "163,200,255,1";
export const INFO_ICON_WEAK_FOCUS = "163,200,255,1";
export const INFO_ICON_WEAK_ACTIVE = "163,200,255,1";
export const INFO_ICON_WEAK_DISABLED = "163,200,255,1";
export const INFO_ICON_IN_DEFAULT = "255,255,255,1";
export const INFO_ICON_IN_HOVER = "255,255,255,1";
export const INFO_ICON_IN_FOCUS = "255,255,255,1";
export const INFO_ICON_IN_ACTIVE = "255,255,255,1";
export const INFO_ICON_IN_DISABLED = "255,255,255,1";
export const INFO_BORDER_DEFAULT = "39,116,217,1";
export const INFO_BORDER_HOVER = "39,116,217,1";
export const INFO_BORDER_FOCUS = "39,116,217,1";
export const INFO_BORDER_ACTIVE = "39,116,217,1";
export const INFO_BORDER_DISABLED = "208,212,217,1";
export const INFO_BORDER_WEAK_DEFAULT = "255,255,255,1";
export const INFO_BORDER_WEAK_HOVER = "255,255,255,1";
export const INFO_BORDER_WEAK_FOCUS = "255,255,255,1";
export const INFO_BORDER_WEAK_ACTIVE = "255,255,255,1";
export const INFO_BORDER_WEAK_DISABLED = "208,212,217,1";
export const DORMANT_BG_DEFAULT = "111,118,128,1";
export const DORMANT_BG_DISABLED = "111,118,128,1";
export const DORMANT_BG_MEDIUM_DEFAULT = "225,228,232,1";
export const DORMANT_BG_MEDIUM_DISABLED = "247,247,247,1";
export const DORMANT_ICON_DEFAULT = "111,118,128,1";
export const DORMANT_ICON_DISABLED = "208,212,217,1";
export const DORMANT_ICON_MEDIUM_DEFAULT = "255,255,255,1";
export const DORMANT_ICON_IN_DEFAULT = "255,255,255,1";
export const DORMANT_ICON_IN_DISABLED = "255,255,255,1";
export const DORMANT_ICON_IN_MEDIUM_DEFAULT = "255,255,255,1";
export const DORMANT_BORDER_DEFAULT = "136,144,153,1";
export const DORMANT_BORDER_DISABLED = "208,212,217,1";
export const ACCENT_A_DEFAULT = "125,138,255,1";
export const ACCENT_A_HOVER = "105,119,240,1";
export const ACCENT_A_WEAK_DEFAULT = "235,237,255,1";
export const ACCENT_B_DEFAULT = "176,40,99,1";
export const ACCENT_B_HOVER = "153,29,83,1";
export const ACCENT_B_WEAK_DEFAULT = "255,232,239,1";
export const ACCENT_C_DEFAULT = "242,99,140,1";
export const ACCENT_C_HOVER = "227,68,124,1";
export const ACCENT_C_WEAK_DEFAULT = "255,212,224,1";
export const ACCENT_D_DEFAULT = "117,59,204,1";
export const ACCENT_D_HOVER = "103,50,184,1";
export const ACCENT_D_WEAK_DEFAULT = "243,235,255,1";
export const ACCENT_E_DEFAULT = "125,161,27,1";
export const ACCENT_E_HOVER = "108,140,20,1";
export const ACCENT_E_WEAK_DEFAULT = "234,242,211,1";
export const ACCENT_F_DEFAULT = "173,57,7,1";
export const ACCENT_F_HOVER = "148,46,3,1";
export const ACCENT_F_WEAK_DEFAULT = "255,234,219,1";
export const ACCENT_G_DEFAULT = "4,164,176,1";
export const ACCENT_G_HOVER = "2,142,153,1";
export const ACCENT_G_WEAK_DEFAULT = "213,245,245,1";
export const ACCENT_H_DEFAULT = "0,103,115,1";
export const ACCENT_H_HOVER = "0,92,102,1";
export const ACCENT_H_WEAK_DEFAULT = "169,235,235,1";
export const ACCENT_I_DEFAULT = "232,95,198,1";
export const ACCENT_I_HOVER = "214,73,179,1";
export const ACCENT_I_WEAK_DEFAULT = "255,232,249,1";
export const ACCENT_J_DEFAULT = "84,92,138,1";
export const ACCENT_J_HOVER = "72,79,122,1";
export const ACCENT_J_WEAK_DEFAULT = "235,237,255,1";
export const ACCENT_K_DEFAULT = "33,166,95,1";
export const ACCENT_K_HOVER = "22,152,85,1";
export const ACCENT_K_WEAK_DEFAULT = "212,245,225,1";
export const IN_PROGRESS_BG_MEDIUM_DEFAULT = "227,238,255,1";
export const IN_PROGRESS_BG_MEDIUM_HOVER = "227,238,255,1";
export const IN_PROGRESS_BG_MEDIUM_FOCUS = "227,238,255,1";
export const IN_PROGRESS_BG_MEDIUM_ACTIVE = "227,238,255,1";
export const IN_PROGRESS_BG_MEDIUM_DISABLED = "247,247,247,1";
export const IN_PROGRESS_ICON_DEFAULT = "39,116,217,1";
export const IN_PROGRESS_ICON_HOVER = "39,116,217,1";
export const IN_PROGRESS_ICON_FOCUS = "39,116,217,1";
export const IN_PROGRESS_ICON_ACTIVE = "39,116,217,1";
export const IN_PROGRESS_ICON_DISABLED = "208,212,217,1";
export const IN_PROGRESS_ICON_IN_DEFAULT = "255,255,255,1";
export const IN_PROGRESS_ICON_IN_HOVER = "255,255,255,1";
export const IN_PROGRESS_ICON_IN_FOCUS = "255,255,255,1";
export const IN_PROGRESS_ICON_IN_ACTIVE = "255,255,255,1";
export const IN_PROGRESS_ICON_IN_DISABLED = "255,255,255,1";
export const IN_PROGRESS_BORDER_DEFAULT = "39,116,217,1";
export const IN_PROGRESS_BORDER_HOVER = "39,116,217,1";
export const IN_PROGRESS_BORDER_FOCUS = "39,116,217,1";
export const IN_PROGRESS_BORDER_ACTIVE = "39,116,217,1";
export const IN_PROGRESS_BORDER_DISABLED = "208,212,217,1";
export const DECENT_BG_MEDIUM_DEFAULT = "234,242,211,1";
export const DECENT_BG_MEDIUM_HOVER = "234,242,211,1";
export const DECENT_BG_MEDIUM_FOCUS = "234,242,211,1";
export const DECENT_BG_MEDIUM_ACTIVE = "234,242,211,1";
export const DECENT_BG_MEDIUM_DISABLED = "247,247,247,1";
export const DECENT_ICON_DEFAULT = "108,140,20,1";
export const DECENT_ICON_HOVER = "108,140,20,1";
export const DECENT_ICON_FOCUS = "108,140,20,1";
export const DECENT_ICON_ACTIVE = "108,140,20,1";
export const DECENT_ICON_DISABLED = "208,212,217,1";
export const DECENT_ICON_IN_DEFAULT = "255,255,255,1";
export const DECENT_ICON_IN_HOVER = "255,255,255,1";
export const DECENT_ICON_IN_FOCUS = "255,255,255,1";
export const DECENT_ICON_IN_ACTIVE = "255,255,255,1";
export const DECENT_ICON_IN_DISABLED = "255,255,255,1";
export const DECENT_BORDER_DEFAULT = "125,161,27,1";
export const DECENT_BORDER_HOVER = "125,161,27,1";
export const DECENT_BORDER_FOCUS = "125,161,27,1";
export const DECENT_BORDER_ACTIVE = "125,161,27,1";
export const DECENT_BORDER_DISABLED = "208,212,217,1";
export const EXCELLENT_BG_MEDIUM_DEFAULT = "212,245,225,1";
export const EXCELLENT_BG_MEDIUM_HOVER = "212,245,225,1";
export const EXCELLENT_BG_MEDIUM_FOCUS = "212,245,225,1";
export const EXCELLENT_BG_MEDIUM_ACTIVE = "212,245,225,1";
export const EXCELLENT_BG_MEDIUM_DISABLED = "247,247,247,1";
export const EXCELLENT_ICON_DEFAULT = "8,112,65,1";
export const EXCELLENT_ICON_HOVER = "8,112,65,1";
export const EXCELLENT_ICON_FOCUS = "8,112,65,1";
export const EXCELLENT_ICON_ACTIVE = "8,112,65,1";
export const EXCELLENT_ICON_DISABLED = "208,212,217,1";
export const EXCELLENT_ICON_IN_DEFAULT = "255,255,255,1";
export const EXCELLENT_ICON_IN_HOVER = "255,255,255,1";
export const EXCELLENT_ICON_IN_FOCUS = "255,255,255,1";
export const EXCELLENT_ICON_IN_ACTIVE = "255,255,255,1";
export const EXCELLENT_ICON_IN_DISABLED = "255,255,255,1";
export const EXCELLENT_BORDER_DEFAULT = "7,94,57,1";
export const EXCELLENT_BORDER_HOVER = "7,94,57,1";
export const EXCELLENT_BORDER_FOCUS = "7,94,57,1";
export const EXCELLENT_BORDER_ACTIVE = "7,94,57,1";
export const EXCELLENT_BORDER_DISABLED = "208,212,217,1";
