// @ciscodesignsystems/cds-react-header/injected/components/help-menu/help-menu.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var phosphorReact = require('phosphor-react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var injected = require('@ciscodesignsystems/cds-react-button/injected');
var headerMenu = require('../header-menu/header-menu.cjs.js');
var helpMenu_module = require('./help-menu.module.cjs.js');

const CDSHelpMenuRoot = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    children,
    sentiment,
    variant,
    contentRef,
    triggerRef
  } = props;
  return jsxRuntime.jsx(headerMenu.CDSHeaderMenuRoot, {
    ref: ref,
    id: "headerHelpMenu",
    "aria-label": "Help menu",
    trigger: jsxRuntime.jsx(injected.CDSButton, {
      ref: triggerRef,
      "aria-label": "Help content",
      sentiment: sentiment,
      icon: jsxRuntime.jsx(phosphorReact.Question, {
        weight: "bold",
        size: "24px"
      }),
      variant: variant
    }),
    children: jsxRuntime.jsx(headerMenu.CDSHeaderMenuContent, {
      className: cdsComponentUtilities.cx(helpMenu_module["default"], 'cds-help-menu'),
      align: "left",
      ref: contentRef,
      children: children
    })
  });
});
CDSHelpMenuRoot.displayName = 'CDSHelpMenuRoot';
const CDSHelpMenuSection = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    description,
    children,
    className,
    ...rest
  } = props;
  return jsxRuntime.jsxs(headerMenu.CDSHeaderMenuRow, {
    ref: ref,
    className: cdsComponentUtilities.cx(helpMenu_module["default"], 'cds-help-menu__section', className),
    ...rest,
    children: [description && jsxRuntime.jsx("span", {
      className: cdsComponentUtilities.cx(helpMenu_module["default"], 'cds-help-menu__section__description'),
      children: description
    }), jsxRuntime.jsx("ul", {
      className: cdsComponentUtilities.cx(helpMenu_module["default"], 'cds-help-menu__section__list'),
      children: children
    })]
  });
});
CDSHelpMenuSection.displayName = 'CDSHelpMenuSection';
CDSHelpMenuSection.defaultProps = {
  description: undefined
};
const CDSHelpMenuLink = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    children,
    interact,
    className,
    ...rest
  } = props;
  const isDisabled = interact === 'disabled';
  return jsxRuntime.jsxs(headerMenu.CDSHeaderMenuRow, {
    className: cdsComponentUtilities.cx(helpMenu_module["default"], 'cds-help-menu__section__row'),
    "data-cds-interact": interact,
    children: [jsxRuntime.jsx(headerMenu.CDSHeaderMenuLink, {
      className: cdsComponentUtilities.cx(helpMenu_module["default"], 'cds-help-menu__section__link', className),
      ref: ref,
      tabIndex: isDisabled ? -1 : undefined,
      "data-cds-interact": interact,
      ...rest,
      children: children
    }), interact === 'active' && jsxRuntime.jsx(phosphorReact.CheckCircle, {
      color: "#4580E5",
      weight: "fill",
      size: 23
    })]
  });
});
CDSHelpMenuLink.displayName = 'CDSHelpMenuLink';
CDSHelpMenuLink.defaultProps = {
  interact: undefined
};
CDSHelpMenuRoot.defaultProps = {
  sentiment: 'inverse',
  variant: 'ghost',
  align: undefined,
  contentRef: undefined,
  triggerRef: undefined
};

exports.CDSHelpMenuLink = CDSHelpMenuLink;
exports.CDSHelpMenuRoot = CDSHelpMenuRoot;
exports.CDSHelpMenuSection = CDSHelpMenuSection;
exports["default"] = CDSHelpMenuRoot;
