// @ciscodesignsystems/cds-react-logos/injected/logos.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var logo_module = require('./logo.module.cjs.js');

const LogoSizes = {
  xs: '12',
  sm: '24',
  md: '48',
  lg: '96',
  xl: '192'
};
const CDSLogo = ({
  size = 'sm',
  primaryTheme = {
    sentiment: 'inverse',
    usage: 'text'
  },
  secondaryTheme = {},
  className = '',
  children
}) => {
  const classes = cdsComponentUtilities.cx(logo_module["default"], 'cds-logo', {
    [`cds-primary-sentiment--${primaryTheme === null || primaryTheme === void 0 ? void 0 : primaryTheme.sentiment}`]: primaryTheme === null || primaryTheme === void 0 ? void 0 : primaryTheme.sentiment,
    [`cds-primary-usage--${primaryTheme === null || primaryTheme === void 0 ? void 0 : primaryTheme.usage}`]: primaryTheme === null || primaryTheme === void 0 ? void 0 : primaryTheme.usage,
    [`cds-primary-prominence--${primaryTheme === null || primaryTheme === void 0 ? void 0 : primaryTheme.prominence}`]: primaryTheme === null || primaryTheme === void 0 ? void 0 : primaryTheme.prominence,
    [`cds-secondary-sentiment--${secondaryTheme === null || secondaryTheme === void 0 ? void 0 : secondaryTheme.sentiment}`]: secondaryTheme === null || secondaryTheme === void 0 ? void 0 : secondaryTheme.sentiment,
    [`cds-secondary-usage--${secondaryTheme === null || secondaryTheme === void 0 ? void 0 : secondaryTheme.usage}`]: secondaryTheme === null || secondaryTheme === void 0 ? void 0 : secondaryTheme.usage,
    [`cds-secondary-prominence--${secondaryTheme === null || secondaryTheme === void 0 ? void 0 : secondaryTheme.prominence}`]: secondaryTheme === null || secondaryTheme === void 0 ? void 0 : secondaryTheme.prominence
  }, className);
  return /*#__PURE__*/react.cloneElement(children, {
    'aria-hidden': true,
    role: 'presentation',
    ...children.props,
    className: classes,
    height: LogoSizes[size]
  });
};

exports.CDSLogo = CDSLogo;
exports.LogoSizes = LogoSizes;
