// @ciscodesignsystems/cds-react-logos/injected/logos.esm.js
import { cloneElement } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './logo.module.esm.js';

const LogoSizes = {
  xs: '12',
  sm: '24',
  md: '48',
  lg: '96',
  xl: '192'
};
const CDSLogo = ({
  size = 'sm',
  primaryTheme = {
    sentiment: 'inverse',
    usage: 'text'
  },
  secondaryTheme = {},
  className = '',
  children
}) => {
  const classes = cx(styles, 'cds-logo', {
    [`cds-primary-sentiment--${primaryTheme?.sentiment}`]: primaryTheme?.sentiment,
    [`cds-primary-usage--${primaryTheme?.usage}`]: primaryTheme?.usage,
    [`cds-primary-prominence--${primaryTheme?.prominence}`]: primaryTheme?.prominence,
    [`cds-secondary-sentiment--${secondaryTheme?.sentiment}`]: secondaryTheme?.sentiment,
    [`cds-secondary-usage--${secondaryTheme?.usage}`]: secondaryTheme?.usage,
    [`cds-secondary-prominence--${secondaryTheme?.prominence}`]: secondaryTheme?.prominence
  }, className);
  return /*#__PURE__*/cloneElement(children, {
    'aria-hidden': true,
    role: 'presentation',
    ...children.props,
    className: classes,
    height: LogoSizes[size]
  });
};

export { CDSLogo, LogoSizes };
