// @ciscodesignsystems/cds-react-text-input/components/suffix/index.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var phosphorReact = require('phosphor-react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var passwordSuffix = require('./password-suffix.cjs.js');
var suffix_module = require('./suffix.module.cjs.js');

const SuffixComp = ({
  type,
  suffix,
  showPassword = undefined,
  setShowPassword = undefined,
  onClear,
  disabled,
  hasValue,
  showPasswordText,
  hidePasswordText
}) => {
  const togglePassword = () => {
    setShowPassword && setShowPassword(prevState => !prevState);
  };
  return jsxRuntime.jsxs("div", {
    className: cdsComponentUtilities.cx(suffix_module["default"], 'cds-text-input__suffix'),
    children: [type === 'password' && showPassword !== undefined && jsxRuntime.jsx(passwordSuffix["default"], {
      showPassword: showPassword,
      togglePassword: togglePassword,
      showPasswordText: showPasswordText,
      hidePasswordText: hidePasswordText
    }), suffix, onClear && hasValue && !disabled && jsxRuntime.jsx(phosphorReact.XCircle, {
      "aria-hidden": true,
      size: 16,
      onClick: onClear,
      className: cdsComponentUtilities.cx(suffix_module["default"], 'cds-text-input__clear-icon')
    })]
  });
};

exports["default"] = SuffixComp;
