// @ciscodesignsystems/cds-react-nav/injected/index.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var nav = require('./nav.cjs.js');
var index = require('./types/index.cjs.js');
var icon = require('./components/icon/icon.cjs.js');
var menu = require('./components/menu/menu.cjs.js');
var menuLabel = require('./components/menu-label/menu-label.cjs.js');
var menuLink = require('./components/menu-link/menu-link.cjs.js');
var menuSection = require('./components/menu-section/menu-section.cjs.js');
var navItem = require('./components/nav-item/nav-item.cjs.js');
var navItemWithSubmenu = require('./components/nav-item/nav-item-with-submenu.cjs.js');
var navItemBase = require('./components/nav-item/nav-item-base.cjs.js');
var section = require('./components/section/section.cjs.js');
var switcher = require('./components/switcher/switcher.cjs.js');



exports.CDSNav = nav.CDSNav;
exports.CDSNavMenuGroupGapArray = index.CDSNavMenuGroupGapArray;
exports.CDSNavMenuSizeArray = index.CDSNavMenuSizeArray;
Object.defineProperty(exports, 'CDSNavReducerActionKind', {
	enumerable: true,
	get: function () { return index.CDSNavReducerActionKind; }
});
exports.CDSNavIcon = icon.CDSNavIcon;
exports.CDSNavMenu = menu.CDSNavMenu;
exports.CDSMenuLabel = menuLabel.CDSMenuLabel;
exports.CDSMenuLink = menuLink.CDSMenuLink;
exports.CDSNavMenuSection = menuSection.CDSNavMenuSection;
exports.CDSNavItem = navItem.CDSNavItem;
exports.CDSNavItemWithSubMenu = navItemWithSubmenu.CDSNavItemWithSubMenu;
exports.CDSNavItemBase = navItemBase.CDSNavItemBase;
exports.CDSNavSection = section.CDSNavSection;
exports.CDSNavSwitcher = switcher.CDSNavSwitcher;
