// @ciscodesignsystems/cds-magnetic-theme-web/js/dist/token-theme-classic-dark-rgb-es6.js
/**
 * This file is auto-generated. Do not edit it directly.
 */

export const BRAND_BG_DEFAULT = "55,60,66,1";
export const BRAND_BG_HOVER = "55,60,66,1";
export const BRAND_BG_FOCUS = "70,76,84,1";
export const BRAND_BG_ACTIVE = "70,76,84,1";
export const BRAND_BG_DISABLED = "225,228,232,1";
export const BRAND_BG_GRADIENT_START = "55,60,66,1";
export const BRAND_BG_GRADIENT_END = "15,18,20,1";
export const BRAND_BG_STRONG_DEFAULT = "55,60,66,1";
export const BRAND_BG_STRONG_HOVER = "55,60,66,1";
export const BRAND_BG_STRONG_FOCUS = "55,60,66,1";
export const BRAND_BG_STRONG_ACTIVE = "55,60,66,1";
export const BRAND_TEXT_DEFAULT = "247,247,247,1";
export const BRAND_TEXT_HOVER = "247,247,247,1";
export const BRAND_TEXT_ACTIVE = "247,247,247,1";
export const BRAND_TEXT_WEAK_DEFAULT = "136,144,153,1";
export const BRAND_TEXT_WEAK_HOVER = "247,247,247,1";
export const BRAND_TEXT_WEAK_FOCUS = "247,247,247,1";
export const BRAND_TEXT_WEAK_ACTIVE = "247,247,247,1";
export const BRAND_TEXT_STRONG_DEFAULT = "240,241,242,1";
export const BRAND_ICON_DEFAULT = "247,247,247,1";
export const BRAND_ICON_HOVER = "255,255,255,1";
export const BRAND_ICON_ACTIVE = "255,255,255,1";
export const BRAND_ICON_WEAK_DEFAULT = "247,247,247,1";
export const BRAND_ICON_WEAK_HOVER = "247,247,247,1";
export const BRAND_ICON_WEAK_FOCUS = "247,247,247,1";
export const BRAND_ICON_WEAK_ACTIVE = "247,247,247,1";
export const BRAND_BORDER_ACTIVE = "204,225,255,1";
export const BRAND_ACCENT_ICON_DEFAULT = "193,198,204,1";
export const BRAND_ACCENT_ICON_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_ICON_ACTIVE = "107,191,65,1";
export const BRAND_ACCENT_ICON_DISABLED = "111,118,128,1";
export const BRAND_ACCENT_ICON_WEAK_DEFAULT = "111,118,128,1";
export const BRAND_ACCENT_ICON_WEAK_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_ICON_WEAK_ACTIVE = "57,133,25,1";
export const BRAND_ACCENT_ICON_WEAK_DISABLED = "255,255,255,1";
export const BRAND_ACCENT_ICON_MEDIUM_DEFAULT = "151,159,168,1";
export const BRAND_ACCENT_ICON_MEDIUM_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_ICON_MEDIUM_ACTIVE = "82,166,43,1";
export const BRAND_ACCENT_ICON_MEDIUM_DISABLED = "255,255,255,1";
export const BRAND_ACCENT_BORDER_DEFAULT = "82,166,43,1";
export const BRAND_ACCENT_BORDER_HOVER = "126,134,143,1";
export const BRAND_ACCENT_BORDER_FOCUS = "82,166,43,1";
export const BRAND_ACCENT_BORDER_ACTIVE = "82,166,43,1";
export const BRAND_ACCENT_BORDER_WEAK_DEFAULT = "208,212,217,1";
export const BRAND_ACCENT_BORDER_WEAK_HOVER = "255,255,255,1";
export const BRAND_ACCENT_BORDER_WEAK_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_BORDER_WEAK_ACTIVE = "57,133,25,1";
export const BRAND_ACCENT_BORDER_WEAK_DISABLED = "255,255,255,1";
export const BRAND_ACCENT_BORDER_MEDIUM_DEFAULT = "89,96,105,1";
export const BRAND_ACCENT_BORDER_MEDIUM_HOVER = "255,255,255,1";
export const BRAND_ACCENT_BORDER_MEDIUM_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_BORDER_MEDIUM_ACTIVE = "43,122,12,1";
export const BRAND_ACCENT_BORDER_MEDIUM_DISABLED = "255,255,255,1";
export const BRAND_ACCENT_BORDER_STRONG_DEFAULT = "15,18,20,1";
export const BRAND_ACCENT_BORDER_STRONG_HOVER = "255,255,255,1";
export const BRAND_ACCENT_BORDER_STRONG_FOCUS = "255,255,255,1";
export const BRAND_ACCENT_BORDER_STRONG_ACTIVE = "15,18,20,1";
export const BRAND_ACCENT_BORDER_STRONG_DISABLED = "255,255,255,1";
export const BASE_BG_DEFAULT = "15,18,20,1";
export const BASE_BG_HOVER = "15,18,20,1";
export const BASE_BG_ACTIVE = "15,18,20,1";
export const BASE_BG_DISABLED = "70,76,84,1";
export const BASE_BG_WEAK_DEFAULT = "35,40,46,1";
export const BASE_BG_WEAK_HOVER = "55,60,66,1";
export const BASE_BG_WEAK_ACTIVE = "35,40,46,1";
export const BASE_BG_WEAK_DISABLED = "70,76,84,1";
export const BASE_BG_MEDIUM_DEFAULT = "55,60,66,1";
export const BASE_BG_MEDIUM_HOVER = "55,60,66,1";
export const BASE_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const BASE_BG_MEDIUM_DISABLED = "232,95,198,1";
export const BASE_BG_STRONG_DEFAULT = "167,173,181,1";
export const BASE_BG_STRONG_HOVER = "208,212,217,1";
export const BASE_BG_STRONG_FOCUS = "208,212,217,1";
export const BASE_BG_STRONG_ACTIVE = "208,212,217,1";
export const BASE_BG_STRONG_DISABLED = "89,96,105,1";
export const BASE_ICON_IN_DEFAULT = "35,40,46,1";
export const BASE_ICON_IN_HOVER = "35,40,46,1";
export const BASE_ICON_IN_FOCUS = "35,40,46,1";
export const BASE_ICON_IN_ACTIVE = "35,40,46,1";
export const BASE_ICON_IN_DISABLED = "111,118,128,1";
export const BASE_TEXT_DEFAULT = "247,247,247,1";
export const BASE_TEXT_DISABLED = "111,118,128,1";
export const BASE_TEXT_WEAK_DEFAULT = "136,144,153,1";
export const BASE_TEXT_WEAK_DISABLED = "111,118,128,1";
export const BASE_TEXT_MEDIUM_DEFAULT = "208,212,217,1";
export const BASE_TEXT_MEDIUM_DISABLED = "111,118,128,1";
export const BASE_TEXT_STRONG_DEFAULT = "240,241,242,1";
export const BASE_TEXT_STRONG_DISABLED = "111,118,128,1";
export const BASE_TEXT_IN_DEFAULT = "35,40,46,1";
export const BASE_TEXT_IN_HOVER = "35,40,46,1";
export const BASE_TEXT_IN_FOCUS = "35,40,46,1";
export const BASE_TEXT_IN_ACTIVE = "35,40,46,1";
export const BASE_TEXT_IN_DISABLED = "35,40,46,1";
export const BASE_TEXT_IN_WEAK_DEFAULT = "255,255,255,1";
export const BASE_BORDER_DEFAULT = "89,96,105,1";
export const BASE_BORDER_WEAK_DEFAULT = "35,40,46,1";
export const BASE_BORDER_MEDIUM_DEFAULT = "70,76,84,1";
export const BASE_BORDER_STRONG_DEFAULT = "193,198,204,1";
export const INVERSE_BG_DEFAULT = "55,60,66,1";
export const INVERSE_BG_HOVER = "55,60,66,1";
export const INVERSE_BG_FOCUS = "70,76,84,1";
export const INVERSE_BG_ACTIVE = "70,76,84,1";
export const INVERSE_BG_DISABLED = "225,228,232,1";
export const INVERSE_BG_GRADIENT_START = "55,60,66,1";
export const INVERSE_BG_GRADIENT_END = "15,18,20,1";
export const INVERSE_BG_WEAK_DEFAULT = "70,76,84,1";
export const INVERSE_TEXT_DEFAULT = "247,247,247,1";
export const INVERSE_TEXT_HOVER = "247,247,247,1";
export const INVERSE_TEXT_FOCUS = "247,247,247,1";
export const INVERSE_TEXT_ACTIVE = "247,247,247,1";
export const INVERSE_TEXT_DISABLED = "101,108,117,1";
export const INVERSE_TEXT_MEDIUM_DEFAULT = "193,198,204,1";
export const INVERSE_TEXT_MEDIUM_DISABLED = "101,108,117,1";
export const INVERSE_ICON_DEFAULT = "247,247,247,1";
export const INVERSE_ICON_HOVER = "255,255,255,1";
export const INVERSE_ICON_FOCUS = "255,255,255,1";
export const INVERSE_ICON_ACTIVE = "255,255,255,1";
export const INVERSE_ICON_DISABLED = "111,118,128,1";
export const INVERSE_ICON_STRONG_DEFAULT = "255,255,255,1";
export const INVERSE_ICON_STRONG_HOVER = "255,255,255,1";
export const INVERSE_ICON_STRONG_FOCUS = "255,255,255,1";
export const INVERSE_ICON_STRONG_ACTIVE = "255,255,255,1";
export const INVERSE_ICON_STRONG_DISABLED = "111,118,128,1";
export const INVERSE_BORDER_DEFAULT = "111,118,128,1";
export const INVERSE_BORDER_HOVER = "100,158,245,1";
export const INVERSE_BORDER_FOCUS = "204,225,255,1";
export const INVERSE_BORDER_ACTIVE = "204,225,255,1";
export const INVERSE_BORDER_DISABLED = "225,228,232,1";
export const CONTROL_BG_DEFAULT = "35,40,46,1";
export const CONTROL_BG_HOVER = "35,40,46,1";
export const CONTROL_BG_FOCUS = "35,40,46,1";
export const CONTROL_BG_ACTIVE = "35,40,46,1";
export const CONTROL_BG_DISABLED = "55,60,66,1";
export const CONTROL_BG_WEAK_DEFAULT = "35,40,46,1";
export const CONTROL_BG_WEAK_HOVER = "55,60,66,1";
export const CONTROL_BG_WEAK_FOCUS = "35,40,46,1";
export const CONTROL_BG_WEAK_ACTIVE = "35,40,46,1";
export const CONTROL_BG_WEAK_DISABLED = "55,60,66,1";
export const CONTROL_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const CONTROL_BG_MEDIUM_HOVER = "55,60,66,1";
export const CONTROL_BG_MEDIUM_FOCUS = "151,159,168,1";
export const CONTROL_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const CONTROL_BG_MEDIUM_DISABLED = "55,60,66,1";
export const CONTROL_BG_STRONG_DEFAULT = "100,158,245,1";
export const CONTROL_BG_STRONG_HOVER = "124,173,247,1";
export const CONTROL_BG_STRONG_FOCUS = "124,173,247,1";
export const CONTROL_BG_STRONG_ACTIVE = "124,173,247,1";
export const CONTROL_BG_STRONG_DISABLED = "89,96,105,1";
export const CONTROL_ICON_DEFAULT = "100,158,245,1";
export const CONTROL_ICON_HOVER = "124,173,247,1";
export const CONTROL_ICON_FOCUS = "124,173,247,1";
export const CONTROL_ICON_ACTIVE = "124,173,247,1";
export const CONTROL_ICON_DISABLED = "204,225,255,1";
export const CONTROL_ICON_WEAK_DEFAULT = "193,198,204,1";
export const CONTROL_ICON_WEAK_HOVER = "100,158,245,1";
export const CONTROL_ICON_WEAK_FOCUS = "100,158,245,1";
export const CONTROL_ICON_WEAK_ACTIVE = "100,158,245,1";
export const CONTROL_ICON_WEAK_DISABLED = "111,118,128,1";
export const CONTROL_ICON_MEDIUM_DEFAULT = "100,158,245,1";
export const CONTROL_ICON_MEDIUM_HOVER = "151,159,168,1";
export const CONTROL_ICON_MEDIUM_FOCUS = "151,159,168,1";
export const CONTROL_ICON_MEDIUM_ACTIVE = "136,144,153,1";
export const CONTROL_ICON_MEDIUM_DISABLED = "225,228,232,1";
export const CONTROL_ICON_STRONG_DEFAULT = "208,212,217,1";
export const CONTROL_ICON_STRONG_HOVER = "240,241,242,1";
export const CONTROL_ICON_STRONG_FOCUS = "100,158,245,1";
export const CONTROL_ICON_STRONG_ACTIVE = "100,158,245,1";
export const CONTROL_ICON_STRONG_DISABLED = "111,118,128,1";
export const CONTROL_ICON_IN_DEFAULT = "35,40,46,1";
export const CONTROL_ICON_IN_HOVER = "35,40,46,1";
export const CONTROL_ICON_IN_FOCUS = "35,40,46,1";
export const CONTROL_ICON_IN_ACTIVE = "35,40,46,1";
export const CONTROL_ICON_IN_DISABLED = "111,118,128,1";
export const CONTROL_ICON_IN_STRONG_DEFAULT = "100,158,245,1";
export const CONTROL_ICON_IN_STRONG_HOVER = "124,173,247,1";
export const CONTROL_ICON_IN_STRONG_FOCUS = "124,173,247,1";
export const CONTROL_ICON_IN_STRONG_ACTIVE = "124,173,247,1";
export const CONTROL_ICON_IN_STRONG_DISABLED = "111,118,128,1";
export const CONTROL_BORDER_DEFAULT = "126,134,143,1";
export const CONTROL_BORDER_HOVER = "124,173,247,1";
export const CONTROL_BORDER_FOCUS = "124,173,247,1";
export const CONTROL_BORDER_ACTIVE = "124,173,247,1";
export const CONTROL_BORDER_DISABLED = "89,96,105,1";
export const CONTROL_BORDER_WEAK_DEFAULT = "101,108,117,1";
export const CONTROL_BORDER_WEAK_HOVER = "124,173,247,1";
export const CONTROL_BORDER_WEAK_FOCUS = "124,173,247,1";
export const CONTROL_BORDER_WEAK_ACTIVE = "124,173,247,1";
export const CONTROL_BORDER_WEAK_DISABLED = "89,96,105,1";
export const CONTROL_BORDER_MEDIUM_DEFAULT = "100,158,245,1";
export const CONTROL_BORDER_MEDIUM_HOVER = "100,158,245,1";
export const CONTROL_BORDER_MEDIUM_FOCUS = "100,158,245,1";
export const CONTROL_BORDER_MEDIUM_ACTIVE = "100,158,245,1";
export const CONTROL_BORDER_MEDIUM_DISABLED = "89,96,105,1";
export const CONTROL_BORDER_STRONG_DEFAULT = "193,198,204,1";
export const CONTROL_BORDER_STRONG_HOVER = "101,108,117,1";
export const CONTROL_BORDER_STRONG_FOCUS = "101,108,117,1";
export const CONTROL_BORDER_STRONG_ACTIVE = "100,158,245,1";
export const CONTROL_BORDER_STRONG_DISABLED = "89,96,105,1";
export const INTERACT_BG_DEFAULT = "100,158,245,1";
export const INTERACT_BG_HOVER = "124,173,247,1";
export const INTERACT_BG_FOCUS = "124,173,247,1";
export const INTERACT_BG_ACTIVE = "124,173,247,1";
export const INTERACT_BG_DISABLED = "35,40,46,1";
export const INTERACT_BG_WEAK_DEFAULT = "35,40,46,1";
export const INTERACT_BG_WEAK_HOVER = "35,40,46,1";
export const INTERACT_BG_WEAK_FOCUS = "35,40,46,1";
export const INTERACT_BG_WEAK_ACTIVE = "35,40,46,1";
export const INTERACT_BG_WEAK_DISABLED = "70,76,84,1";
export const INTERACT_BG_MEDIUM_DEFAULT = "100,158,245,1";
export const INTERACT_BG_MEDIUM_HOVER = "89,96,105,1";
export const INTERACT_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const INTERACT_BG_STRONG_HOVER = "89,96,105,1";
export const INTERACT_BG_STRONG_ACTIVE = "89,96,105,1";
export const INTERACT_TEXT_DEFAULT = "100,158,245,1";
export const INTERACT_TEXT_HOVER = "124,173,247,1";
export const INTERACT_TEXT_FOCUS = "124,173,247,1";
export const INTERACT_TEXT_ACTIVE = "124,173,247,1";
export const INTERACT_TEXT_DISABLED = "35,40,46,1";
export const INTERACT_TEXT_WEAK_DEFAULT = "193,198,204,1";
export const INTERACT_TEXT_WEAK_HOVER = "100,158,245,1";
export const INTERACT_TEXT_WEAK_FOCUS = "100,158,245,1";
export const INTERACT_TEXT_WEAK_ACTIVE = "100,158,245,1";
export const INTERACT_TEXT_WEAK_DISABLED = "111,118,128,1";
export const INTERACT_TEXT_IN_DEFAULT = "35,40,46,1";
export const INTERACT_TEXT_IN_HOVER = "35,40,46,1";
export const INTERACT_TEXT_IN_FOCUS = "35,40,46,1";
export const INTERACT_TEXT_IN_ACTIVE = "35,40,46,1";
export const INTERACT_TEXT_IN_DISABLED = "35,40,46,1";
export const INTERACT_ICON_DEFAULT = "100,158,245,1";
export const INTERACT_ICON_HOVER = "124,173,247,1";
export const INTERACT_ICON_FOCUS = "124,173,247,1";
export const INTERACT_ICON_ACTIVE = "124,173,247,1";
export const INTERACT_ICON_DISABLED = "35,40,46,1";
export const INTERACT_ICON_WEAK_DEFAULT = "193,198,204,1";
export const INTERACT_ICON_WEAK_HOVER = "208,212,217,1";
export const INTERACT_ICON_WEAK_FOCUS = "208,212,217,1";
export const INTERACT_ICON_WEAK_ACTIVE = "208,212,217,1";
export const INTERACT_ICON_WEAK_DISABLED = "111,118,128,1";
export const INTERACT_ICON_MEDIUM_DEFAULT = "193,198,204,1";
export const INTERACT_ICON_MEDIUM_HOVER = "100,158,245,1";
export const INTERACT_ICON_MEDIUM_FOCUS = "100,158,245,1";
export const INTERACT_ICON_MEDIUM_ACTIVE = "100,158,245,1";
export const INTERACT_ICON_MEDIUM_DISABLED = "111,118,128,1";
export const INTERACT_ICON_STRONG_DEFAULT = "208,212,217,1";
export const INTERACT_ICON_STRONG_HOVER = "240,241,242,1";
export const INTERACT_ICON_STRONG_FOCUS = "240,241,242,1";
export const INTERACT_ICON_STRONG_ACTIVE = "240,241,242,1";
export const INTERACT_ICON_STRONG_DISABLED = "111,118,128,1";
export const INTERACT_ICON_IN_DEFAULT = "35,40,46,1";
export const INTERACT_ICON_IN_HOVER = "35,40,46,1";
export const INTERACT_ICON_IN_FOCUS = "35,40,46,1";
export const INTERACT_ICON_IN_ACTIVE = "35,40,46,1";
export const INTERACT_ICON_IN_DISABLED = "35,40,46,1";
export const INTERACT_BORDER_DEFAULT = "100,158,245,1";
export const INTERACT_BORDER_HOVER = "124,173,247,1";
export const INTERACT_BORDER_FOCUS = "124,173,247,1";
export const INTERACT_BORDER_ACTIVE = "124,173,247,1";
export const INTERACT_BORDER_DISABLED = "35,40,46,1";
export const INTERACT_BORDER_WEAK_DEFAULT = "101,108,117,1";
export const INTERACT_BORDER_WEAK_HOVER = "204,225,255,1";
export const INTERACT_BORDER_WEAK_FOCUS = "225,228,232,1";
export const INTERACT_BORDER_WEAK_ACTIVE = "225,228,232,1";
export const INTERACT_BORDER_WEAK_DISABLED = "89,96,105,1";
export const INTERACT_BORDER_MEDIUM_DEFAULT = "100,158,245,1";
export const INTERACT_BORDER_MEDIUM_HOVER = "163,200,255,1";
export const INTERACT_BORDER_MEDIUM_FOCUS = "100,158,245,1";
export const INTERACT_BORDER_MEDIUM_ACTIVE = "100,158,245,1";
export const INTERACT_BORDER_MEDIUM_DISABLED = "225,228,232,1";
export const POSITIVE_BG_DEFAULT = "107,191,65,1";
export const POSITIVE_BG_HOVER = "107,191,65,1";
export const POSITIVE_BG_FOCUS = "107,191,65,1";
export const POSITIVE_BG_ACTIVE = "107,191,65,1";
export const POSITIVE_BG_DISABLED = "107,191,65,1";
export const POSITIVE_BG_WEAK_DEFAULT = "35,40,46,1";
export const POSITIVE_BG_WEAK_HOVER = "35,40,46,1";
export const POSITIVE_BG_WEAK_FOCUS = "35,40,46,1";
export const POSITIVE_BG_WEAK_ACTIVE = "35,40,46,1";
export const POSITIVE_BG_WEAK_DISABLED = "35,40,46,1";
export const POSITIVE_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const POSITIVE_BG_MEDIUM_HOVER = "35,40,46,1";
export const POSITIVE_BG_MEDIUM_FOCUS = "35,40,46,1";
export const POSITIVE_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const POSITIVE_BG_MEDIUM_DISABLED = "55,60,66,1";
export const POSITIVE_BG_STRONG_DEFAULT = "15,18,20,1";
export const POSITIVE_ICON_DEFAULT = "107,191,65,1";
export const POSITIVE_ICON_HOVER = "107,191,65,1";
export const POSITIVE_ICON_FOCUS = "107,191,65,1";
export const POSITIVE_ICON_ACTIVE = "107,191,65,1";
export const POSITIVE_ICON_DISABLED = "89,96,105,1";
export const POSITIVE_ICON_WEAK_DEFAULT = "107,191,65,1";
export const POSITIVE_ICON_WEAK_HOVER = "107,191,65,1";
export const POSITIVE_ICON_WEAK_FOCUS = "107,191,65,1";
export const POSITIVE_ICON_WEAK_ACTIVE = "107,191,65,1";
export const POSITIVE_ICON_WEAK_DISABLED = "89,96,105,1";
export const POSITIVE_ICON_STRONG_DEFAULT = "107,191,65,1";
export const POSITIVE_ICON_STRONG_HOVER = "107,191,65,1";
export const POSITIVE_ICON_STRONG_FOCUS = "107,191,65,1";
export const POSITIVE_ICON_STRONG_ACTIVE = "107,191,65,1";
export const POSITIVE_ICON_STRONG_DISABLED = "35,40,46,1";
export const POSITIVE_ICON_IN_DEFAULT = "35,40,46,1";
export const POSITIVE_ICON_IN_HOVER = "35,40,46,1";
export const POSITIVE_ICON_IN_FOCUS = "35,40,46,1";
export const POSITIVE_ICON_IN_ACTIVE = "35,40,46,1";
export const POSITIVE_ICON_IN_DISABLED = "35,40,46,1";
export const POSITIVE_BORDER_DEFAULT = "107,191,65,1";
export const POSITIVE_BORDER_HOVER = "107,191,65,1";
export const POSITIVE_BORDER_FOCUS = "107,191,65,1";
export const POSITIVE_BORDER_ACTIVE = "107,191,65,1";
export const POSITIVE_BORDER_DISABLED = "107,191,65,1";
export const POSITIVE_BORDER_WEAK_DEFAULT = "35,40,46,1";
export const POSITIVE_BORDER_WEAK_HOVER = "107,191,65,1";
export const POSITIVE_BORDER_WEAK_FOCUS = "107,191,65,1";
export const POSITIVE_BORDER_WEAK_ACTIVE = "107,191,65,1";
export const POSITIVE_BORDER_WEAK_DISABLED = "89,96,105,1";
export const POSITIVE_BORDER_STRONG_DEFAULT = "247,247,247,1";
export const POSITIVE_BORDER_STRONG_HOVER = "247,247,247,1";
export const POSITIVE_BORDER_STRONG_FOCUS = "247,247,247,1";
export const POSITIVE_BORDER_STRONG_ACTIVE = "247,247,247,1";
export const POSITIVE_BORDER_STRONG_DISABLED = "247,247,247,1";
export const POSITIVE_TEXT_IN_DEFAULT = "35,40,46,1";
export const POSITIVE_TEXT_IN_HOVER = "35,40,46,1";
export const POSITIVE_TEXT_IN_FOCUS = "35,40,46,1";
export const POSITIVE_TEXT_IN_ACTIVE = "35,40,46,1";
export const POSITIVE_TEXT_IN_DISABLED = "35,40,46,1";
export const WARNING_BG_DEFAULT = "240,194,67,1";
export const WARNING_BG_HOVER = "240,194,67,1";
export const WARNING_BG_FOCUS = "240,194,67,1";
export const WARNING_BG_ACTIVE = "240,194,67,1";
export const WARNING_BG_DISABLED = "240,194,67,1";
export const WARNING_BG_WEAK_DEFAULT = "35,40,46,1";
export const WARNING_BG_WEAK_HOVER = "35,40,46,1";
export const WARNING_BG_WEAK_FOCUS = "35,40,46,1";
export const WARNING_BG_WEAK_ACTIVE = "35,40,46,1";
export const WARNING_BG_WEAK_DISABLED = "35,40,46,1";
export const WARNING_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const WARNING_BG_MEDIUM_HOVER = "35,40,46,1";
export const WARNING_BG_MEDIUM_FOCUS = "35,40,46,1";
export const WARNING_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const WARNING_BG_MEDIUM_DISABLED = "55,60,66,1";
export const WARNING_BG_STRONG_DEFAULT = "15,18,20,1";
export const WARNING_ICON_DEFAULT = "240,194,67,1";
export const WARNING_ICON_HOVER = "240,194,67,1";
export const WARNING_ICON_FOCUS = "240,194,67,1";
export const WARNING_ICON_ACTIVE = "240,194,67,1";
export const WARNING_ICON_DISABLED = "89,96,105,1";
export const WARNING_ICON_STRONG_DEFAULT = "240,194,67,1";
export const WARNING_ICON_STRONG_HOVER = "240,194,67,1";
export const WARNING_ICON_STRONG_FOCUS = "240,194,67,1";
export const WARNING_ICON_STRONG_ACTIVE = "240,194,67,1";
export const WARNING_ICON_STRONG_DISABLED = "240,194,67,1";
export const WARNING_ICON_IN_DEFAULT = "35,40,46,1";
export const WARNING_ICON_IN_HOVER = "35,40,46,1";
export const WARNING_ICON_IN_FOCUS = "35,40,46,1";
export const WARNING_ICON_IN_ACTIVE = "35,40,46,1";
export const WARNING_ICON_IN_DISABLED = "35,40,46,1";
export const WARNING_BORDER_DEFAULT = "240,194,67,1";
export const WARNING_BORDER_HOVER = "240,194,67,1";
export const WARNING_BORDER_FOCUS = "240,194,67,1";
export const WARNING_BORDER_ACTIVE = "240,194,67,1";
export const WARNING_BORDER_DISABLED = "240,194,67,1";
export const WARNING_BORDER_WEAK_DEFAULT = "35,40,46,1";
export const WARNING_BORDER_WEAK_HOVER = "35,40,46,1";
export const WARNING_BORDER_WEAK_FOCUS = "35,40,46,1";
export const WARNING_BORDER_WEAK_ACTIVE = "35,40,46,1";
export const WARNING_BORDER_WEAK_DISABLED = "89,96,105,1";
export const WARNING_BORDER_STRONG_DEFAULT = "247,247,247,1";
export const WARNING_BORDER_STRONG_HOVER = "247,247,247,1";
export const WARNING_BORDER_STRONG_FOCUS = "247,247,247,1";
export const WARNING_BORDER_STRONG_ACTIVE = "247,247,247,1";
export const WARNING_BORDER_STRONG_DISABLED = "247,247,247,1";
export const WARNING_TEXT_IN_DEFAULT = "35,40,46,1";
export const WARNING_TEXT_IN_HOVER = "35,40,46,1";
export const WARNING_TEXT_IN_FOCUS = "35,40,46,1";
export const WARNING_TEXT_IN_ACTIVE = "35,40,46,1";
export const WARNING_TEXT_IN_DISABLED = "35,40,46,1";
export const SEVERE_WARNING_BG_DEFAULT = "247,120,47,1";
export const SEVERE_WARNING_BG_HOVER = "247,120,47,1";
export const SEVERE_WARNING_BG_FOCUS = "247,120,47,1";
export const SEVERE_WARNING_BG_ACTIVE = "247,120,47,1";
export const SEVERE_WARNING_BG_DISABLED = "247,120,47,1";
export const SEVERE_WARNING_BG_WEAK_DEFAULT = "35,40,46,1";
export const SEVERE_WARNING_BG_WEAK_HOVER = "35,40,46,1";
export const SEVERE_WARNING_BG_WEAK_FOCUS = "35,40,46,1";
export const SEVERE_WARNING_BG_WEAK_ACTIVE = "35,40,46,1";
export const SEVERE_WARNING_BG_WEAK_DISABLED = "35,40,46,1";
export const SEVERE_WARNING_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const SEVERE_WARNING_BG_MEDIUM_HOVER = "35,40,46,1";
export const SEVERE_WARNING_BG_MEDIUM_FOCUS = "35,40,46,1";
export const SEVERE_WARNING_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const SEVERE_WARNING_BG_MEDIUM_DISABLED = "55,60,66,1";
export const SEVERE_WARNING_BG_STRONG_DEFAULT = "15,18,20,1";
export const SEVERE_WARNING_ICON_DEFAULT = "247,120,47,1";
export const SEVERE_WARNING_ICON_HOVER = "247,120,47,1";
export const SEVERE_WARNING_ICON_FOCUS = "247,120,47,1";
export const SEVERE_WARNING_ICON_ACTIVE = "247,120,47,1";
export const SEVERE_WARNING_ICON_DISABLED = "89,96,105,1";
export const SEVERE_WARNING_ICON_STRONG_DEFAULT = "252,141,76,1";
export const SEVERE_WARNING_ICON_STRONG_HOVER = "252,141,76,1";
export const SEVERE_WARNING_ICON_STRONG_FOCUS = "252,141,76,1";
export const SEVERE_WARNING_ICON_STRONG_ACTIVE = "252,141,76,1";
export const SEVERE_WARNING_ICON_STRONG_DISABLED = "252,141,76,1";
export const SEVERE_WARNING_ICON_IN_DEFAULT = "35,40,46,1";
export const SEVERE_WARNING_ICON_IN_HOVER = "35,40,46,1";
export const SEVERE_WARNING_ICON_IN_FOCUS = "35,40,46,1";
export const SEVERE_WARNING_ICON_IN_ACTIVE = "35,40,46,1";
export const SEVERE_WARNING_ICON_IN_DISABLED = "35,40,46,1";
export const SEVERE_WARNING_BORDER_DEFAULT = "247,120,47,1";
export const SEVERE_WARNING_BORDER_HOVER = "247,120,47,1";
export const SEVERE_WARNING_BORDER_FOCUS = "247,120,47,1";
export const SEVERE_WARNING_BORDER_ACTIVE = "247,120,47,1";
export const SEVERE_WARNING_BORDER_DISABLED = "247,120,47,1";
export const SEVERE_WARNING_BORDER_WEAK_DEFAULT = "35,40,46,1";
export const SEVERE_WARNING_BORDER_WEAK_HOVER = "35,40,46,1";
export const SEVERE_WARNING_BORDER_WEAK_FOCUS = "35,40,46,1";
export const SEVERE_WARNING_BORDER_WEAK_ACTIVE = "35,40,46,1";
export const SEVERE_WARNING_BORDER_WEAK_DISABLED = "89,96,105,1";
export const NEGATIVE_BG_DEFAULT = "250,87,98,1";
export const NEGATIVE_BG_HOVER = "255,110,114,1";
export const NEGATIVE_BG_FOCUS = "255,110,114,1";
export const NEGATIVE_BG_ACTIVE = "255,110,114,1";
export const NEGATIVE_BG_DISABLED = "35,40,46,1";
export const NEGATIVE_BG_WEAK_DEFAULT = "35,40,46,1";
export const NEGATIVE_BG_WEAK_HOVER = "55,60,66,1";
export const NEGATIVE_BG_WEAK_FOCUS = "55,60,66,1";
export const NEGATIVE_BG_WEAK_ACTIVE = "55,60,66,1";
export const NEGATIVE_BG_WEAK_DISABLED = "55,60,66,1";
export const NEGATIVE_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const NEGATIVE_BG_MEDIUM_HOVER = "35,40,46,1";
export const NEGATIVE_BG_MEDIUM_FOCUS = "35,40,46,1";
export const NEGATIVE_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const NEGATIVE_BG_MEDIUM_DISABLED = "55,60,66,1";
export const NEGATIVE_BG_STRONG_DEFAULT = "15,18,20,1";
export const NEGATIVE_TEXT_DEFAULT = "250,87,98,1";
export const NEGATIVE_TEXT_HOVER = "255,135,139,1";
export const NEGATIVE_TEXT_FOCUS = "255,135,139,1";
export const NEGATIVE_TEXT_ACTIVE = "255,135,139,1";
export const NEGATIVE_TEXT_DISABLED = "35,40,46,1";
export const NEGATIVE_TEXT_IN_DEFAULT = "35,40,46,1";
export const NEGATIVE_TEXT_IN_HOVER = "35,40,46,1";
export const NEGATIVE_TEXT_IN_FOCUS = "35,40,46,1";
export const NEGATIVE_TEXT_IN_ACTIVE = "35,40,46,1";
export const NEGATIVE_TEXT_IN_DISABLED = "35,40,46,1";
export const NEGATIVE_ICON_DEFAULT = "250,87,98,1";
export const NEGATIVE_ICON_HOVER = "255,135,139,1";
export const NEGATIVE_ICON_FOCUS = "255,135,139,1";
export const NEGATIVE_ICON_ACTIVE = "255,135,139,1";
export const NEGATIVE_ICON_DISABLED = "35,40,46,1";
export const NEGATIVE_ICON_WEAK_DEFAULT = "255,178,181,1";
export const NEGATIVE_ICON_WEAK_HOVER = "255,178,181,1";
export const NEGATIVE_ICON_WEAK_FOCUS = "255,178,181,1";
export const NEGATIVE_ICON_WEAK_ACTIVE = "255,178,181,1";
export const NEGATIVE_ICON_WEAK_DISABLED = "89,96,105,1";
export const NEGATIVE_ICON_IN_DEFAULT = "35,40,46,1";
export const NEGATIVE_ICON_IN_HOVER = "35,40,46,1";
export const NEGATIVE_ICON_IN_FOCUS = "35,40,46,1";
export const NEGATIVE_ICON_IN_ACTIVE = "35,40,46,1";
export const NEGATIVE_ICON_IN_DISABLED = "35,40,46,1";
export const NEGATIVE_BORDER_DEFAULT = "250,87,98,1";
export const NEGATIVE_BORDER_HOVER = "255,135,139,1";
export const NEGATIVE_BORDER_FOCUS = "255,135,139,1";
export const NEGATIVE_BORDER_ACTIVE = "255,135,139,1";
export const NEGATIVE_BORDER_DISABLED = "35,40,46,1";
export const NEGATIVE_BORDER_WEAK_DEFAULT = "35,40,46,1";
export const NEGATIVE_BORDER_WEAK_HOVER = "35,40,46,1";
export const NEGATIVE_BORDER_WEAK_FOCUS = "35,40,46,1";
export const NEGATIVE_BORDER_WEAK_ACTIVE = "35,40,46,1";
export const NEGATIVE_BORDER_WEAK_DISABLED = "89,96,105,1";
export const NEGATIVE_BORDER_MEDIUM_DEFAULT = "235,70,81,1";
export const NEGATIVE_BORDER_MEDIUM_HOVER = "255,135,139,1";
export const NEGATIVE_BORDER_MEDIUM_FOCUS = "255,135,139,1";
export const NEGATIVE_BORDER_MEDIUM_ACTIVE = "255,135,139,1";
export const NEGATIVE_BORDER_MEDIUM_DISABLED = "35,40,46,1";
export const NEGATIVE_BORDER_STRONG_DEFAULT = "247,247,247,1";
export const NEGATIVE_BORDER_STRONG_HOVER = "255,255,255,1";
export const NEGATIVE_BORDER_STRONG_FOCUS = "247,247,247,1";
export const NEGATIVE_BORDER_STRONG_ACTIVE = "247,247,247,1";
export const NEGATIVE_BORDER_STRONG_DISABLED = "247,247,247,1";
export const INFO_BG_DEFAULT = "124,173,247,1";
export const INFO_BG_HOVER = "124,173,247,1";
export const INFO_BG_FOCUS = "124,173,247,1";
export const INFO_BG_ACTIVE = "124,173,247,1";
export const INFO_BG_DISABLED = "124,173,247,1";
export const INFO_BG_WEAK_DEFAULT = "35,40,46,1";
export const INFO_BG_WEAK_HOVER = "35,40,46,1";
export const INFO_BG_WEAK_FOCUS = "35,40,46,1";
export const INFO_BG_WEAK_ACTIVE = "35,40,46,1";
export const INFO_BG_WEAK_DISABLED = "35,40,46,1";
export const INFO_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const INFO_BG_MEDIUM_HOVER = "35,40,46,1";
export const INFO_BG_MEDIUM_FOCUS = "35,40,46,1";
export const INFO_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const INFO_BG_MEDIUM_DISABLED = "55,60,66,1";
export const INFO_BG_STRONG_DEFAULT = "15,18,20,1";
export const INFO_ICON_DEFAULT = "124,173,247,1";
export const INFO_ICON_HOVER = "124,173,247,1";
export const INFO_ICON_FOCUS = "124,173,247,1";
export const INFO_ICON_ACTIVE = "124,173,247,1";
export const INFO_ICON_DISABLED = "89,96,105,1";
export const INFO_ICON_WEAK_DEFAULT = "124,173,247,1";
export const INFO_ICON_WEAK_HOVER = "124,173,247,1";
export const INFO_ICON_WEAK_FOCUS = "124,173,247,1";
export const INFO_ICON_WEAK_ACTIVE = "124,173,247,1";
export const INFO_ICON_WEAK_DISABLED = "89,96,105,1";
export const INFO_ICON_IN_DEFAULT = "35,40,46,1";
export const INFO_ICON_IN_HOVER = "35,40,46,1";
export const INFO_ICON_IN_FOCUS = "35,40,46,1";
export const INFO_ICON_IN_ACTIVE = "35,40,46,1";
export const INFO_ICON_IN_DISABLED = "35,40,46,1";
export const INFO_BORDER_DEFAULT = "124,173,247,1";
export const INFO_BORDER_HOVER = "124,173,247,1";
export const INFO_BORDER_FOCUS = "124,173,247,1";
export const INFO_BORDER_ACTIVE = "124,173,247,1";
export const INFO_BORDER_DISABLED = "124,173,247,1";
export const INFO_BORDER_WEAK_DEFAULT = "35,40,46,1";
export const INFO_BORDER_WEAK_HOVER = "35,40,46,1";
export const INFO_BORDER_WEAK_FOCUS = "35,40,46,1";
export const INFO_BORDER_WEAK_ACTIVE = "35,40,46,1";
export const INFO_BORDER_WEAK_DISABLED = "89,96,105,1";
export const DORMANT_BG_DEFAULT = "111,118,128,1";
export const DORMANT_BG_DISABLED = "111,118,128,1";
export const DORMANT_BG_MEDIUM_DEFAULT = "70,76,84,1";
export const DORMANT_BG_MEDIUM_DISABLED = "55,60,66,1";
export const DORMANT_ICON_DEFAULT = "193,198,204,1";
export const DORMANT_ICON_DISABLED = "89,96,105,1";
export const DORMANT_ICON_MEDIUM_DEFAULT = "255,255,255,1";
export const DORMANT_ICON_IN_DEFAULT = "35,40,46,1";
export const DORMANT_ICON_IN_DISABLED = "35,40,46,1";
export const DORMANT_ICON_IN_MEDIUM_DEFAULT = "35,40,46,1";
export const DORMANT_BORDER_DEFAULT = "167,173,181,1";
export const DORMANT_BORDER_DISABLED = "89,96,105,1";
export const ACCENT_A_DEFAULT = "156,166,255,1";
export const ACCENT_A_HOVER = "138,149,255,1";
export const ACCENT_A_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_B_DEFAULT = "227,68,124,1";
export const ACCENT_B_HOVER = "207,58,122,1";
export const ACCENT_B_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_C_DEFAULT = "252,179,200,1";
export const ACCENT_C_HOVER = "255,135,169,1";
export const ACCENT_C_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_D_DEFAULT = "155,95,245,1";
export const ACCENT_D_HOVER = "141,78,237,1";
export const ACCENT_D_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_E_DEFAULT = "157,186,76,1";
export const ACCENT_E_HOVER = "137,171,44,1";
export const ACCENT_E_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_F_DEFAULT = "217,90,26,1";
export const ACCENT_F_HOVER = "196,79,20,1";
export const ACCENT_F_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_G_DEFAULT = "74,217,217,1";
export const ACCENT_G_HOVER = "23,194,194,1";
export const ACCENT_G_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_H_DEFAULT = "2,142,153,1";
export const ACCENT_H_HOVER = "1,129,140,1";
export const ACCENT_H_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_I_DEFAULT = "245,130,216,1";
export const ACCENT_I_HOVER = "242,109,209,1";
export const ACCENT_I_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_J_DEFAULT = "118,126,178,1";
export const ACCENT_J_HOVER = "104,113,163,1";
export const ACCENT_J_WEAK_DEFAULT = "35,40,46,1";
export const ACCENT_K_DEFAULT = "76,191,127,1";
export const ACCENT_K_HOVER = "54,178,110,1";
export const ACCENT_K_WEAK_DEFAULT = "35,40,46,1";
export const IN_PROGRESS_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const IN_PROGRESS_BG_MEDIUM_HOVER = "35,40,46,1";
export const IN_PROGRESS_BG_MEDIUM_FOCUS = "35,40,46,1";
export const IN_PROGRESS_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const IN_PROGRESS_BG_MEDIUM_DISABLED = "55,60,66,1";
export const IN_PROGRESS_ICON_DEFAULT = "124,173,247,1";
export const IN_PROGRESS_ICON_HOVER = "124,173,247,1";
export const IN_PROGRESS_ICON_FOCUS = "124,173,247,1";
export const IN_PROGRESS_ICON_ACTIVE = "124,173,247,1";
export const IN_PROGRESS_ICON_DISABLED = "89,96,105,1";
export const IN_PROGRESS_ICON_IN_DEFAULT = "35,40,46,1";
export const IN_PROGRESS_ICON_IN_HOVER = "35,40,46,1";
export const IN_PROGRESS_ICON_IN_FOCUS = "35,40,46,1";
export const IN_PROGRESS_ICON_IN_ACTIVE = "35,40,46,1";
export const IN_PROGRESS_ICON_IN_DISABLED = "35,40,46,1";
export const IN_PROGRESS_BORDER_DEFAULT = "124,173,247,1";
export const IN_PROGRESS_BORDER_HOVER = "124,173,247,1";
export const IN_PROGRESS_BORDER_FOCUS = "124,173,247,1";
export const IN_PROGRESS_BORDER_ACTIVE = "124,173,247,1";
export const IN_PROGRESS_BORDER_DISABLED = "124,173,247,1";
export const DECENT_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const DECENT_BG_MEDIUM_HOVER = "35,40,46,1";
export const DECENT_BG_MEDIUM_FOCUS = "35,40,46,1";
export const DECENT_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const DECENT_BG_MEDIUM_DISABLED = "55,60,66,1";
export const DECENT_ICON_DEFAULT = "125,161,27,1";
export const DECENT_ICON_HOVER = "125,161,27,1";
export const DECENT_ICON_FOCUS = "125,161,27,1";
export const DECENT_ICON_ACTIVE = "125,161,27,1";
export const DECENT_ICON_DISABLED = "89,96,105,1";
export const DECENT_ICON_IN_DEFAULT = "35,40,46,1";
export const DECENT_ICON_IN_HOVER = "35,40,46,1";
export const DECENT_ICON_IN_FOCUS = "35,40,46,1";
export const DECENT_ICON_IN_ACTIVE = "35,40,46,1";
export const DECENT_ICON_IN_DISABLED = "35,40,46,1";
export const DECENT_BORDER_DEFAULT = "108,140,20,1";
export const DECENT_BORDER_HOVER = "108,140,20,1";
export const DECENT_BORDER_FOCUS = "108,140,20,1";
export const DECENT_BORDER_ACTIVE = "108,140,20,1";
export const DECENT_BORDER_DISABLED = "108,140,20,1";
export const EXCELLENT_BG_MEDIUM_DEFAULT = "35,40,46,1";
export const EXCELLENT_BG_MEDIUM_HOVER = "35,40,46,1";
export const EXCELLENT_BG_MEDIUM_FOCUS = "35,40,46,1";
export const EXCELLENT_BG_MEDIUM_ACTIVE = "35,40,46,1";
export const EXCELLENT_BG_MEDIUM_DISABLED = "55,60,66,1";
export const EXCELLENT_ICON_DEFAULT = "76,191,127,1";
export const EXCELLENT_ICON_HOVER = "76,191,127,1";
export const EXCELLENT_ICON_FOCUS = "76,191,127,1";
export const EXCELLENT_ICON_ACTIVE = "76,191,127,1";
export const EXCELLENT_ICON_DISABLED = "89,96,105,1";
export const EXCELLENT_ICON_IN_DEFAULT = "35,40,46,1";
export const EXCELLENT_ICON_IN_HOVER = "35,40,46,1";
export const EXCELLENT_ICON_IN_FOCUS = "35,40,46,1";
export const EXCELLENT_ICON_IN_ACTIVE = "35,40,46,1";
export const EXCELLENT_ICON_IN_DISABLED = "35,40,46,1";
export const EXCELLENT_BORDER_DEFAULT = "76,191,127,1";
export const EXCELLENT_BORDER_HOVER = "76,191,127,1";
export const EXCELLENT_BORDER_FOCUS = "76,191,127,1";
export const EXCELLENT_BORDER_ACTIVE = "76,191,127,1";
export const EXCELLENT_BORDER_DISABLED = "76,191,127,1";
