// @ciscodesignsystems/cds-magnetic-theme-web/js/dist/token-theme-classic-light-es6.js
/**
 * This file is auto-generated. Do not edit it directly.
 */

export const BRAND_BG_DEFAULT = "#373c42";
export const BRAND_BG_HOVER = "#373c42";
export const BRAND_BG_FOCUS = "#464c54";
export const BRAND_BG_ACTIVE = "#464c54";
export const BRAND_BG_DISABLED = "#e1e4e8";
export const BRAND_BG_GRADIENT_START = "#373c42";
export const BRAND_BG_GRADIENT_END = "#0f1214";
export const BRAND_BG_STRONG_DEFAULT = "#373c42";
export const BRAND_BG_STRONG_HOVER = "#373c42";
export const BRAND_BG_STRONG_FOCUS = "#373c42";
export const BRAND_BG_STRONG_ACTIVE = "#373c42";
export const BRAND_TEXT_DEFAULT = "#f7f7f7";
export const BRAND_TEXT_HOVER = "#f7f7f7";
export const BRAND_TEXT_ACTIVE = "#f7f7f7";
export const BRAND_TEXT_WEAK_DEFAULT = "#889099";
export const BRAND_TEXT_WEAK_HOVER = "#f7f7f7";
export const BRAND_TEXT_WEAK_FOCUS = "#f7f7f7";
export const BRAND_TEXT_WEAK_ACTIVE = "#f7f7f7";
export const BRAND_TEXT_STRONG_DEFAULT = "#f0f1f2";
export const BRAND_ICON_DEFAULT = "#f7f7f7";
export const BRAND_ICON_HOVER = "#ffffff";
export const BRAND_ICON_ACTIVE = "#ffffff";
export const BRAND_ICON_WEAK_DEFAULT = "#f7f7f7";
export const BRAND_ICON_WEAK_HOVER = "#f7f7f7";
export const BRAND_ICON_WEAK_FOCUS = "#f7f7f7";
export const BRAND_ICON_WEAK_ACTIVE = "#f7f7f7";
export const BRAND_BORDER_ACTIVE = "#cce1ff";
export const BRAND_ACCENT_ICON_DEFAULT = "#7e868f";
export const BRAND_ACCENT_ICON_FOCUS = "#ffffff";
export const BRAND_ACCENT_ICON_ACTIVE = "#45991f";
export const BRAND_ACCENT_ICON_DISABLED = "#d0d4d9";
export const BRAND_ACCENT_ICON_WEAK_DEFAULT = "#c1c6cc";
export const BRAND_ACCENT_ICON_WEAK_FOCUS = "#ffffff";
export const BRAND_ACCENT_ICON_WEAK_ACTIVE = "#6bbf41";
export const BRAND_ACCENT_ICON_WEAK_DISABLED = "#d0d4d9";
export const BRAND_ACCENT_ICON_MEDIUM_DEFAULT = "#a7adb5";
export const BRAND_ACCENT_ICON_MEDIUM_FOCUS = "#ffffff";
export const BRAND_ACCENT_ICON_MEDIUM_ACTIVE = "#52a62b";
export const BRAND_ACCENT_ICON_MEDIUM_DISABLED = "#d0d4d9";
export const BRAND_ACCENT_BORDER_DEFAULT = "#52a62b";
export const BRAND_ACCENT_BORDER_HOVER = "#e1e4e8";
export const BRAND_ACCENT_BORDER_FOCUS = "#45991f";
export const BRAND_ACCENT_BORDER_ACTIVE = "#45991f";
export const BRAND_ACCENT_BORDER_WEAK_DEFAULT = "#d0d4d9";
export const BRAND_ACCENT_BORDER_WEAK_HOVER = "#ffffff";
export const BRAND_ACCENT_BORDER_WEAK_FOCUS = "#ffffff";
export const BRAND_ACCENT_BORDER_WEAK_ACTIVE = "#6bbf41";
export const BRAND_ACCENT_BORDER_WEAK_DISABLED = "#ffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_DEFAULT = "#f0f1f2";
export const BRAND_ACCENT_BORDER_MEDIUM_HOVER = "#ffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_FOCUS = "#ffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_ACTIVE = "#e0f5d5";
export const BRAND_ACCENT_BORDER_MEDIUM_DISABLED = "#ffffff";
export const BRAND_ACCENT_BORDER_STRONG_DEFAULT = "#f7f7f7";
export const BRAND_ACCENT_BORDER_STRONG_HOVER = "#f7f7f7";
export const BRAND_ACCENT_BORDER_STRONG_FOCUS = "#f7f7f7";
export const BRAND_ACCENT_BORDER_STRONG_ACTIVE = "#f7f7f7";
export const BRAND_ACCENT_BORDER_STRONG_DISABLED = "#ffffff";
export const BASE_BG_DEFAULT = "#f7f7f7";
export const BASE_BG_HOVER = "#f7f7f7";
export const BASE_BG_ACTIVE = "#f7f7f7";
export const BASE_BG_DISABLED = "#f0f1f2";
export const BASE_BG_WEAK_DEFAULT = "#ffffff";
export const BASE_BG_WEAK_HOVER = "#e1e4e8cc";
export const BASE_BG_WEAK_ACTIVE = "#f0f1f2";
export const BASE_BG_WEAK_DISABLED = "#f0f1f2";
export const BASE_BG_MEDIUM_DEFAULT = "#f7f7f7";
export const BASE_BG_MEDIUM_HOVER = "#d0d4d9";
export const BASE_BG_MEDIUM_ACTIVE = "#e1e4e8";
export const BASE_BG_MEDIUM_DISABLED = "#f7f7f7";
export const BASE_BG_STRONG_DEFAULT = "#656c75";
export const BASE_BG_STRONG_HOVER = "#23282e";
export const BASE_BG_STRONG_FOCUS = "#23282e";
export const BASE_BG_STRONG_ACTIVE = "#23282e";
export const BASE_BG_STRONG_DISABLED = "#d0d4d9";
export const BASE_ICON_IN_DEFAULT = "#ffffff";
export const BASE_ICON_IN_HOVER = "#ffffff";
export const BASE_ICON_IN_FOCUS = "#ffffff";
export const BASE_ICON_IN_ACTIVE = "#ffffff";
export const BASE_ICON_IN_DISABLED = "#ffffff";
export const BASE_TEXT_DEFAULT = "#23282e";
export const BASE_TEXT_DISABLED = "#c1c6cc";
export const BASE_TEXT_WEAK_DEFAULT = "#889099";
export const BASE_TEXT_WEAK_DISABLED = "#c1c6cc";
export const BASE_TEXT_MEDIUM_DEFAULT = "#596069";
export const BASE_TEXT_MEDIUM_DISABLED = "#c1c6cc";
export const BASE_TEXT_STRONG_DEFAULT = "#373c42";
export const BASE_TEXT_STRONG_DISABLED = "#c1c6cc";
export const BASE_TEXT_IN_DEFAULT = "#ffffff";
export const BASE_TEXT_IN_HOVER = "#ffffff";
export const BASE_TEXT_IN_FOCUS = "#ffffff";
export const BASE_TEXT_IN_ACTIVE = "#ffffff";
export const BASE_TEXT_IN_DISABLED = "#ffffff";
export const BASE_TEXT_IN_WEAK_DEFAULT = "#f7f7f7";
export const BASE_BORDER_DEFAULT = "#e1e4e8";
export const BASE_BORDER_WEAK_DEFAULT = "#ffffff";
export const BASE_BORDER_MEDIUM_DEFAULT = "#e1e4e8";
export const BASE_BORDER_STRONG_DEFAULT = "#656c75";
export const INVERSE_BG_DEFAULT = "#373c42";
export const INVERSE_BG_HOVER = "#373c42";
export const INVERSE_BG_FOCUS = "#464c54";
export const INVERSE_BG_ACTIVE = "#464c54";
export const INVERSE_BG_DISABLED = "#e1e4e8";
export const INVERSE_BG_GRADIENT_START = "#373c42";
export const INVERSE_BG_GRADIENT_END = "#0f1214";
export const INVERSE_BG_WEAK_DEFAULT = "#373c42";
export const INVERSE_TEXT_DEFAULT = "#f7f7f7";
export const INVERSE_TEXT_HOVER = "#f7f7f7";
export const INVERSE_TEXT_FOCUS = "#f7f7f7";
export const INVERSE_TEXT_ACTIVE = "#f7f7f7";
export const INVERSE_TEXT_DISABLED = "#656c75";
export const INVERSE_TEXT_MEDIUM_DEFAULT = "#c1c6cc";
export const INVERSE_TEXT_MEDIUM_DISABLED = "#656c75";
export const INVERSE_ICON_DEFAULT = "#f7f7f7";
export const INVERSE_ICON_HOVER = "#ffffff";
export const INVERSE_ICON_FOCUS = "#ffffff";
export const INVERSE_ICON_ACTIVE = "#ffffff";
export const INVERSE_ICON_DISABLED = "#6f7680";
export const INVERSE_ICON_STRONG_DEFAULT = "#ffffff";
export const INVERSE_ICON_STRONG_HOVER = "#ffffff";
export const INVERSE_ICON_STRONG_FOCUS = "#ffffff";
export const INVERSE_ICON_STRONG_ACTIVE = "#ffffff";
export const INVERSE_ICON_STRONG_DISABLED = "#ffffff";
export const INVERSE_BORDER_DEFAULT = "#6f7680";
export const INVERSE_BORDER_HOVER = "#649ef5";
export const INVERSE_BORDER_FOCUS = "#cce1ff";
export const INVERSE_BORDER_ACTIVE = "#cce1ff";
export const INVERSE_BORDER_DISABLED = "#e1e4e8";
export const CONTROL_BG_DEFAULT = "#ffffff";
export const CONTROL_BG_HOVER = "#e3eeff";
export const CONTROL_BG_FOCUS = "#f0f1f2";
export const CONTROL_BG_ACTIVE = "#f0f1f2";
export const CONTROL_BG_DISABLED = "#f7f7f7";
export const CONTROL_BG_WEAK_DEFAULT = "#ffffff";
export const CONTROL_BG_WEAK_HOVER = "#cce1ff";
export const CONTROL_BG_WEAK_FOCUS = "#e3eeff";
export const CONTROL_BG_WEAK_ACTIVE = "#e3eeff";
export const CONTROL_BG_WEAK_DISABLED = "#f7f7f7";
export const CONTROL_BG_MEDIUM_DEFAULT = "#ffffff";
export const CONTROL_BG_MEDIUM_HOVER = "#f7f7f7";
export const CONTROL_BG_MEDIUM_FOCUS = "#e3eeff";
export const CONTROL_BG_MEDIUM_ACTIVE = "#e3eeff";
export const CONTROL_BG_MEDIUM_DISABLED = "#e1e4e8";
export const CONTROL_BG_STRONG_DEFAULT = "#2774d9";
export const CONTROL_BG_STRONG_HOVER = "#0d5cbd";
export const CONTROL_BG_STRONG_FOCUS = "#0d5cbd";
export const CONTROL_BG_STRONG_ACTIVE = "#0d5cbd";
export const CONTROL_BG_STRONG_DISABLED = "#bad6ff";
export const CONTROL_ICON_DEFAULT = "#1d69cc";
export const CONTROL_ICON_HOVER = "#0d5cbd";
export const CONTROL_ICON_FOCUS = "#0d5cbd";
export const CONTROL_ICON_ACTIVE = "#0d5cbd";
export const CONTROL_ICON_DISABLED = "#a3c8ff";
export const CONTROL_ICON_WEAK_DEFAULT = "#7e868f";
export const CONTROL_ICON_WEAK_HOVER = "#2774d9";
export const CONTROL_ICON_WEAK_FOCUS = "#2774d9";
export const CONTROL_ICON_WEAK_ACTIVE = "#2774d9";
export const CONTROL_ICON_WEAK_DISABLED = "#d0d4d9";
export const CONTROL_ICON_MEDIUM_DEFAULT = "#2774d9";
export const CONTROL_ICON_MEDIUM_HOVER = "#0d5cbd";
export const CONTROL_ICON_MEDIUM_FOCUS = "#0d5cbd";
export const CONTROL_ICON_MEDIUM_ACTIVE = "#0d5cbd";
export const CONTROL_ICON_MEDIUM_DISABLED = "#a3c8ff";
export const CONTROL_ICON_STRONG_DEFAULT = "#656c75";
export const CONTROL_ICON_STRONG_HOVER = "#464c54";
export const CONTROL_ICON_STRONG_FOCUS = "#1d69cc";
export const CONTROL_ICON_STRONG_ACTIVE = "#1d69cc";
export const CONTROL_ICON_STRONG_DISABLED = "#c1c6cc";
export const CONTROL_ICON_IN_DEFAULT = "#ffffff";
export const CONTROL_ICON_IN_HOVER = "#ffffff";
export const CONTROL_ICON_IN_FOCUS = "#ffffff";
export const CONTROL_ICON_IN_ACTIVE = "#ffffff";
export const CONTROL_ICON_IN_DISABLED = "#c1c6cc";
export const CONTROL_ICON_IN_STRONG_DEFAULT = "#2774d9";
export const CONTROL_ICON_IN_STRONG_HOVER = "#2774d9";
export const CONTROL_ICON_IN_STRONG_FOCUS = "#2774d9";
export const CONTROL_ICON_IN_STRONG_ACTIVE = "#2774d9";
export const CONTROL_ICON_IN_STRONG_DISABLED = "#c1c6cc";
export const CONTROL_BORDER_DEFAULT = "#889099";
export const CONTROL_BORDER_HOVER = "#3e84e5";
export const CONTROL_BORDER_FOCUS = "#3e84e5";
export const CONTROL_BORDER_ACTIVE = "#3e84e5";
export const CONTROL_BORDER_DISABLED = "#d0d4d9";
export const CONTROL_BORDER_WEAK_DEFAULT = "#e1e4e8";
export const CONTROL_BORDER_WEAK_HOVER = "#7cadf7";
export const CONTROL_BORDER_WEAK_FOCUS = "#7cadf7";
export const CONTROL_BORDER_WEAK_ACTIVE = "#7cadf7";
export const CONTROL_BORDER_WEAK_DISABLED = "#d0d4d9";
export const CONTROL_BORDER_MEDIUM_DEFAULT = "#2774d9";
export const CONTROL_BORDER_MEDIUM_HOVER = "#bad6ff";
export const CONTROL_BORDER_MEDIUM_FOCUS = "#2774d9";
export const CONTROL_BORDER_MEDIUM_ACTIVE = "#2774d9";
export const CONTROL_BORDER_MEDIUM_DISABLED = "#d0d4d9";
export const CONTROL_BORDER_STRONG_DEFAULT = "#656c75";
export const CONTROL_BORDER_STRONG_HOVER = "#656c75";
export const CONTROL_BORDER_STRONG_FOCUS = "#2774d9";
export const CONTROL_BORDER_STRONG_ACTIVE = "#2774d9";
export const CONTROL_BORDER_STRONG_DISABLED = "#d0d4d9";
export const INTERACT_BG_DEFAULT = "#1d69cc";
export const INTERACT_BG_HOVER = "#0d5cbd";
export const INTERACT_BG_FOCUS = "#0d5cbd";
export const INTERACT_BG_ACTIVE = "#0d5cbd";
export const INTERACT_BG_DISABLED = "#a3c8ff";
export const INTERACT_BG_WEAK_DEFAULT = "#ffffff";
export const INTERACT_BG_WEAK_HOVER = "#cce1ff";
export const INTERACT_BG_WEAK_FOCUS = "#cce1ff";
export const INTERACT_BG_WEAK_ACTIVE = "#cce1ff";
export const INTERACT_BG_WEAK_DISABLED = "#ffffff";
export const INTERACT_BG_MEDIUM_DEFAULT = "#2774d9";
export const INTERACT_BG_MEDIUM_ACTIVE = "#cce1ff";
export const INTERACT_BG_STRONG_ACTIVE = "#f7f7f7";
export const INTERACT_BG_STRONG_HOVER = "#f7f7f7";
export const INTERACT_TEXT_DEFAULT = "#1d69cc";
export const INTERACT_TEXT_HOVER = "#0d5cbd";
export const INTERACT_TEXT_FOCUS = "#0d5cbd";
export const INTERACT_TEXT_ACTIVE = "#0d5cbd";
export const INTERACT_TEXT_DISABLED = "#a3c8ff";
export const INTERACT_TEXT_WEAK_DEFAULT = "#656c75";
export const INTERACT_TEXT_WEAK_HOVER = "#1d69cc";
export const INTERACT_TEXT_WEAK_FOCUS = "#1d69cc";
export const INTERACT_TEXT_WEAK_ACTIVE = "#1d69cc";
export const INTERACT_TEXT_WEAK_DISABLED = "#c1c6cc";
export const INTERACT_TEXT_IN_DEFAULT = "#ffffff";
export const INTERACT_TEXT_IN_HOVER = "#ffffff";
export const INTERACT_TEXT_IN_FOCUS = "#ffffff";
export const INTERACT_TEXT_IN_ACTIVE = "#ffffff";
export const INTERACT_TEXT_IN_DISABLED = "#ffffff";
export const INTERACT_ICON_DEFAULT = "#1d69cc";
export const INTERACT_ICON_HOVER = "#0d5cbd";
export const INTERACT_ICON_FOCUS = "#0d5cbd";
export const INTERACT_ICON_ACTIVE = "#0d5cbd";
export const INTERACT_ICON_DISABLED = "#a3c8ff";
export const INTERACT_ICON_WEAK_DEFAULT = "#7e868f";
export const INTERACT_ICON_WEAK_HOVER = "#6f7680";
export const INTERACT_ICON_WEAK_FOCUS = "#6f7680";
export const INTERACT_ICON_WEAK_ACTIVE = "#6f7680";
export const INTERACT_ICON_WEAK_DISABLED = "#d0d4d9";
export const INTERACT_ICON_MEDIUM_DEFAULT = "#7e868f";
export const INTERACT_ICON_MEDIUM_HOVER = "#2774d9";
export const INTERACT_ICON_MEDIUM_FOCUS = "#2774d9";
export const INTERACT_ICON_MEDIUM_ACTIVE = "#2774d9";
export const INTERACT_ICON_MEDIUM_DISABLED = "#c1c6cc";
export const INTERACT_ICON_STRONG_DEFAULT = "#656c75";
export const INTERACT_ICON_STRONG_HOVER = "#464c54";
export const INTERACT_ICON_STRONG_FOCUS = "#464c54";
export const INTERACT_ICON_STRONG_ACTIVE = "#464c54";
export const INTERACT_ICON_STRONG_DISABLED = "#a7adb5";
export const INTERACT_ICON_IN_DEFAULT = "#ffffff";
export const INTERACT_ICON_IN_HOVER = "#ffffff";
export const INTERACT_ICON_IN_FOCUS = "#ffffff";
export const INTERACT_ICON_IN_ACTIVE = "#ffffff";
export const INTERACT_ICON_IN_DISABLED = "#ffffff";
export const INTERACT_BORDER_DEFAULT = "#1d69cc";
export const INTERACT_BORDER_HOVER = "#0d5cbd";
export const INTERACT_BORDER_FOCUS = "#0d5cbd";
export const INTERACT_BORDER_ACTIVE = "#0d5cbd";
export const INTERACT_BORDER_DISABLED = "#a3c8ff";
export const INTERACT_BORDER_WEAK_DEFAULT = "#e1e4e8";
export const INTERACT_BORDER_WEAK_HOVER = "#cce1ff";
export const INTERACT_BORDER_WEAK_FOCUS = "#e1e4e8";
export const INTERACT_BORDER_WEAK_ACTIVE = "#e1e4e8";
export const INTERACT_BORDER_WEAK_DISABLED = "#e1e4e8";
export const INTERACT_BORDER_MEDIUM_DEFAULT = "#2774d9";
export const INTERACT_BORDER_MEDIUM_HOVER = "#bad6ff";
export const INTERACT_BORDER_MEDIUM_FOCUS = "#2774d9";
export const INTERACT_BORDER_MEDIUM_ACTIVE = "#2774d9";
export const INTERACT_BORDER_MEDIUM_DISABLED = "#e1e4e8";
export const POSITIVE_BG_DEFAULT = "#45991f";
export const POSITIVE_BG_HOVER = "#45991f";
export const POSITIVE_BG_FOCUS = "#45991f";
export const POSITIVE_BG_ACTIVE = "#45991f";
export const POSITIVE_BG_DISABLED = "#b0e396";
export const POSITIVE_BG_WEAK_DEFAULT = "#ffffff";
export const POSITIVE_BG_WEAK_HOVER = "#ffffff";
export const POSITIVE_BG_WEAK_FOCUS = "#ffffff";
export const POSITIVE_BG_WEAK_ACTIVE = "#ffffff";
export const POSITIVE_BG_WEAK_DISABLED = "#ffffff";
export const POSITIVE_BG_MEDIUM_DEFAULT = "#e0f5d5";
export const POSITIVE_BG_MEDIUM_HOVER = "#e0f5d5";
export const POSITIVE_BG_MEDIUM_FOCUS = "#e0f5d5";
export const POSITIVE_BG_MEDIUM_ACTIVE = "#e0f5d5";
export const POSITIVE_BG_MEDIUM_DISABLED = "#f7f7f7";
export const POSITIVE_BG_STRONG_DEFAULT = "#e0f5d5";
export const POSITIVE_ICON_DEFAULT = "#45991f";
export const POSITIVE_ICON_HOVER = "#45991f";
export const POSITIVE_ICON_FOCUS = "#45991f";
export const POSITIVE_ICON_ACTIVE = "#45991f";
export const POSITIVE_ICON_DISABLED = "#d0d4d9";
export const POSITIVE_ICON_WEAK_DEFAULT = "#b0e396";
export const POSITIVE_ICON_WEAK_HOVER = "#b0e396";
export const POSITIVE_ICON_WEAK_FOCUS = "#b0e396";
export const POSITIVE_ICON_WEAK_ACTIVE = "#b0e396";
export const POSITIVE_ICON_WEAK_DISABLED = "#b0e396";
export const POSITIVE_ICON_STRONG_DEFAULT = "#45991f";
export const POSITIVE_ICON_STRONG_HOVER = "#45991f";
export const POSITIVE_ICON_STRONG_FOCUS = "#45991f";
export const POSITIVE_ICON_STRONG_ACTIVE = "#45991f";
export const POSITIVE_ICON_STRONG_DISABLED = "#d0d4d9";
export const POSITIVE_ICON_IN_DEFAULT = "#ffffff";
export const POSITIVE_ICON_IN_HOVER = "#ffffff";
export const POSITIVE_ICON_IN_FOCUS = "#ffffff";
export const POSITIVE_ICON_IN_ACTIVE = "#ffffff";
export const POSITIVE_ICON_IN_DISABLED = "#ffffff";
export const POSITIVE_BORDER_DEFAULT = "#45991f";
export const POSITIVE_BORDER_HOVER = "#45991f";
export const POSITIVE_BORDER_FOCUS = "#45991f";
export const POSITIVE_BORDER_ACTIVE = "#45991f";
export const POSITIVE_BORDER_DISABLED = "#d0d4d9";
export const POSITIVE_BORDER_WEAK_DEFAULT = "#ffffff";
export const POSITIVE_BORDER_WEAK_HOVER = "#ffffff";
export const POSITIVE_BORDER_WEAK_FOCUS = "#ffffff";
export const POSITIVE_BORDER_WEAK_ACTIVE = "#ffffff";
export const POSITIVE_BORDER_WEAK_DISABLED = "#d0d4d9";
export const POSITIVE_BORDER_STRONG_DEFAULT = "#398519";
export const POSITIVE_BORDER_STRONG_HOVER = "#398519";
export const POSITIVE_BORDER_STRONG_FOCUS = "#398519";
export const POSITIVE_BORDER_STRONG_ACTIVE = "#398519";
export const POSITIVE_BORDER_STRONG_DISABLED = "#398519";
export const POSITIVE_TEXT_IN_DEFAULT = "#ffffff";
export const POSITIVE_TEXT_IN_HOVER = "#ffffff";
export const POSITIVE_TEXT_IN_FOCUS = "#ffffff";
export const POSITIVE_TEXT_IN_ACTIVE = "#ffffff";
export const POSITIVE_TEXT_IN_DISABLED = "#ffffff";
export const WARNING_BG_DEFAULT = "#cc8604";
export const WARNING_BG_HOVER = "#cc8604";
export const WARNING_BG_FOCUS = "#cc8604";
export const WARNING_BG_ACTIVE = "#cc8604";
export const WARNING_BG_DISABLED = "#f2d268";
export const WARNING_BG_WEAK_DEFAULT = "#ffffff";
export const WARNING_BG_WEAK_HOVER = "#ffffff";
export const WARNING_BG_WEAK_FOCUS = "#ffffff";
export const WARNING_BG_WEAK_ACTIVE = "#ffffff";
export const WARNING_BG_WEAK_DISABLED = "#ffffff";
export const WARNING_BG_MEDIUM_DEFAULT = "#faefb9";
export const WARNING_BG_MEDIUM_HOVER = "#faefb9";
export const WARNING_BG_MEDIUM_FOCUS = "#faefb9";
export const WARNING_BG_MEDIUM_ACTIVE = "#faefb9";
export const WARNING_BG_MEDIUM_DISABLED = "#f7f7f7";
export const WARNING_BG_STRONG_DEFAULT = "#faefb9";
export const WARNING_ICON_DEFAULT = "#cc8604";
export const WARNING_ICON_HOVER = "#cc8604";
export const WARNING_ICON_FOCUS = "#cc8604";
export const WARNING_ICON_ACTIVE = "#cc8604";
export const WARNING_ICON_DISABLED = "#d0d4d9";
export const WARNING_ICON_STRONG_DEFAULT = "#bd7202";
export const WARNING_ICON_STRONG_HOVER = "#bd7202";
export const WARNING_ICON_STRONG_FOCUS = "#bd7202";
export const WARNING_ICON_STRONG_ACTIVE = "#bd7202";
export const WARNING_ICON_STRONG_DISABLED = "#d0d4d9";
export const WARNING_ICON_IN_DEFAULT = "#ffffff";
export const WARNING_ICON_IN_HOVER = "#ffffff";
export const WARNING_ICON_IN_FOCUS = "#ffffff";
export const WARNING_ICON_IN_ACTIVE = "#ffffff";
export const WARNING_ICON_IN_DISABLED = "#ffffff";
export const WARNING_BORDER_DEFAULT = "#cc8604";
export const WARNING_BORDER_HOVER = "#cc8604";
export const WARNING_BORDER_FOCUS = "#cc8604";
export const WARNING_BORDER_ACTIVE = "#cc8604";
export const WARNING_BORDER_DISABLED = "#d0d4d9";
export const WARNING_BORDER_WEAK_DEFAULT = "#ffffff";
export const WARNING_BORDER_WEAK_HOVER = "#ffffff";
export const WARNING_BORDER_WEAK_FOCUS = "#ffffff";
export const WARNING_BORDER_WEAK_ACTIVE = "#ffffff";
export const WARNING_BORDER_WEAK_DISABLED = "#d0d4d9";
export const WARNING_BORDER_STRONG_DEFAULT = "#a65503";
export const WARNING_BORDER_STRONG_HOVER = "#a65503";
export const WARNING_BORDER_STRONG_FOCUS = "#a65503";
export const WARNING_BORDER_STRONG_ACTIVE = "#a65503";
export const WARNING_BORDER_STRONG_DISABLED = "#a65503";
export const WARNING_TEXT_IN_DEFAULT = "#ffffff";
export const WARNING_TEXT_IN_HOVER = "#ffffff";
export const WARNING_TEXT_IN_FOCUS = "#ffffff";
export const WARNING_TEXT_IN_ACTIVE = "#ffffff";
export const WARNING_TEXT_IN_DISABLED = "#ffffff";
export const SEVERE_WARNING_BG_DEFAULT = "#f26722";
export const SEVERE_WARNING_BG_HOVER = "#f26722";
export const SEVERE_WARNING_BG_FOCUS = "#f26722";
export const SEVERE_WARNING_BG_ACTIVE = "#f26722";
export const SEVERE_WARNING_BG_DISABLED = "#fcc9a7";
export const SEVERE_WARNING_BG_WEAK_DEFAULT = "#ffffff";
export const SEVERE_WARNING_BG_WEAK_HOVER = "#ffffff";
export const SEVERE_WARNING_BG_WEAK_FOCUS = "#ffffff";
export const SEVERE_WARNING_BG_WEAK_ACTIVE = "#ffffff";
export const SEVERE_WARNING_BG_WEAK_DISABLED = "#ffffff";
export const SEVERE_WARNING_BG_MEDIUM_DEFAULT = "#ffeadb";
export const SEVERE_WARNING_BG_MEDIUM_HOVER = "#ffeadb";
export const SEVERE_WARNING_BG_MEDIUM_FOCUS = "#ffeadb";
export const SEVERE_WARNING_BG_MEDIUM_ACTIVE = "#ffeadb";
export const SEVERE_WARNING_BG_MEDIUM_DISABLED = "#f7f7f7";
export const SEVERE_WARNING_BG_STRONG_DEFAULT = "#ffeadb";
export const SEVERE_WARNING_ICON_DEFAULT = "#f26722";
export const SEVERE_WARNING_ICON_HOVER = "#d95a1a";
export const SEVERE_WARNING_ICON_FOCUS = "#c44f14";
export const SEVERE_WARNING_ICON_ACTIVE = "#c44f14";
export const SEVERE_WARNING_ICON_DISABLED = "#d0d4d9";
export const SEVERE_WARNING_ICON_STRONG_DEFAULT = "#d95a1a";
export const SEVERE_WARNING_ICON_STRONG_HOVER = "#d95a1a";
export const SEVERE_WARNING_ICON_STRONG_FOCUS = "#d95a1a";
export const SEVERE_WARNING_ICON_STRONG_ACTIVE = "#d95a1a";
export const SEVERE_WARNING_ICON_STRONG_DISABLED = "#d95a1a";
export const SEVERE_WARNING_ICON_IN_DEFAULT = "#ffffff";
export const SEVERE_WARNING_ICON_IN_HOVER = "#ffffff";
export const SEVERE_WARNING_ICON_IN_FOCUS = "#ffffff";
export const SEVERE_WARNING_ICON_IN_ACTIVE = "#ffffff";
export const SEVERE_WARNING_ICON_IN_DISABLED = "#ffffff";
export const SEVERE_WARNING_BORDER_DEFAULT = "#f26722";
export const SEVERE_WARNING_BORDER_HOVER = "#f26722";
export const SEVERE_WARNING_BORDER_FOCUS = "#f26722";
export const SEVERE_WARNING_BORDER_ACTIVE = "#f26722";
export const SEVERE_WARNING_BORDER_DISABLED = "#f26722";
export const SEVERE_WARNING_BORDER_WEAK_DEFAULT = "#ffffff";
export const SEVERE_WARNING_BORDER_WEAK_HOVER = "#ffffff";
export const SEVERE_WARNING_BORDER_WEAK_FOCUS = "#ffffff";
export const SEVERE_WARNING_BORDER_WEAK_ACTIVE = "#ffffff";
export const SEVERE_WARNING_BORDER_WEAK_DISABLED = "#d0d4d9";
export const NEGATIVE_BG_DEFAULT = "#cc2d37";
export const NEGATIVE_BG_HOVER = "#cc2d37";
export const NEGATIVE_BG_FOCUS = "#cc2d37";
export const NEGATIVE_BG_ACTIVE = "#cc2d37";
export const NEGATIVE_BG_DISABLED = "#ffb2b5";
export const NEGATIVE_BG_WEAK_DEFAULT = "#ffffff";
export const NEGATIVE_BG_WEAK_HOVER = "#ffd4d5";
export const NEGATIVE_BG_WEAK_FOCUS = "#ffd4d5";
export const NEGATIVE_BG_WEAK_ACTIVE = "#ffd4d5";
export const NEGATIVE_BG_WEAK_DISABLED = "#ffffff";
export const NEGATIVE_BG_MEDIUM_DEFAULT = "#ffe8e9";
export const NEGATIVE_BG_MEDIUM_HOVER = "#ffe8e9";
export const NEGATIVE_BG_MEDIUM_FOCUS = "#ffe8e9";
export const NEGATIVE_BG_MEDIUM_ACTIVE = "#ffe8e9";
export const NEGATIVE_BG_MEDIUM_DISABLED = "#f7f7f7";
export const NEGATIVE_BG_STRONG_DEFAULT = "#ffe8e9";
export const NEGATIVE_TEXT_DEFAULT = "#cc2d37";
export const NEGATIVE_TEXT_HOVER = "#b2242d";
export const NEGATIVE_TEXT_FOCUS = "#b2242d";
export const NEGATIVE_TEXT_ACTIVE = "#b2242d";
export const NEGATIVE_TEXT_DISABLED = "#ffb2b5";
export const NEGATIVE_TEXT_IN_DEFAULT = "#ffffff";
export const NEGATIVE_TEXT_IN_HOVER = "#ffffff";
export const NEGATIVE_TEXT_IN_FOCUS = "#ffffff";
export const NEGATIVE_TEXT_IN_ACTIVE = "#ffffff";
export const NEGATIVE_TEXT_IN_DISABLED = "#ffffff";
export const NEGATIVE_ICON_DEFAULT = "#cc2d37";
export const NEGATIVE_ICON_HOVER = "#b2242d";
export const NEGATIVE_ICON_FOCUS = "#b2242d";
export const NEGATIVE_ICON_ACTIVE = "#b2242d";
export const NEGATIVE_ICON_DISABLED = "#ffb2b5";
export const NEGATIVE_ICON_WEAK_DEFAULT = "#ffc7c9";
export const NEGATIVE_ICON_WEAK_HOVER = "#ffc7c9";
export const NEGATIVE_ICON_WEAK_FOCUS = "#ffc7c9";
export const NEGATIVE_ICON_WEAK_ACTIVE = "#ffc7c9";
export const NEGATIVE_ICON_WEAK_DISABLED = "#d0d4d9";
export const NEGATIVE_ICON_IN_DEFAULT = "#ffffff";
export const NEGATIVE_ICON_IN_HOVER = "#ffffff";
export const NEGATIVE_ICON_IN_FOCUS = "#ffffff";
export const NEGATIVE_ICON_IN_ACTIVE = "#ffffff";
export const NEGATIVE_ICON_IN_DISABLED = "#ffffff";
export const NEGATIVE_BORDER_DEFAULT = "#cc2d37";
export const NEGATIVE_BORDER_HOVER = "#b2242d";
export const NEGATIVE_BORDER_FOCUS = "#b2242d";
export const NEGATIVE_BORDER_ACTIVE = "#b2242d";
export const NEGATIVE_BORDER_DISABLED = "#ffb2b5";
export const NEGATIVE_BORDER_WEAK_DEFAULT = "#ffffff";
export const NEGATIVE_BORDER_WEAK_HOVER = "#ffffff";
export const NEGATIVE_BORDER_WEAK_FOCUS = "#ffffff";
export const NEGATIVE_BORDER_WEAK_ACTIVE = "#ffffff";
export const NEGATIVE_BORDER_WEAK_DISABLED = "#d0d4d9";
export const NEGATIVE_BORDER_MEDIUM_DEFAULT = "#d93843";
export const NEGATIVE_BORDER_MEDIUM_HOVER = "#d93843";
export const NEGATIVE_BORDER_MEDIUM_FOCUS = "#d93843";
export const NEGATIVE_BORDER_MEDIUM_ACTIVE = "#d93843";
export const NEGATIVE_BORDER_MEDIUM_DISABLED = "#ffb2b5";
export const NEGATIVE_BORDER_STRONG_DEFAULT = "#b2242d";
export const NEGATIVE_BORDER_STRONG_HOVER = "#b2242d";
export const NEGATIVE_BORDER_STRONG_FOCUS = "#b2242d";
export const NEGATIVE_BORDER_STRONG_ACTIVE = "#b2242d";
export const NEGATIVE_BORDER_STRONG_DISABLED = "#ffb2b5";
export const INFO_BG_DEFAULT = "#2774d9";
export const INFO_BG_HOVER = "#2774d9";
export const INFO_BG_FOCUS = "#2774d9";
export const INFO_BG_ACTIVE = "#2774d9";
export const INFO_BG_DISABLED = "#2774d9";
export const INFO_BG_WEAK_DEFAULT = "#ffffff";
export const INFO_BG_WEAK_HOVER = "#ffffff";
export const INFO_BG_WEAK_FOCUS = "#ffffff";
export const INFO_BG_WEAK_ACTIVE = "#ffffff";
export const INFO_BG_WEAK_DISABLED = "#ffffff";
export const INFO_BG_MEDIUM_DEFAULT = "#e3eeff";
export const INFO_BG_MEDIUM_HOVER = "#e3eeff";
export const INFO_BG_MEDIUM_FOCUS = "#e3eeff";
export const INFO_BG_MEDIUM_ACTIVE = "#e3eeff";
export const INFO_BG_MEDIUM_DISABLED = "#f7f7f7";
export const INFO_BG_STRONG_DEFAULT = "#e3eeff";
export const INFO_ICON_DEFAULT = "#2774d9";
export const INFO_ICON_HOVER = "#2774d9";
export const INFO_ICON_FOCUS = "#2774d9";
export const INFO_ICON_ACTIVE = "#2774d9";
export const INFO_ICON_DISABLED = "#d0d4d9";
export const INFO_ICON_WEAK_DEFAULT = "#a3c8ff";
export const INFO_ICON_WEAK_HOVER = "#a3c8ff";
export const INFO_ICON_WEAK_FOCUS = "#a3c8ff";
export const INFO_ICON_WEAK_ACTIVE = "#a3c8ff";
export const INFO_ICON_WEAK_DISABLED = "#a3c8ff";
export const INFO_ICON_IN_DEFAULT = "#ffffff";
export const INFO_ICON_IN_HOVER = "#ffffff";
export const INFO_ICON_IN_FOCUS = "#ffffff";
export const INFO_ICON_IN_ACTIVE = "#ffffff";
export const INFO_ICON_IN_DISABLED = "#ffffff";
export const INFO_BORDER_DEFAULT = "#2774d9";
export const INFO_BORDER_HOVER = "#2774d9";
export const INFO_BORDER_FOCUS = "#2774d9";
export const INFO_BORDER_ACTIVE = "#2774d9";
export const INFO_BORDER_DISABLED = "#d0d4d9";
export const INFO_BORDER_WEAK_DEFAULT = "#ffffff";
export const INFO_BORDER_WEAK_HOVER = "#ffffff";
export const INFO_BORDER_WEAK_FOCUS = "#ffffff";
export const INFO_BORDER_WEAK_ACTIVE = "#ffffff";
export const INFO_BORDER_WEAK_DISABLED = "#d0d4d9";
export const DORMANT_BG_DEFAULT = "#6f7680";
export const DORMANT_BG_DISABLED = "#6f7680";
export const DORMANT_BG_MEDIUM_DEFAULT = "#e1e4e8";
export const DORMANT_BG_MEDIUM_DISABLED = "#f7f7f7";
export const DORMANT_ICON_DEFAULT = "#6f7680";
export const DORMANT_ICON_DISABLED = "#d0d4d9";
export const DORMANT_ICON_MEDIUM_DEFAULT = "#ffffff";
export const DORMANT_ICON_IN_DEFAULT = "#ffffff";
export const DORMANT_ICON_IN_DISABLED = "#ffffff";
export const DORMANT_ICON_IN_MEDIUM_DEFAULT = "#ffffff";
export const DORMANT_BORDER_DEFAULT = "#889099";
export const DORMANT_BORDER_DISABLED = "#d0d4d9";
export const ACCENT_A_DEFAULT = "#7d8aff";
export const ACCENT_A_HOVER = "#6977f0";
export const ACCENT_A_WEAK_DEFAULT = "#ebedff";
export const ACCENT_B_DEFAULT = "#b02863";
export const ACCENT_B_HOVER = "#991d53";
export const ACCENT_B_WEAK_DEFAULT = "#ffe8ef";
export const ACCENT_C_DEFAULT = "#f2638c";
export const ACCENT_C_HOVER = "#e3447c";
export const ACCENT_C_WEAK_DEFAULT = "#ffd4e0";
export const ACCENT_D_DEFAULT = "#753bcc";
export const ACCENT_D_HOVER = "#6732b8";
export const ACCENT_D_WEAK_DEFAULT = "#f3ebff";
export const ACCENT_E_DEFAULT = "#7da11b";
export const ACCENT_E_HOVER = "#6c8c14";
export const ACCENT_E_WEAK_DEFAULT = "#eaf2d3";
export const ACCENT_F_DEFAULT = "#ad3907";
export const ACCENT_F_HOVER = "#942e03";
export const ACCENT_F_WEAK_DEFAULT = "#ffeadb";
export const ACCENT_G_DEFAULT = "#04a4b0";
export const ACCENT_G_HOVER = "#028e99";
export const ACCENT_G_WEAK_DEFAULT = "#d5f5f5";
export const ACCENT_H_DEFAULT = "#006773";
export const ACCENT_H_HOVER = "#005c66";
export const ACCENT_H_WEAK_DEFAULT = "#a9ebeb";
export const ACCENT_I_DEFAULT = "#e85fc6";
export const ACCENT_I_HOVER = "#d649b3";
export const ACCENT_I_WEAK_DEFAULT = "#ffe8f9";
export const ACCENT_J_DEFAULT = "#545c8a";
export const ACCENT_J_HOVER = "#484f7a";
export const ACCENT_J_WEAK_DEFAULT = "#ebedff";
export const ACCENT_K_DEFAULT = "#21a65f";
export const ACCENT_K_HOVER = "#169855";
export const ACCENT_K_WEAK_DEFAULT = "#d4f5e1";
export const IN_PROGRESS_BG_MEDIUM_DEFAULT = "#e3eeff";
export const IN_PROGRESS_BG_MEDIUM_HOVER = "#e3eeff";
export const IN_PROGRESS_BG_MEDIUM_FOCUS = "#e3eeff";
export const IN_PROGRESS_BG_MEDIUM_ACTIVE = "#e3eeff";
export const IN_PROGRESS_BG_MEDIUM_DISABLED = "#f7f7f7";
export const IN_PROGRESS_ICON_DEFAULT = "#2774d9";
export const IN_PROGRESS_ICON_HOVER = "#2774d9";
export const IN_PROGRESS_ICON_FOCUS = "#2774d9";
export const IN_PROGRESS_ICON_ACTIVE = "#2774d9";
export const IN_PROGRESS_ICON_DISABLED = "#d0d4d9";
export const IN_PROGRESS_ICON_IN_DEFAULT = "#ffffff";
export const IN_PROGRESS_ICON_IN_HOVER = "#ffffff";
export const IN_PROGRESS_ICON_IN_FOCUS = "#ffffff";
export const IN_PROGRESS_ICON_IN_ACTIVE = "#ffffff";
export const IN_PROGRESS_ICON_IN_DISABLED = "#ffffff";
export const IN_PROGRESS_BORDER_DEFAULT = "#2774d9";
export const IN_PROGRESS_BORDER_HOVER = "#2774d9";
export const IN_PROGRESS_BORDER_FOCUS = "#2774d9";
export const IN_PROGRESS_BORDER_ACTIVE = "#2774d9";
export const IN_PROGRESS_BORDER_DISABLED = "#d0d4d9";
export const DECENT_BG_MEDIUM_DEFAULT = "#eaf2d3";
export const DECENT_BG_MEDIUM_HOVER = "#eaf2d3";
export const DECENT_BG_MEDIUM_FOCUS = "#eaf2d3";
export const DECENT_BG_MEDIUM_ACTIVE = "#eaf2d3";
export const DECENT_BG_MEDIUM_DISABLED = "#f7f7f7";
export const DECENT_ICON_DEFAULT = "#6c8c14";
export const DECENT_ICON_HOVER = "#6c8c14";
export const DECENT_ICON_FOCUS = "#6c8c14";
export const DECENT_ICON_ACTIVE = "#6c8c14";
export const DECENT_ICON_DISABLED = "#d0d4d9";
export const DECENT_ICON_IN_DEFAULT = "#ffffff";
export const DECENT_ICON_IN_HOVER = "#ffffff";
export const DECENT_ICON_IN_FOCUS = "#ffffff";
export const DECENT_ICON_IN_ACTIVE = "#ffffff";
export const DECENT_ICON_IN_DISABLED = "#ffffff";
export const DECENT_BORDER_DEFAULT = "#7da11b";
export const DECENT_BORDER_HOVER = "#7da11b";
export const DECENT_BORDER_FOCUS = "#7da11b";
export const DECENT_BORDER_ACTIVE = "#7da11b";
export const DECENT_BORDER_DISABLED = "#d0d4d9";
export const EXCELLENT_BG_MEDIUM_DEFAULT = "#d4f5e1";
export const EXCELLENT_BG_MEDIUM_HOVER = "#d4f5e1";
export const EXCELLENT_BG_MEDIUM_FOCUS = "#d4f5e1";
export const EXCELLENT_BG_MEDIUM_ACTIVE = "#d4f5e1";
export const EXCELLENT_BG_MEDIUM_DISABLED = "#f7f7f7";
export const EXCELLENT_ICON_DEFAULT = "#087041";
export const EXCELLENT_ICON_HOVER = "#087041";
export const EXCELLENT_ICON_FOCUS = "#087041";
export const EXCELLENT_ICON_ACTIVE = "#087041";
export const EXCELLENT_ICON_DISABLED = "#d0d4d9";
export const EXCELLENT_ICON_IN_DEFAULT = "#ffffff";
export const EXCELLENT_ICON_IN_HOVER = "#ffffff";
export const EXCELLENT_ICON_IN_FOCUS = "#ffffff";
export const EXCELLENT_ICON_IN_ACTIVE = "#ffffff";
export const EXCELLENT_ICON_IN_DISABLED = "#ffffff";
export const EXCELLENT_BORDER_DEFAULT = "#075e39";
export const EXCELLENT_BORDER_HOVER = "#075e39";
export const EXCELLENT_BORDER_FOCUS = "#075e39";
export const EXCELLENT_BORDER_ACTIVE = "#075e39";
export const EXCELLENT_BORDER_DISABLED = "#d0d4d9";
