// @ciscodesignsystems/cds-react-nav/hooks/use-nav-reducer.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('react');
var index = require('../types/index.cjs.js');

const stateWithAnimating = (prevState, newState) => {
  const animating = prevState.isAnimating || prevState.isOpen !== newState.isOpen;
  return {
    ...newState,
    isAnimating: animating
  };
};
const reducer = (state, action) => {
  const {
    payload,
    type
  } = action;
  const closedState = stateWithAnimating(state, {
    ...state,
    isOpen: false,
    event: undefined
  });
  switch (type) {
    case index.CDSNavReducerActionKind.ANIMATION_END:
      return {
        ...state,
        isAnimating: false
      };
    case index.CDSNavReducerActionKind.HOVER:
      if (state.isOpen && state.event === 'click') {
        return state;
      } else if (!payload.menu) {
        return closedState;
      }
      return stateWithAnimating(state, {
        isOpen: true,
        id: payload.id,
        event: 'hover',
        subMenuStack: state.isOpen && state.id === payload.id ? state.subMenuStack : [payload.menu]
      });
    case index.CDSNavReducerActionKind.UNHOVER:
      if (state.event === 'click' || state.id !== payload.id) {
        return state;
      }
      return closedState;
    case index.CDSNavReducerActionKind.CLICK:
      if (state.event === 'click' && state.id === payload.id || !payload.menu) {
        return closedState;
      }
      return stateWithAnimating(state, {
        isOpen: true,
        id: payload.id,
        event: 'click',
        subMenuStack: [payload.menu]
      });
    case index.CDSNavReducerActionKind.CLOSE:
      return closedState;
    case index.CDSNavReducerActionKind.OPEN_SUBMENU:
      if (!payload.menu) {
        return closedState;
      }
      return {
        ...state,
        subMenuStack: [...state.subMenuStack, payload.menu]
      };
    case index.CDSNavReducerActionKind.BACK_SUBMENU:
      return {
        ...state,
        subMenuStack: [...state.subMenuStack.slice(0, -1)]
      };
    default:
      return state;
  }
};
const useNavReducer = () => {
  return react.useReducer(reducer, {
    isOpen: false,
    event: undefined,
    subMenuStack: [],
    isAnimating: false
  });
};

exports.useNavReducer = useNavReducer;
