// @ciscodesignsystems/cds-react-toggle/toggle.esm.js
import { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './toggle.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';

const CDSToggle = /*#__PURE__*/forwardRef(({
  className,
  checked,
  size,
  disabled,
  children,
  ...rest
}, ref) => {
  const classes = cx(styles, 'cds-toggle', `cds-toggle--${size}`, {
    'cds-toggle--disabled': disabled,
    'cds-toggle--checked': checked
  }, className);
  return /*#__PURE__*/jsxs("label", {
    className: classes,
    children: [/*#__PURE__*/jsx("input", {
      ref: ref,
      ...rest,
      type: "checkbox",
      role: "switch",
      "aria-checked": checked,
      checked: checked,
      disabled: disabled
    }), /*#__PURE__*/jsxs("svg", {
      "aria-hidden": "true",
      className: cx(styles, 'cds-toggle-icon'),
      fill: "none",
      focusable: "false",
      viewBox: "0 0 40 20",
      children: [/*#__PURE__*/jsx("rect", {
        className: cx(styles, 'cds-toggle-icon__background'),
        height: "20",
        rx: "10",
        width: "40"
      }), /*#__PURE__*/jsx("circle", {
        className: cx(styles, 'cds-toggle-icon__thumb'),
        cy: "10",
        r: "7"
      })]
    }), children]
  });
});
CDSToggle.displayName = 'CDSToggle';
CDSToggle.defaultProps = {
  size: 'md'
};

export { CDSToggle, CDSToggle as default };
