// @ciscodesignsystems/cds-react-toggle/toggle.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var toggle_module = require('./toggle.module.cjs.js');

const CDSToggle = /*#__PURE__*/react.forwardRef(({
  className,
  checked,
  size,
  disabled,
  children,
  ...rest
}, ref) => {
  const classes = cdsComponentUtilities.cx(toggle_module["default"], 'cds-toggle', `cds-toggle--${size}`, {
    'cds-toggle--disabled': disabled,
    'cds-toggle--checked': checked
  }, className);
  return jsxRuntime.jsxs("label", {
    className: classes,
    children: [jsxRuntime.jsx("input", {
      ref: ref,
      ...rest,
      type: "checkbox",
      role: "switch",
      "aria-checked": checked,
      checked: checked,
      disabled: disabled
    }), jsxRuntime.jsxs("svg", {
      "aria-hidden": "true",
      className: cdsComponentUtilities.cx(toggle_module["default"], 'cds-toggle-icon'),
      fill: "none",
      focusable: "false",
      viewBox: "0 0 40 20",
      children: [jsxRuntime.jsx("rect", {
        className: cdsComponentUtilities.cx(toggle_module["default"], 'cds-toggle-icon__background'),
        height: "20",
        rx: "10",
        width: "40"
      }), jsxRuntime.jsx("circle", {
        className: cdsComponentUtilities.cx(toggle_module["default"], 'cds-toggle-icon__thumb'),
        cy: "10",
        r: "7"
      })]
    }), children]
  });
});
CDSToggle.displayName = 'CDSToggle';
CDSToggle.defaultProps = {
  size: 'md'
};

exports.CDSToggle = CDSToggle;
exports["default"] = CDSToggle;
