// @ciscodesignsystems/cds-react-table/components/table-pagination/table-pagination.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var phosphorReact = require('phosphor-react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var cdsReactFlex = require('@ciscodesignsystems/cds-react-flex');
var cdsReactSelect = require('@ciscodesignsystems/cds-react-select');
var tablePagination_module = require('./table-pagination.module.cjs.js');

const CDSTablePaginationButton = ({
  index,
  active = false,
  disabled = false,
  onClick = undefined
}) => {
  const classes = cdsComponentUtilities.cx(tablePagination_module["default"], 'cds-table__pagination__button', 'cds-table__pagination__item', {
    'cds-table__pagination__button--active': active
  });
  return jsxRuntime.jsx("button", {
    className: classes,
    disabled: disabled,
    onClick: onClick,
    children: index + 1
  }, index);
};
const CDSTablePagination = ({
  table,
  className = undefined,
  paginationConfig
}) => {
  var _a;
  const pages = table.getPageCount();
  const currentPage = table.getState().pagination.pageIndex;
  const computedPageSizeOptions = ((_a = paginationConfig.pageSizeOptions) === null || _a === void 0 ? void 0 : _a.map(value => {
    return {
      value: value.toString(),
      label: value.toString(),
      selected: value === table.getState().pagination.pageSize
    };
  })) || [];
  return jsxRuntime.jsxs(cdsReactFlex.CDSFlex, {
    gap: 4,
    margin: [16, 0],
    justify: "flex-end",
    className: className,
    children: [paginationConfig.showPageSizeChanger && computedPageSizeOptions.length > 0 ? jsxRuntime.jsxs(cdsReactFlex.CDSFlex, {
      gap: 8,
      justify: "flex-end",
      children: [jsxRuntime.jsx("span", {
        className: cdsComponentUtilities.cx(tablePagination_module["default"], 'cds-table__pagination__page-size__label'),
        children: "Rows per page"
      }), jsxRuntime.jsx(cdsReactSelect.CDSSelect, {
        style: {
          width: 'fit-content'
        },
        options: computedPageSizeOptions,
        value: `${table.getState().pagination.pageSize}`,
        onChange: option => {
          table.setPageSize(Number(option.value));
        }
      })]
    }) : null, jsxRuntime.jsx("button", {
      className: cdsComponentUtilities.cx(tablePagination_module["default"], 'cds-table__pagination__button', 'cds-table__pagination__button-arrow', 'cds-table__pagination__item'),
      disabled: currentPage === 0 || pages === 0,
      "data-testid": "cds-table-pagination-prev",
      onClick: () => table.setPageIndex(currentPage - 1),
      children: jsxRuntime.jsx(phosphorReact.CaretLeft, {
        size: 18,
        weight: "bold"
      })
    }), currentPage > 1 && jsxRuntime.jsx(CDSTablePaginationButton, {
      index: 0,
      onClick: () => table.setPageIndex(0)
    }), currentPage > 2 && jsxRuntime.jsx("div", {
      className: cdsComponentUtilities.cx(tablePagination_module["default"], 'cds-table__pagination__item'),
      children: "..."
    }), currentPage !== 0 && jsxRuntime.jsx(CDSTablePaginationButton, {
      index: currentPage - 1,
      onClick: () => table.setPageIndex(currentPage - 1)
    }), jsxRuntime.jsx(CDSTablePaginationButton, {
      index: currentPage,
      active: true
    }), pages > 0 && currentPage !== pages - 1 && jsxRuntime.jsx(CDSTablePaginationButton, {
      index: currentPage + 1,
      onClick: () => table.setPageIndex(currentPage + 1)
    }), currentPage < pages - 3 && jsxRuntime.jsx("div", {
      className: cdsComponentUtilities.cx(tablePagination_module["default"], 'cds-table__pagination__item'),
      children: "..."
    }), currentPage < pages - 2 && jsxRuntime.jsx(CDSTablePaginationButton, {
      index: pages - 1,
      onClick: () => table.setPageIndex(pages - 1)
    }), jsxRuntime.jsx("button", {
      className: cdsComponentUtilities.cx(tablePagination_module["default"], 'cds-table__pagination__button', 'cds-table__pagination__button-arrow', 'cds-table__pagination__item'),
      "data-testid": "cds-table-pagination-next",
      disabled: currentPage + 1 === pages || pages === 0,
      onClick: () => table.setPageIndex(currentPage + 1),
      children: jsxRuntime.jsx(phosphorReact.CaretRight, {
        size: 18,
        weight: "bold"
      })
    })]
  });
};

exports.CDSTablePagination = CDSTablePagination;
