// @ciscodesignsystems/cds-react-container/injected/container.esm.js
import { forwardRef } from 'react';
import { handleMarginInput } from '@ciscodesignsystems/cds-component-types';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './container.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const CDSContainer = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    background,
    children,
    className,
    margin,
    style,
    ...rest
  } = props;
  const marginDataAttr = handleMarginInput(margin);
  const classNames = cx(styles, 'cds-container', `cds-container__margin-top--${marginDataAttr.top}`, `cds-container__margin-bottom--${marginDataAttr.bottom}`, `cds-container__margin-right--${marginDataAttr.right}`, `cds-container__margin-left--${marginDataAttr.left}`, className);
  const computedBackground = () => {
    if (background) {
      const {
        image: backgroundImage,
        position: backgroundPosition = 'center',
        repeat: backgroundRepeat = 'no-repeat',
        size: backgroundSize = 'cover'
      } = {
        ...(typeof background === 'string' ? {
          image: background
        } : background)
      };
      return {
        backgroundImage,
        backgroundPosition,
        backgroundRepeat,
        backgroundSize
      };
    }
    return {};
  };
  return /*#__PURE__*/jsx("div", {
    className: classNames,
    ref: ref,
    style: {
      ...style,
      ...computedBackground()
    },
    ...rest,
    children: children
  });
});
CDSContainer.displayName = 'CDSContainer';
CDSContainer.defaultProps = {
  background: undefined,
  margin: 0
};

export { CDSContainer, CDSContainer as default };
