// @ciscodesignsystems/cds-react-drawer/components/heading/index.esm.js
import { X } from 'phosphor-react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { CDSFlex } from '@ciscodesignsystems/cds-react-flex';
import { useDrawerContext } from '../../util/context/index.esm.js';
import styles from '../../drawer.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';

const CDSDrawerHeading = ({
  children,
  className = '',
  ...rest
}) => {
  const {
    onClose
  } = useDrawerContext();
  return /*#__PURE__*/jsxs(CDSFlex, {
    align: "center",
    className: cx(styles, `cds-drawer__header ${className}`),
    ...rest,
    children: [/*#__PURE__*/jsx("h2", {
      className: cx(styles, 'cds-drawer__header__content'),
      children: children
    }), onClose && /*#__PURE__*/jsx("button", {
      "aria-label": "Close Drawer",
      className: cx(styles, 'cds-drawer__header__close-button'),
      type: "button",
      onClick: onClose,
      children: /*#__PURE__*/jsx(X, {
        size: 20
      })
    })]
  });
};
CDSDrawerHeading.displayName = 'CDSDrawer.Heading';

export { CDSDrawerHeading };
