// @ciscodesignsystems/cds-react-drawer/injected/components/content/index.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var drawer_module = require('../../drawer.module.cjs.js');

const CDSDrawerContent = ({
  children,
  className,
  ...rest
}) => {
  const contentRef = react.useRef(null);
  const [scrolled, setScrolled] = react.useState(false);
  const computedClassName = cdsComponentUtilities.cx(drawer_module["default"], 'cds-drawer__content', {
    'cds-drawer__content--scrolled': scrolled
  }, className);
  react.useEffect(() => {
    const contentTarget = contentRef.current;
    const options = {
      root: document.querySelector('.cds-drawer__content'),
      threshold: 1.0
    };
    const observer = new IntersectionObserver(([entry]) => {
      setScrolled(entry && !entry.isIntersecting);
    }, options);
    contentTarget && observer.observe(contentTarget);
    return () => {
      contentTarget && observer.unobserve(contentTarget);
    };
  }, [contentRef]);
  return jsxRuntime.jsxs("div", {
    className: computedClassName,
    ...rest,
    children: [jsxRuntime.jsx("span", {
      ref: contentRef,
      "aria-hidden": true
    }), children]
  });
};
CDSDrawerContent.displayName = 'CDSDrawer.Content';

exports.CDSDrawerContent = CDSDrawerContent;
