// @ciscodesignsystems/cds-react-header/injected/components/tenant-switcher/tenant-switcher.esm.js
import { forwardRef } from 'react';
import { Users, User, CaretDown } from 'phosphor-react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './tenant-switcher.module.esm.js';
import { jsx, jsxs } from 'react/jsx-runtime';

/* eslint-disable @typescript-eslint/no-explicit-any */
const CDSTenantSwitcher = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    as,
    sentiment,
    switcherType,
    userName,
    tenantName,
    width,
    toggleCaret,
    ...rest
  } = props;
  const Component = as || 'button';
  return /*#__PURE__*/jsx(Component, {
    "data-cds-tenant-switcher": true,
    "data-cds-sentiment": sentiment,
    "aria-label": "Tenant switcher menu",
    ref: ref,
    ...rest,
    children: /*#__PURE__*/jsxs("div", {
      className: cx(styles, 'tenant-switcher__content'),
      children: [switcherType === 'tenant' ? /*#__PURE__*/jsx(Users, {
        size: "24px",
        weight: "bold",
        style: {
          flexShrink: '0'
        }
      }) : /*#__PURE__*/jsx(User, {
        size: "24px",
        weight: "bold",
        style: {
          flexShrink: '0'
        }
      }), (userName || tenantName) && /*#__PURE__*/jsxs("div", {
        style: {
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'flex-start',
          padding: '0 1em',
          flex: 1,
          whiteSpace: 'nowrap',
          width: width
        },
        children: [(switcherType === 'combo' || switcherType === 'user') && /*#__PURE__*/jsx("span", {
          style: {
            textOverflow: 'ellipsis',
            overflow: 'hidden',
            width: '100%'
          },
          children: userName
        }), (switcherType === 'combo' || switcherType === 'tenant') && /*#__PURE__*/jsx("span", {
          style: {
            fontWeight: tenantName && userName ? 'normal' : 'inherit',
            textOverflow: 'ellipsis',
            overflow: 'hidden',
            width: '100%'
          },
          children: tenantName
        })]
      }), (userName || tenantName) && /*#__PURE__*/jsx(CaretDown, {
        size: "16px",
        weight: "bold",
        style: {
          flexShrink: '0'
        },
        className: toggleCaret ? cx(styles, 'tenant-switcher-caret') : ''
      })]
    })
  });
});
CDSTenantSwitcher.displayName = 'CDSTenantSwitcher';
CDSTenantSwitcher.defaultProps = {
  as: undefined,
  switcherType: 'combo',
  sentiment: 'brand',
  width: '120px',
  toggleCaret: true,
  userName: undefined,
  tenantName: undefined
};

export { CDSTenantSwitcher, CDSTenantSwitcher as default };
