// @ciscodesignsystems/cds-react-dropdown/injected/util/hooks/use-menu-interactions.esm.js
import { useFloating, offset, flip, shift, autoUpdate, useHover, safePolygon, useClick, useRole, useDismiss, useListNavigation, useInteractions } from '@floating-ui/react';
import { usePlacement } from './use-placement.esm.js';

const useMenuInteractions = ({
  activeIndex,
  allowHover,
  isNested,
  isOpen,
  listItemsRef,
  nodeId,
  setActiveIndex,
  setIsOpen
}) => {
  const placement = usePlacement(isNested);
  const {
    context,
    refs,
    strategy,
    x,
    y
  } = useFloating({
    nodeId,
    open: isOpen,
    onOpenChange: setIsOpen,
    strategy: 'fixed',
    placement,
    middleware: [offset({
      mainAxis: 4
    }), flip(), shift()],
    whileElementsMounted: autoUpdate
  });
  const hover = useHover(context, {
    enabled: isNested && allowHover,
    delay: {
      open: 75
    },
    handleClose: safePolygon({
      blockPointerEvents: true
    })
  });
  const click = useClick(context, {
    event: 'mousedown',
    toggle: !isNested || !allowHover,
    ignoreMouse: isNested
  });
  const role = useRole(context, {
    role: 'menu'
  });
  const dismiss = useDismiss(context);
  const listNavigation = useListNavigation(context, {
    listRef: listItemsRef,
    activeIndex,
    nested: isNested,
    onNavigate: setActiveIndex
  });
  const {
    getFloatingProps,
    getItemProps,
    getReferenceProps
  } = useInteractions([hover, click, role, dismiss, listNavigation]);
  return {
    context,
    getFloatingProps,
    getItemProps,
    getReferenceProps,
    refs,
    strategy,
    x,
    y
  };
};

export { useMenuInteractions };
