// @ciscodesignsystems/cds-react-tag/components/binary/index.esm.js
import { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { IcActive, IcInactive } from '@ciscodesignsystems/cds-react-icons';
import { CDSTag } from '../default-tag/index.esm.js';
import styles from './binary.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const labelMap = {
  active: 'Active',
  inactive: 'Inactive'
};
const CDSTagBinary = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    active,
    className,
    label,
    ...rest
  } = props;
  const classes = cx(styles, 'cds-tag--binary', `cds-tag--binary--${active ? 'active' : 'inactive'}`, className);
  return /*#__PURE__*/jsx(CDSTag, {
    ...rest,
    ref: ref,
    className: classes,
    icon: active ? /*#__PURE__*/jsx(IcActive, {}) : /*#__PURE__*/jsx(IcInactive, {}),
    children: label ?? labelMap[active ? 'active' : 'inactive']
  });
});
CDSTagBinary.displayName = 'CDSTag.Binary';
CDSTagBinary.defaultProps = {
  active: true,
  className: undefined,
  label: undefined,
  size: 'md'
};

export { CDSTagBinary };
