// @ciscodesignsystems/cds-react-flex/injected/flex.esm.js
import { forwardRef } from 'react';
import { handleMarginInput, spacingSizes } from '@ciscodesignsystems/cds-component-types';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './flex.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const CDSFlex = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    as,
    className,
    align,
    direction,
    gap,
    grow,
    justify,
    margin,
    reverse,
    wrap,
    children,
    style,
    ...rest
  } = props;
  const Component = as || 'div';
  const marginDataAttr = handleMarginInput(margin);
  const classes = cx(styles, 'cds-flex', {
    'cds-flex--grow': grow
  }, `cds-margin-top--${marginDataAttr.top}`, `cds-margin-bottom--${marginDataAttr.bottom}`, `cds-margin-right--${marginDataAttr.right}`, `cds-margin-left--${marginDataAttr.left}`, className);
  return /*#__PURE__*/jsx(Component, {
    className: classes,
    ref: ref,
    style: {
      ...style,
      alignItems: align,
      display: 'flex',
      flexDirection: `${direction === 'horizontal' ? 'row' : 'column'}${reverse ? '-reverse' : ''}`,
      flexWrap: wrap ? 'wrap' : 'nowrap',
      gap: typeof gap === 'string' ? spacingSizes[gap] : gap,
      justifyContent: justify
    },
    ...rest,
    children: children
  });
});
CDSFlex.displayName = 'CDSFlex';
CDSFlex.defaultProps = {
  align: 'stretch',
  direction: 'horizontal',
  gap: 0,
  grow: false,
  justify: 'flex-start',
  reverse: false,
  wrap: false,
  margin: 0
};

export { CDSFlex, CDSFlex as default };
