// @ciscodesignsystems/cds-component-types/index.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var sentimentThemeTypes = require('./lib/sentiment-theme-types.cjs.js');
var spacingTypes = require('./lib/basic-types/spacing-types.cjs.js');
var statusTypes = require('./lib/basic-types/status-types.cjs.js');
var conditionalWrap = require('./lib/conditional-wrap.cjs.js');
var marginHelper = require('./util/margin-helper/margin-helper.cjs.js');
var useKeyboard = require('./util/use-keyboard.cjs.js');
var useOutsideClick = require('./util/use-outside-click.cjs.js');
var useMergeRef = require('./util/use-merge-ref.cjs.js');



exports.AllSentimentArray = sentimentThemeTypes.AllSentimentArray;
exports.BrandSentimentArray = sentimentThemeTypes.BrandSentimentArray;
exports.CategoricalSentimentArray = sentimentThemeTypes.CategoricalSentimentArray;
exports.InteractionSentimentArray = sentimentThemeTypes.InteractionSentimentArray;
exports.SentimentArray = sentimentThemeTypes.SentimentArray;
exports.SentimentGradientArray = sentimentThemeTypes.SentimentGradientArray;
exports.SentimentInteractArray = sentimentThemeTypes.SentimentInteractArray;
exports.SentimentProminenceArray = sentimentThemeTypes.SentimentProminenceArray;
exports.SentimentUsageArray = sentimentThemeTypes.SentimentUsageArray;
exports.StatusSentimentArray = sentimentThemeTypes.StatusSentimentArray;
exports.spacingSizes = spacingTypes.spacingSizes;
exports.CDSStatusArray = statusTypes.CDSStatusArray;
exports.CDSStatusIconArray = statusTypes.CDSStatusIconArray;
exports.ConditionalWrap = conditionalWrap.ConditionalWrap;
exports.handleMarginInput = marginHelper.handleMarginInput;
exports.useKeyboard = useKeyboard.useKeyboard;
exports.useOutsideClick = useOutsideClick.useOutsideClick;
exports.useMergeRefs = useMergeRef.useMergeRefs;
