// @ciscodesignsystems/cds-react-header/components/tenant-switcher/tenant-switcher.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var phosphorReact = require('phosphor-react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var tenantSwitcher_module = require('./tenant-switcher.module.cjs.js');

const CDSTenantSwitcher = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    as,
    sentiment,
    switcherType,
    userName,
    tenantName,
    width,
    toggleCaret,
    ...rest
  } = props;
  const Component = as || 'button';
  return jsxRuntime.jsx(Component, {
    "data-cds-tenant-switcher": true,
    "data-cds-sentiment": sentiment,
    "aria-label": "Tenant switcher menu",
    ref: ref,
    ...rest,
    children: jsxRuntime.jsxs("div", {
      className: cdsComponentUtilities.cx(tenantSwitcher_module["default"], 'tenant-switcher__content'),
      children: [switcherType === 'tenant' ? jsxRuntime.jsx(phosphorReact.Users, {
        size: "24px",
        weight: "bold",
        style: {
          flexShrink: '0'
        }
      }) : jsxRuntime.jsx(phosphorReact.User, {
        size: "24px",
        weight: "bold",
        style: {
          flexShrink: '0'
        }
      }), (userName || tenantName) && jsxRuntime.jsxs("div", {
        style: {
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'flex-start',
          padding: '0 1em',
          flex: 1,
          whiteSpace: 'nowrap',
          width: width
        },
        children: [(switcherType === 'combo' || switcherType === 'user') && jsxRuntime.jsx("span", {
          style: {
            textOverflow: 'ellipsis',
            overflow: 'hidden',
            width: '100%'
          },
          children: userName
        }), (switcherType === 'combo' || switcherType === 'tenant') && jsxRuntime.jsx("span", {
          style: {
            fontWeight: tenantName && userName ? 'normal' : 'inherit',
            textOverflow: 'ellipsis',
            overflow: 'hidden',
            width: '100%'
          },
          children: tenantName
        })]
      }), (userName || tenantName) && jsxRuntime.jsx(phosphorReact.CaretDown, {
        size: "16px",
        weight: "bold",
        style: {
          flexShrink: '0'
        },
        className: toggleCaret ? cdsComponentUtilities.cx(tenantSwitcher_module["default"], 'tenant-switcher-caret') : ''
      })]
    })
  });
});
CDSTenantSwitcher.displayName = 'CDSTenantSwitcher';
CDSTenantSwitcher.defaultProps = {
  as: undefined,
  switcherType: 'combo',
  sentiment: 'brand',
  width: '120px',
  toggleCaret: true,
  userName: undefined,
  tenantName: undefined
};

exports.CDSTenantSwitcher = CDSTenantSwitcher;
exports["default"] = CDSTenantSwitcher;
