// @ciscodesignsystems/cds-react-text-input/components/suffix/index.esm.js
import { XCircle } from 'phosphor-react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import PasswordSuffix from './password-suffix.esm.js';
import styles from './suffix.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';

const SuffixComp = ({
  type,
  suffix,
  showPassword = undefined,
  setShowPassword = undefined,
  onClear,
  disabled,
  hasValue,
  showPasswordText,
  hidePasswordText
}) => {
  const togglePassword = () => {
    setShowPassword && setShowPassword(prevState => !prevState);
  };
  return /*#__PURE__*/jsxs("div", {
    className: cx(styles, 'cds-text-input__suffix'),
    children: [type === 'password' && showPassword !== undefined && /*#__PURE__*/jsx(PasswordSuffix, {
      showPassword: showPassword,
      togglePassword: togglePassword,
      showPasswordText: showPasswordText,
      hidePasswordText: hidePasswordText
    }), suffix, onClear && hasValue && !disabled && /*#__PURE__*/jsx(XCircle, {
      "aria-hidden": true,
      size: 16,
      onClick: onClear,
      className: cx(styles, 'cds-text-input__clear-icon')
    })]
  });
};
var SuffixComp$1 = SuffixComp;

export { SuffixComp$1 as default };
