// @ciscodesignsystems/cds-react-table/injected/components/table-cell/table-cell-with-expanded-indicator.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var phosphorReact = require('phosphor-react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var injected = require('@ciscodesignsystems/cds-react-button/injected');
var tableCellWithExpandedIndicator_module = require('./table-cell-with-expanded-indicator.module.cjs.js');

const TableCellWithExpandedIndication = ({
  row
}) => {
  const className = cdsComponentUtilities.cx(tableCellWithExpandedIndicator_module["default"], 'cds-table__row--expandable__trigger', {
    'cds-table__row--expandable__trigger--expanded': row.getIsExpanded()
  });
  return row.getCanExpand() && jsxRuntime.jsx(injected.CDSButton, {
    className: className,
    variant: "tertiary",
    onClick: () => row.toggleExpanded(),
    children: jsxRuntime.jsx(phosphorReact.CaretRight, {
      size: 18,
      weight: "bold"
    })
  });
};

exports.TableCellWithExpandedIndication = TableCellWithExpandedIndication;
