// @ciscodesignsystems/cds-react-logos/cisco/index.esm.js
import { CDSLogo } from '../logos.esm.js';
import { jsx } from 'react/jsx-runtime';

const CDSLogoCisco = props => {
  return /*#__PURE__*/jsx(CDSLogo, {
    ...props,
    children: /*#__PURE__*/jsx("svg", {
      version: "1.1",
      xmlns: "http://www.w3.org/2000/svg",
      x: "0px",
      y: "0px",
      viewBox: "0 0 48 24",
      style: {
        flexShrink: 0
      },
      preserveAspectRatio: "xMidYMid meet",
      children: /*#__PURE__*/jsx("g", {
        children: /*#__PURE__*/jsx("path", {
          className: "cds-logo-fill-primary",
          d: "M15.8 15.3h1.6V22h-1.6zM25.8 18.6c0 2.1 1.6 3.4 3.5 3.4.7 0 1.3-.2 1.5-.2V20c-.1 0-.6.3-1.4.3-1.1 0-1.8-.8-1.8-1.8s.7-1.8 1.8-1.8c.8 0 1.3.3 1.4.4v-1.8c-.2 0-.7-.2-1.5-.2-2 .1-3.5 1.5-3.5 3.5zM8.6 18.6c0 2.1 1.6 3.4 3.5 3.4.7 0 1.3-.2 1.5-.2V20c-.1 0-.6.3-1.4.3-1.1 0-1.8-.8-1.8-1.8s.7-1.8 1.8-1.8c.8 0 1.3.3 1.4.4v-1.8c-.2 0-.7-.2-1.5-.2-2 .1-3.5 1.5-3.5 3.5zM36 15.2c-2 0-3.5 1.5-3.5 3.4S33.9 22 36 22c2 0 3.5-1.5 3.5-3.4-.1-1.9-1.5-3.4-3.5-3.4zm0 5.1c-1 0-1.7-.8-1.7-1.7 0-1 .7-1.7 1.7-1.7s1.7.8 1.7 1.7c0 .9-.8 1.7-1.7 1.7zM22.6 17.9l-.5-.1c-.3-.1-.8-.2-.8-.6 0-.3.4-.5 1-.5s1.3.2 1.3.2v-1.4s-.8-.2-1.6-.2c-1.5 0-2.4.8-2.4 2.1 0 1.1.8 1.6 1.7 1.9.1 0 .3.1.4.1.4.1.7.3.7.7 0 .4-.4.6-1.2.6-.7 0-1.4-.2-1.6-.2V22c.1 0 .9.2 1.8.2 1.3 0 2.7-.6 2.7-2.2.1-1-.4-1.8-1.5-2.1zM6 7.4c-.5 0-.8.4-.8.8v1.7c0 .5.4.8.8.8.5 0 .8-.4.8-.8V8.2c0-.5-.3-.8-.8-.8zM10.5 10.8c.5 0 .8-.4.8-.8V6c0-.5-.4-.8-.8-.8-.5 0-.8.4-.8.8v4c0 .4.4.8.8.8zM15 2c-.5 0-.8.4-.8.8v8.8c0 .5.4.8.8.8.5 0 .8-.4.8-.8V2.8c0-.4-.3-.8-.8-.8zM20.3 9.9v-4c0-.5-.4-.8-.8-.8s-.8.4-.8.8v4c0 .5.4.8.8.8.5.1.8-.3.8-.8zM24 7.4c-.5 0-.8.4-.8.8v1.7c0 .5.4.8.8.8s.8-.4.8-.8V8.2c0-.5-.3-.8-.8-.8zM28.5 10.8c.4 0 .8-.4.8-.8V6c0-.5-.4-.8-.8-.8-.5 0-.8.4-.8.8v4c0 .4.3.8.8.8zM33.8 11.6V2.8c0-.4-.4-.8-.8-.8-.5 0-.8.4-.8.8v8.8c0 .5.4.8.8.8s.8-.4.8-.8zM37.5 10.8c.5 0 .8-.4.8-.8V6c0-.5-.4-.8-.8-.8-.5 0-.8.4-.8.8v4c0 .4.3.8.8.8zM42 7.4c-.5 0-.8.4-.8.8v1.7c0 .5.4.8.8.8s.8-.4.8-.8V8.2c0-.5-.4-.8-.8-.8z"
        })
      })
    })
  });
};

export { CDSLogoCisco };
