// @ciscodesignsystems/cds-component-utilities/data-viz/theme.esm.js
import { mergeDeepRight } from 'ramda';

const getDefaultChartTheme = fontSize => ({
  background: 'var(--base-bg-weak-default)',
  fontFamily: '',
  textColor: 'var(--base-text-default)',
  fontSize: 12,
  margin: {
    top: fontSize,
    right: 0,
    bottom: fontSize * 2,
    left: fontSize * 4
  },
  colors: datum => 'var(--base-text-weak-default)',
  pointSize: 0,
  axis: {
    domain: {
      line: {
        stroke: 'var(--base-text-weak-default)',
        strokeWidth: 0
      }
    },
    ticks: {
      line: {
        stroke: 'var(--base-text-weak-default)',
        strokeWidth: 0
      },
      text: {
        fill: 'var(--base-text-weak-default)',
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        //@ts-ignore required because nivo's own types are wrong
        fontSize: fontSize ?? 14
      }
    }
  },
  crosshair: {
    line: {
      stroke: 'var(--base-border-strong-default)',
      strokeWidth: 2,
      strokeDasharray: '0'
    }
  },
  grid: {
    line: {
      stroke: 'var(--base-border-medium-default)',
      strokeWidth: 1,
      strokeDasharray: '3 3'
    }
  }
});
const mergeTheme = (fontSize, theme) => theme ? mergeDeepRight({
  ...getDefaultChartTheme(fontSize),
  fontSize
}, theme) : getDefaultChartTheme(fontSize);

export { getDefaultChartTheme, mergeTheme };
