// @ciscodesignsystems/cds-react-header/components/help-menu/help-menu.esm.js
import { forwardRef } from 'react';
import { Question, CheckCircle } from 'phosphor-react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { CDSButton } from '@ciscodesignsystems/cds-react-button';
import styles from './help-menu.module.esm.js';
import { jsx, jsxs } from 'react/jsx-runtime';
import { CDSHeaderMenuRoot, CDSHeaderMenuContent, CDSHeaderMenuRow, CDSHeaderMenuLink } from '../header-menu/header-menu.esm.js';

const CDSHelpMenuRoot = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    sentiment,
    variant,
    contentRef,
    triggerRef
  } = props;
  return /*#__PURE__*/jsx(CDSHeaderMenuRoot, {
    ref: ref,
    id: "headerHelpMenu",
    "aria-label": "Help menu",
    trigger: /*#__PURE__*/jsx(CDSButton, {
      ref: triggerRef,
      "aria-label": "Help content",
      sentiment: sentiment,
      icon: /*#__PURE__*/jsx(Question, {
        weight: "bold",
        size: "24px"
      }),
      variant: variant
    }),
    children: /*#__PURE__*/jsx(CDSHeaderMenuContent, {
      className: cx(styles, 'cds-help-menu'),
      align: "left",
      ref: contentRef,
      children: children
    })
  });
});
CDSHelpMenuRoot.displayName = 'CDSHelpMenuRoot';
const CDSHelpMenuSection = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    description,
    children,
    className,
    ...rest
  } = props;
  return /*#__PURE__*/jsxs(CDSHeaderMenuRow, {
    ref: ref,
    className: cx(styles, 'cds-help-menu__section', className),
    ...rest,
    children: [description && /*#__PURE__*/jsx("span", {
      className: cx(styles, 'cds-help-menu__section__description'),
      children: description
    }), /*#__PURE__*/jsx("ul", {
      className: cx(styles, 'cds-help-menu__section__list'),
      children: children
    })]
  });
});
CDSHelpMenuSection.displayName = 'CDSHelpMenuSection';
CDSHelpMenuSection.defaultProps = {
  description: undefined
};
const CDSHelpMenuLink = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    interact,
    className,
    ...rest
  } = props;
  const isDisabled = interact === 'disabled';
  return /*#__PURE__*/jsxs(CDSHeaderMenuRow, {
    className: cx(styles, 'cds-help-menu__section__row'),
    "data-cds-interact": interact,
    children: [/*#__PURE__*/jsx(CDSHeaderMenuLink, {
      className: cx(styles, 'cds-help-menu__section__link', className),
      ref: ref,
      tabIndex: isDisabled ? -1 : undefined,
      "data-cds-interact": interact,
      ...rest,
      children: children
    }), interact === 'active' && /*#__PURE__*/jsx(CheckCircle, {
      color: "#4580E5",
      weight: "fill",
      size: 23
    })]
  });
});
CDSHelpMenuLink.displayName = 'CDSHelpMenuLink';
CDSHelpMenuLink.defaultProps = {
  interact: undefined
};
CDSHelpMenuRoot.defaultProps = {
  sentiment: 'inverse',
  variant: 'ghost',
  align: undefined,
  contentRef: undefined,
  triggerRef: undefined
};

export { CDSHelpMenuLink, CDSHelpMenuRoot, CDSHelpMenuSection, CDSHelpMenuRoot as default };
