// @ciscodesignsystems/cds-react-header/components/menu/menu.esm.js
import { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './menu.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { CDSMenuItem } from './components/menu-item/menu-item.esm.js';

const CDSMenu = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    width,
    trigger,
    align,
    children,
    showMenuContent,
    className,
    ...rest
  } = props;
  const classes = cx(styles, 'cds-menu', `cds-menu--${align}`, className);
  return /*#__PURE__*/jsxs("div", {
    className: classes,
    ref: ref,
    "data-aria-orientation": "horizontal",
    dir: "ltr",
    ...rest,
    children: [/*#__PURE__*/jsx("div", {
      className: cx(styles, 'cds-menu__trigger'),
      children: trigger
    }), showMenuContent ? /*#__PURE__*/jsx("ul", {
      style: {
        width
      },
      className: cx(styles, 'cds-menu__content'),
      "data-aria-orientation": "horizontal",
      dir: "ltr",
      children: children
    }) : null]
  });
});
CDSMenu.displayName = 'CDSMenu';
CDSMenu.defaultProps = {
  align: 'left',
  showMenuContent: false,
  width: '397px'
};
const CDSMenuNamespace = Object.assign(CDSMenu, {
  /**
   * The `CDSMenu.Item` represents the portion of `CDSMenu` that
   * will be rendered when menu content is shown.
   */
  Item: CDSMenuItem
});
CDSMenuNamespace.displayName = 'CDSMenu';

export { CDSMenuNamespace as CDSMenu, CDSMenuNamespace as default };
