// @ciscodesignsystems/cds-react-header/components/menu/menu.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var menuItem = require('./components/menu-item/menu-item.cjs.js');
var menu_module = require('./menu.module.cjs.js');

const CDSMenu = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    width,
    trigger,
    align,
    children,
    showMenuContent,
    className,
    ...rest
  } = props;
  const classes = cdsComponentUtilities.cx(menu_module["default"], 'cds-menu', `cds-menu--${align}`, className);
  return jsxRuntime.jsxs("div", {
    className: classes,
    ref: ref,
    "data-aria-orientation": "horizontal",
    dir: "ltr",
    ...rest,
    children: [jsxRuntime.jsx("div", {
      className: cdsComponentUtilities.cx(menu_module["default"], 'cds-menu__trigger'),
      children: trigger
    }), showMenuContent ? jsxRuntime.jsx("ul", {
      style: {
        width
      },
      className: cdsComponentUtilities.cx(menu_module["default"], 'cds-menu__content'),
      "data-aria-orientation": "horizontal",
      dir: "ltr",
      children: children
    }) : null]
  });
});
CDSMenu.displayName = 'CDSMenu';
CDSMenu.defaultProps = {
  align: 'left',
  showMenuContent: false,
  width: '397px'
};
const CDSMenuNamespace = Object.assign(CDSMenu, {
  /**
   * The `CDSMenu.Item` represents the portion of `CDSMenu` that
   * will be rendered when menu content is shown.
   */
  Item: menuItem.CDSMenuItem
});
CDSMenuNamespace.displayName = 'CDSMenu';

exports.CDSMenu = CDSMenuNamespace;
exports["default"] = CDSMenuNamespace;
