// @ciscodesignsystems/cds-react-dropdown/components/item/index.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var react$1 = require('@floating-ui/react');
var cdsComponentTypes = require('@ciscodesignsystems/cds-component-types');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var panelContext = require('../../util/panel-context.cjs.js');
var item_module = require('./item.module.cjs.js');

const CDSDropdownItem = /*#__PURE__*/react.forwardRef(({
  destructive = false,
  icon,
  label,
  className,
  onClick: onClickFromProps,
  onMouseEnter: onMouseEnterFromProps,
  ...rest
}, ref) => {
  const {
    ref: listItemRef,
    index
  } = react$1.useListItem();
  const mergeRefs = cdsComponentTypes.useMergeRefs([ref, listItemRef]);
  const classNames = cdsComponentUtilities.cx(item_module["default"], 'cds-dropdown__item', {
    'cds-dropdown__item--destructive': destructive
  }, className);
  const {
    tree,
    activeIndex,
    setActiveIndex,
    isOpen,
    allowHover,
    getItemProps
  } = panelContext.useDropdownPanelContext();
  return jsxRuntime.jsxs("button", {
    ...rest,
    className: classNames,
    ref: mergeRefs,
    role: "menuitem",
    type: "button",
    tabIndex: activeIndex === index ? 0 : -1,
    ...getItemProps({
      onClick(e) {
        onClickFromProps === null || onClickFromProps === void 0 ? void 0 : onClickFromProps(e);
        if (!rest['aria-haspopup']) {
          tree === null || tree === void 0 ? void 0 : tree.events.emit('click');
        }
      },
      onMouseEnter(e) {
        onMouseEnterFromProps === null || onMouseEnterFromProps === void 0 ? void 0 : onMouseEnterFromProps(e);
        if (allowHover && isOpen) {
          setActiveIndex(index);
        }
      }
    }),
    children: [icon, label]
  });
});
CDSDropdownItem.displayName = 'CDSDropdown.Item';
CDSDropdownItem.defaultProps = {
  icon: undefined,
  destructive: false
};

exports.CDSDropdownItem = CDSDropdownItem;
exports["default"] = CDSDropdownItem;
