// @ciscodesignsystems/cds-react-header/injected/components/menu/components/menu-item/menu-item.esm.js
import React, { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import styles from './menu-item.module.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';

const getTitle = node => {
  if (typeof node === 'string') {
    return node;
  }
  if (Array.isArray(node)) {
    return node.map(child => getTitle(child)).join('');
  }
  if ( /*#__PURE__*/React.isValidElement(node) && node.props && node.props.children) {
    return getTitle(node.props.children);
  }
  return '';
};
const CDSMenuItem = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    label,
    title,
    subtitle,
    children,
    icon,
    action,
    truncate,
    className,
    ...rest
  } = props;
  const classes = cx(styles, 'cds-menu__item', className);
  const contentClasses = cx(styles, 'cds-menu__item__content', {
    'cds-menu__item__content--truncated': truncate
  });
  return /*#__PURE__*/jsxs("li", {
    ref: ref,
    className: classes,
    ...rest,
    children: [icon ? /*#__PURE__*/jsx("div", {
      className: cx(styles, 'cds-menu__item__affix', 'cds-menu__item__affix--left'),
      children: icon
    }) : null, /*#__PURE__*/jsxs("div", {
      className: contentClasses,
      children: [label !== '' ? /*#__PURE__*/jsx("div", {
        className: cx(styles, 'cds-menu__item__content__label'),
        title: getTitle(label),
        children: label
      }) : null, /*#__PURE__*/jsx("div", {
        className: cx(styles, 'cds-menu__item__content__title'),
        title: getTitle(title),
        children: title
      }), subtitle !== '' ? /*#__PURE__*/jsx("div", {
        className: cx(styles, 'cds-menu__item__content__subtitle'),
        title: getTitle(subtitle),
        children: subtitle
      }) : null]
    }), /*#__PURE__*/jsx("div", {
      className: cx(styles, 'cds-menu__item__affix', 'cds-menu__item__affix--right'),
      children: action
    })]
  });
});
CDSMenuItem.displayName = 'CDSMenuItem';
CDSMenuItem.defaultProps = {
  icon: false,
  label: '',
  subtitle: '',
  truncate: false
};

export { CDSMenuItem, CDSMenuItem as default, getTitle };
