// @ciscodesignsystems/cds-magnetic-theme-web/js/dist/token-theme-classic-dark-hex8-es6.js
/**
 * This file is auto-generated. Do not edit it directly.
 */

export const BRAND_BG_DEFAULT = "#373c42ff";
export const BRAND_BG_HOVER = "#373c42ff";
export const BRAND_BG_FOCUS = "#464c54ff";
export const BRAND_BG_ACTIVE = "#464c54ff";
export const BRAND_BG_DISABLED = "#e1e4e8ff";
export const BRAND_BG_GRADIENT_START = "#373c42ff";
export const BRAND_BG_GRADIENT_END = "#0f1214ff";
export const BRAND_BG_STRONG_DEFAULT = "#373c42ff";
export const BRAND_BG_STRONG_HOVER = "#373c42ff";
export const BRAND_BG_STRONG_FOCUS = "#373c42ff";
export const BRAND_BG_STRONG_ACTIVE = "#373c42ff";
export const BRAND_TEXT_DEFAULT = "#f7f7f7ff";
export const BRAND_TEXT_HOVER = "#f7f7f7ff";
export const BRAND_TEXT_ACTIVE = "#f7f7f7ff";
export const BRAND_TEXT_WEAK_DEFAULT = "#889099ff";
export const BRAND_TEXT_WEAK_HOVER = "#f7f7f7ff";
export const BRAND_TEXT_WEAK_FOCUS = "#f7f7f7ff";
export const BRAND_TEXT_WEAK_ACTIVE = "#f7f7f7ff";
export const BRAND_TEXT_STRONG_DEFAULT = "#f0f1f2ff";
export const BRAND_ICON_DEFAULT = "#f7f7f7ff";
export const BRAND_ICON_HOVER = "#ffffffff";
export const BRAND_ICON_ACTIVE = "#ffffffff";
export const BRAND_ICON_WEAK_DEFAULT = "#f7f7f7ff";
export const BRAND_ICON_WEAK_HOVER = "#f7f7f7ff";
export const BRAND_ICON_WEAK_FOCUS = "#f7f7f7ff";
export const BRAND_ICON_WEAK_ACTIVE = "#f7f7f7ff";
export const BRAND_BORDER_ACTIVE = "#cce1ffff";
export const BRAND_ACCENT_ICON_DEFAULT = "#c1c6ccff";
export const BRAND_ACCENT_ICON_FOCUS = "#ffffffff";
export const BRAND_ACCENT_ICON_ACTIVE = "#6bbf41ff";
export const BRAND_ACCENT_ICON_DISABLED = "#6f7680ff";
export const BRAND_ACCENT_ICON_WEAK_DEFAULT = "#6f7680ff";
export const BRAND_ACCENT_ICON_WEAK_FOCUS = "#ffffffff";
export const BRAND_ACCENT_ICON_WEAK_ACTIVE = "#398519ff";
export const BRAND_ACCENT_ICON_WEAK_DISABLED = "#ffffffff";
export const BRAND_ACCENT_ICON_MEDIUM_DEFAULT = "#979fa8ff";
export const BRAND_ACCENT_ICON_MEDIUM_FOCUS = "#ffffffff";
export const BRAND_ACCENT_ICON_MEDIUM_ACTIVE = "#52a62bff";
export const BRAND_ACCENT_ICON_MEDIUM_DISABLED = "#ffffffff";
export const BRAND_ACCENT_BORDER_DEFAULT = "#52a62bff";
export const BRAND_ACCENT_BORDER_HOVER = "#7e868fff";
export const BRAND_ACCENT_BORDER_FOCUS = "#52a62bff";
export const BRAND_ACCENT_BORDER_ACTIVE = "#52a62bff";
export const BRAND_ACCENT_BORDER_WEAK_DEFAULT = "#d0d4d9ff";
export const BRAND_ACCENT_BORDER_WEAK_HOVER = "#ffffffff";
export const BRAND_ACCENT_BORDER_WEAK_FOCUS = "#ffffffff";
export const BRAND_ACCENT_BORDER_WEAK_ACTIVE = "#398519ff";
export const BRAND_ACCENT_BORDER_WEAK_DISABLED = "#ffffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_DEFAULT = "#596069ff";
export const BRAND_ACCENT_BORDER_MEDIUM_HOVER = "#ffffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_FOCUS = "#ffffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_ACTIVE = "#2b7a0cff";
export const BRAND_ACCENT_BORDER_MEDIUM_DISABLED = "#ffffffff";
export const BRAND_ACCENT_BORDER_STRONG_DEFAULT = "#0f1214ff";
export const BRAND_ACCENT_BORDER_STRONG_HOVER = "#ffffffff";
export const BRAND_ACCENT_BORDER_STRONG_FOCUS = "#ffffffff";
export const BRAND_ACCENT_BORDER_STRONG_ACTIVE = "#0f1214ff";
export const BRAND_ACCENT_BORDER_STRONG_DISABLED = "#ffffffff";
export const BASE_BG_DEFAULT = "#0f1214ff";
export const BASE_BG_HOVER = "#0f1214ff";
export const BASE_BG_ACTIVE = "#0f1214ff";
export const BASE_BG_DISABLED = "#464c54ff";
export const BASE_BG_WEAK_DEFAULT = "#23282eff";
export const BASE_BG_WEAK_HOVER = "#373c42ff";
export const BASE_BG_WEAK_ACTIVE = "#23282eff";
export const BASE_BG_WEAK_DISABLED = "#464c54ff";
export const BASE_BG_MEDIUM_DEFAULT = "#373c42ff";
export const BASE_BG_MEDIUM_HOVER = "#373c42ff";
export const BASE_BG_MEDIUM_ACTIVE = "#23282eff";
export const BASE_BG_MEDIUM_DISABLED = "#e85fc6ff";
export const BASE_BG_STRONG_DEFAULT = "#a7adb5ff";
export const BASE_BG_STRONG_HOVER = "#d0d4d9ff";
export const BASE_BG_STRONG_FOCUS = "#d0d4d9ff";
export const BASE_BG_STRONG_ACTIVE = "#d0d4d9ff";
export const BASE_BG_STRONG_DISABLED = "#596069ff";
export const BASE_ICON_IN_DEFAULT = "#23282eff";
export const BASE_ICON_IN_HOVER = "#23282eff";
export const BASE_ICON_IN_FOCUS = "#23282eff";
export const BASE_ICON_IN_ACTIVE = "#23282eff";
export const BASE_ICON_IN_DISABLED = "#6f7680ff";
export const BASE_TEXT_DEFAULT = "#f7f7f7ff";
export const BASE_TEXT_DISABLED = "#6f7680ff";
export const BASE_TEXT_WEAK_DEFAULT = "#889099ff";
export const BASE_TEXT_WEAK_DISABLED = "#6f7680ff";
export const BASE_TEXT_MEDIUM_DEFAULT = "#d0d4d9ff";
export const BASE_TEXT_MEDIUM_DISABLED = "#6f7680ff";
export const BASE_TEXT_STRONG_DEFAULT = "#f0f1f2ff";
export const BASE_TEXT_STRONG_DISABLED = "#6f7680ff";
export const BASE_TEXT_IN_DEFAULT = "#23282eff";
export const BASE_TEXT_IN_HOVER = "#23282eff";
export const BASE_TEXT_IN_FOCUS = "#23282eff";
export const BASE_TEXT_IN_ACTIVE = "#23282eff";
export const BASE_TEXT_IN_DISABLED = "#23282eff";
export const BASE_TEXT_IN_WEAK_DEFAULT = "#ffffffff";
export const BASE_BORDER_DEFAULT = "#596069ff";
export const BASE_BORDER_WEAK_DEFAULT = "#23282eff";
export const BASE_BORDER_MEDIUM_DEFAULT = "#464c54ff";
export const BASE_BORDER_STRONG_DEFAULT = "#c1c6ccff";
export const INVERSE_BG_DEFAULT = "#373c42ff";
export const INVERSE_BG_HOVER = "#373c42ff";
export const INVERSE_BG_FOCUS = "#464c54ff";
export const INVERSE_BG_ACTIVE = "#464c54ff";
export const INVERSE_BG_DISABLED = "#e1e4e8ff";
export const INVERSE_BG_GRADIENT_START = "#373c42ff";
export const INVERSE_BG_GRADIENT_END = "#0f1214ff";
export const INVERSE_BG_WEAK_DEFAULT = "#464c54ff";
export const INVERSE_TEXT_DEFAULT = "#f7f7f7ff";
export const INVERSE_TEXT_HOVER = "#f7f7f7ff";
export const INVERSE_TEXT_FOCUS = "#f7f7f7ff";
export const INVERSE_TEXT_ACTIVE = "#f7f7f7ff";
export const INVERSE_TEXT_DISABLED = "#656c75ff";
export const INVERSE_TEXT_MEDIUM_DEFAULT = "#c1c6ccff";
export const INVERSE_TEXT_MEDIUM_DISABLED = "#656c75ff";
export const INVERSE_ICON_DEFAULT = "#f7f7f7ff";
export const INVERSE_ICON_HOVER = "#ffffffff";
export const INVERSE_ICON_FOCUS = "#ffffffff";
export const INVERSE_ICON_ACTIVE = "#ffffffff";
export const INVERSE_ICON_DISABLED = "#6f7680ff";
export const INVERSE_ICON_STRONG_DEFAULT = "#ffffffff";
export const INVERSE_ICON_STRONG_HOVER = "#ffffffff";
export const INVERSE_ICON_STRONG_FOCUS = "#ffffffff";
export const INVERSE_ICON_STRONG_ACTIVE = "#ffffffff";
export const INVERSE_ICON_STRONG_DISABLED = "#6f7680ff";
export const INVERSE_BORDER_DEFAULT = "#6f7680ff";
export const INVERSE_BORDER_HOVER = "#649ef5ff";
export const INVERSE_BORDER_FOCUS = "#cce1ffff";
export const INVERSE_BORDER_ACTIVE = "#cce1ffff";
export const INVERSE_BORDER_DISABLED = "#e1e4e8ff";
export const CONTROL_BG_DEFAULT = "#23282eff";
export const CONTROL_BG_HOVER = "#2a3442ff";
export const CONTROL_BG_FOCUS = "#2a3442ff";
export const CONTROL_BG_ACTIVE = "#2a3442ff";
export const CONTROL_BG_DISABLED = "#373c42ff";
export const CONTROL_BG_WEAK_DEFAULT = "#23282eff";
export const CONTROL_BG_WEAK_HOVER = "#373c42ff";
export const CONTROL_BG_WEAK_FOCUS = "#23282eff";
export const CONTROL_BG_WEAK_ACTIVE = "#2a3442ff";
export const CONTROL_BG_WEAK_DISABLED = "#373c42ff";
export const CONTROL_BG_MEDIUM_DEFAULT = "#23282eff";
export const CONTROL_BG_MEDIUM_HOVER = "#373c42ff";
export const CONTROL_BG_MEDIUM_FOCUS = "#979fa8ff";
export const CONTROL_BG_MEDIUM_ACTIVE = "#2a3442ff";
export const CONTROL_BG_MEDIUM_DISABLED = "#373c42ff";
export const CONTROL_BG_STRONG_DEFAULT = "#649ef5ff";
export const CONTROL_BG_STRONG_HOVER = "#7cadf7ff";
export const CONTROL_BG_STRONG_FOCUS = "#7cadf7ff";
export const CONTROL_BG_STRONG_ACTIVE = "#7cadf7ff";
export const CONTROL_BG_STRONG_DISABLED = "#596069ff";
export const CONTROL_ICON_DEFAULT = "#649ef5ff";
export const CONTROL_ICON_HOVER = "#7cadf7ff";
export const CONTROL_ICON_FOCUS = "#7cadf7ff";
export const CONTROL_ICON_ACTIVE = "#7cadf7ff";
export const CONTROL_ICON_DISABLED = "#cce1ffff";
export const CONTROL_ICON_WEAK_DEFAULT = "#c1c6ccff";
export const CONTROL_ICON_WEAK_HOVER = "#649ef5ff";
export const CONTROL_ICON_WEAK_FOCUS = "#649ef5ff";
export const CONTROL_ICON_WEAK_ACTIVE = "#649ef5ff";
export const CONTROL_ICON_WEAK_DISABLED = "#6f7680ff";
export const CONTROL_ICON_MEDIUM_DEFAULT = "#649ef5ff";
export const CONTROL_ICON_MEDIUM_HOVER = "#979fa8ff";
export const CONTROL_ICON_MEDIUM_FOCUS = "#979fa8ff";
export const CONTROL_ICON_MEDIUM_ACTIVE = "#889099ff";
export const CONTROL_ICON_MEDIUM_DISABLED = "#e1e4e8ff";
export const CONTROL_ICON_STRONG_DEFAULT = "#d0d4d9ff";
export const CONTROL_ICON_STRONG_HOVER = "#f0f1f2ff";
export const CONTROL_ICON_STRONG_FOCUS = "#649ef5ff";
export const CONTROL_ICON_STRONG_ACTIVE = "#649ef5ff";
export const CONTROL_ICON_STRONG_DISABLED = "#6f7680ff";
export const CONTROL_ICON_IN_DEFAULT = "#23282eff";
export const CONTROL_ICON_IN_HOVER = "#23282eff";
export const CONTROL_ICON_IN_FOCUS = "#23282eff";
export const CONTROL_ICON_IN_ACTIVE = "#23282eff";
export const CONTROL_ICON_IN_DISABLED = "#6f7680ff";
export const CONTROL_ICON_IN_STRONG_DEFAULT = "#649ef5ff";
export const CONTROL_ICON_IN_STRONG_HOVER = "#7cadf7ff";
export const CONTROL_ICON_IN_STRONG_FOCUS = "#7cadf7ff";
export const CONTROL_ICON_IN_STRONG_ACTIVE = "#7cadf7ff";
export const CONTROL_ICON_IN_STRONG_DISABLED = "#6f7680ff";
export const CONTROL_BORDER_DEFAULT = "#7e868fff";
export const CONTROL_BORDER_HOVER = "#7cadf7ff";
export const CONTROL_BORDER_FOCUS = "#7cadf7ff";
export const CONTROL_BORDER_ACTIVE = "#7cadf7ff";
export const CONTROL_BORDER_DISABLED = "#596069ff";
export const CONTROL_BORDER_WEAK_DEFAULT = "#656c75ff";
export const CONTROL_BORDER_WEAK_HOVER = "#7cadf7ff";
export const CONTROL_BORDER_WEAK_FOCUS = "#7cadf7ff";
export const CONTROL_BORDER_WEAK_ACTIVE = "#7cadf7ff";
export const CONTROL_BORDER_WEAK_DISABLED = "#596069ff";
export const CONTROL_BORDER_MEDIUM_DEFAULT = "#649ef5ff";
export const CONTROL_BORDER_MEDIUM_HOVER = "#649ef5ff";
export const CONTROL_BORDER_MEDIUM_FOCUS = "#649ef5ff";
export const CONTROL_BORDER_MEDIUM_ACTIVE = "#649ef5ff";
export const CONTROL_BORDER_MEDIUM_DISABLED = "#596069ff";
export const CONTROL_BORDER_STRONG_DEFAULT = "#c1c6ccff";
export const CONTROL_BORDER_STRONG_HOVER = "#656c75ff";
export const CONTROL_BORDER_STRONG_FOCUS = "#656c75ff";
export const CONTROL_BORDER_STRONG_ACTIVE = "#649ef5ff";
export const CONTROL_BORDER_STRONG_DISABLED = "#596069ff";
export const INTERACT_BG_DEFAULT = "#649ef5ff";
export const INTERACT_BG_HOVER = "#7cadf7ff";
export const INTERACT_BG_FOCUS = "#7cadf7ff";
export const INTERACT_BG_ACTIVE = "#7cadf7ff";
export const INTERACT_BG_DISABLED = "#446392ff";
export const INTERACT_BG_WEAK_DEFAULT = "#23282eff";
export const INTERACT_BG_WEAK_HOVER = "#2f3d52ff";
export const INTERACT_BG_WEAK_FOCUS = "#2f3d52ff";
export const INTERACT_BG_WEAK_ACTIVE = "#2f3d52ff";
export const INTERACT_BG_WEAK_DISABLED = "#464c54ff";
export const INTERACT_BG_MEDIUM_DEFAULT = "#649ef5ff";
export const INTERACT_BG_MEDIUM_HOVER = "#596069ff";
export const INTERACT_BG_MEDIUM_ACTIVE = "#3a5276ff";
export const INTERACT_BG_STRONG_HOVER = "#596069ff";
export const INTERACT_BG_STRONG_ACTIVE = "#596069ff";
export const INTERACT_TEXT_DEFAULT = "#649ef5ff";
export const INTERACT_TEXT_HOVER = "#7cadf7ff";
export const INTERACT_TEXT_FOCUS = "#7cadf7ff";
export const INTERACT_TEXT_ACTIVE = "#7cadf7ff";
export const INTERACT_TEXT_DISABLED = "#446392ff";
export const INTERACT_TEXT_WEAK_DEFAULT = "#c1c6ccff";
export const INTERACT_TEXT_WEAK_HOVER = "#649ef5ff";
export const INTERACT_TEXT_WEAK_FOCUS = "#649ef5ff";
export const INTERACT_TEXT_WEAK_ACTIVE = "#649ef5ff";
export const INTERACT_TEXT_WEAK_DISABLED = "#6f7680ff";
export const INTERACT_TEXT_IN_DEFAULT = "#23282eff";
export const INTERACT_TEXT_IN_HOVER = "#23282eff";
export const INTERACT_TEXT_IN_FOCUS = "#23282eff";
export const INTERACT_TEXT_IN_ACTIVE = "#23282eff";
export const INTERACT_TEXT_IN_DISABLED = "#23282eff";
export const INTERACT_ICON_DEFAULT = "#649ef5ff";
export const INTERACT_ICON_HOVER = "#7cadf7ff";
export const INTERACT_ICON_FOCUS = "#7cadf7ff";
export const INTERACT_ICON_ACTIVE = "#7cadf7ff";
export const INTERACT_ICON_DISABLED = "#446392ff";
export const INTERACT_ICON_WEAK_DEFAULT = "#c1c6ccff";
export const INTERACT_ICON_WEAK_HOVER = "#d0d4d9ff";
export const INTERACT_ICON_WEAK_FOCUS = "#d0d4d9ff";
export const INTERACT_ICON_WEAK_ACTIVE = "#d0d4d9ff";
export const INTERACT_ICON_WEAK_DISABLED = "#6f7680ff";
export const INTERACT_ICON_MEDIUM_DEFAULT = "#c1c6ccff";
export const INTERACT_ICON_MEDIUM_HOVER = "#649ef5ff";
export const INTERACT_ICON_MEDIUM_FOCUS = "#649ef5ff";
export const INTERACT_ICON_MEDIUM_ACTIVE = "#649ef5ff";
export const INTERACT_ICON_MEDIUM_DISABLED = "#6f7680ff";
export const INTERACT_ICON_STRONG_DEFAULT = "#d0d4d9ff";
export const INTERACT_ICON_STRONG_HOVER = "#f0f1f2ff";
export const INTERACT_ICON_STRONG_FOCUS = "#f0f1f2ff";
export const INTERACT_ICON_STRONG_ACTIVE = "#f0f1f2ff";
export const INTERACT_ICON_STRONG_DISABLED = "#6f7680ff";
export const INTERACT_ICON_IN_DEFAULT = "#23282eff";
export const INTERACT_ICON_IN_HOVER = "#23282eff";
export const INTERACT_ICON_IN_FOCUS = "#23282eff";
export const INTERACT_ICON_IN_ACTIVE = "#23282eff";
export const INTERACT_ICON_IN_DISABLED = "#23282eff";
export const INTERACT_BORDER_DEFAULT = "#649ef5ff";
export const INTERACT_BORDER_HOVER = "#7cadf7ff";
export const INTERACT_BORDER_FOCUS = "#7cadf7ff";
export const INTERACT_BORDER_ACTIVE = "#7cadf7ff";
export const INTERACT_BORDER_DISABLED = "#446392ff";
export const INTERACT_BORDER_WEAK_DEFAULT = "#656c75ff";
export const INTERACT_BORDER_WEAK_HOVER = "#cce1ffff";
export const INTERACT_BORDER_WEAK_FOCUS = "#e1e4e8ff";
export const INTERACT_BORDER_WEAK_ACTIVE = "#e1e4e8ff";
export const INTERACT_BORDER_WEAK_DISABLED = "#596069ff";
export const INTERACT_BORDER_MEDIUM_DEFAULT = "#649ef5ff";
export const INTERACT_BORDER_MEDIUM_HOVER = "#a3c8ffff";
export const INTERACT_BORDER_MEDIUM_FOCUS = "#649ef5ff";
export const INTERACT_BORDER_MEDIUM_ACTIVE = "#649ef5ff";
export const INTERACT_BORDER_MEDIUM_DISABLED = "#e1e4e8ff";
export const POSITIVE_BG_DEFAULT = "#6bbf41ff";
export const POSITIVE_BG_HOVER = "#6bbf41ff";
export const POSITIVE_BG_FOCUS = "#6bbf41ff";
export const POSITIVE_BG_ACTIVE = "#6bbf41ff";
export const POSITIVE_BG_DISABLED = "#6bbf41ff";
export const POSITIVE_BG_WEAK_DEFAULT = "#23282eff";
export const POSITIVE_BG_WEAK_HOVER = "#23282eff";
export const POSITIVE_BG_WEAK_FOCUS = "#23282eff";
export const POSITIVE_BG_WEAK_ACTIVE = "#23282eff";
export const POSITIVE_BG_WEAK_DISABLED = "#23282eff";
export const POSITIVE_BG_MEDIUM_DEFAULT = "#395534ff";
export const POSITIVE_BG_MEDIUM_HOVER = "#395534ff";
export const POSITIVE_BG_MEDIUM_FOCUS = "#395534ff";
export const POSITIVE_BG_MEDIUM_ACTIVE = "#395534ff";
export const POSITIVE_BG_MEDIUM_DISABLED = "#373c42ff";
export const POSITIVE_BG_STRONG_DEFAULT = "#23381eff";
export const POSITIVE_ICON_DEFAULT = "#6bbf41ff";
export const POSITIVE_ICON_HOVER = "#6bbf41ff";
export const POSITIVE_ICON_FOCUS = "#6bbf41ff";
export const POSITIVE_ICON_ACTIVE = "#6bbf41ff";
export const POSITIVE_ICON_DISABLED = "#596069ff";
export const POSITIVE_ICON_WEAK_DEFAULT = "#6bbf41ff";
export const POSITIVE_ICON_WEAK_HOVER = "#6bbf41ff";
export const POSITIVE_ICON_WEAK_FOCUS = "#6bbf41ff";
export const POSITIVE_ICON_WEAK_ACTIVE = "#6bbf41ff";
export const POSITIVE_ICON_WEAK_DISABLED = "#596069ff";
export const POSITIVE_ICON_STRONG_DEFAULT = "#6bbf41ff";
export const POSITIVE_ICON_STRONG_HOVER = "#6bbf41ff";
export const POSITIVE_ICON_STRONG_FOCUS = "#6bbf41ff";
export const POSITIVE_ICON_STRONG_ACTIVE = "#6bbf41ff";
export const POSITIVE_ICON_STRONG_DISABLED = "#477438ff";
export const POSITIVE_ICON_IN_DEFAULT = "#23282eff";
export const POSITIVE_ICON_IN_HOVER = "#23282eff";
export const POSITIVE_ICON_IN_FOCUS = "#23282eff";
export const POSITIVE_ICON_IN_ACTIVE = "#23282eff";
export const POSITIVE_ICON_IN_DISABLED = "#23282eff";
export const POSITIVE_BORDER_DEFAULT = "#6bbf41ff";
export const POSITIVE_BORDER_HOVER = "#6bbf41ff";
export const POSITIVE_BORDER_FOCUS = "#6bbf41ff";
export const POSITIVE_BORDER_ACTIVE = "#6bbf41ff";
export const POSITIVE_BORDER_DISABLED = "#6bbf41ff";
export const POSITIVE_BORDER_WEAK_DEFAULT = "#23282eff";
export const POSITIVE_BORDER_WEAK_HOVER = "#6bbf41ff";
export const POSITIVE_BORDER_WEAK_FOCUS = "#6bbf41ff";
export const POSITIVE_BORDER_WEAK_ACTIVE = "#6bbf41ff";
export const POSITIVE_BORDER_WEAK_DISABLED = "#596069ff";
export const POSITIVE_BORDER_STRONG_DEFAULT = "#f7f7f7ff";
export const POSITIVE_BORDER_STRONG_HOVER = "#f7f7f7ff";
export const POSITIVE_BORDER_STRONG_FOCUS = "#f7f7f7ff";
export const POSITIVE_BORDER_STRONG_ACTIVE = "#f7f7f7ff";
export const POSITIVE_BORDER_STRONG_DISABLED = "#f7f7f7ff";
export const POSITIVE_TEXT_IN_DEFAULT = "#23282eff";
export const POSITIVE_TEXT_IN_HOVER = "#23282eff";
export const POSITIVE_TEXT_IN_FOCUS = "#23282eff";
export const POSITIVE_TEXT_IN_ACTIVE = "#23282eff";
export const POSITIVE_TEXT_IN_DISABLED = "#23282eff";
export const WARNING_BG_DEFAULT = "#f0c243ff";
export const WARNING_BG_HOVER = "#f0c243ff";
export const WARNING_BG_FOCUS = "#f0c243ff";
export const WARNING_BG_ACTIVE = "#f0c243ff";
export const WARNING_BG_DISABLED = "#f0c243ff";
export const WARNING_BG_WEAK_DEFAULT = "#23282eff";
export const WARNING_BG_WEAK_HOVER = "#23282eff";
export const WARNING_BG_WEAK_FOCUS = "#23282eff";
export const WARNING_BG_WEAK_ACTIVE = "#23282eff";
export const WARNING_BG_WEAK_DISABLED = "#23282eff";
export const WARNING_BG_MEDIUM_DEFAULT = "#5c4d28ff";
export const WARNING_BG_MEDIUM_HOVER = "#5c4d28ff";
export const WARNING_BG_MEDIUM_FOCUS = "#5c4d28ff";
export const WARNING_BG_MEDIUM_ACTIVE = "#5c4d28ff";
export const WARNING_BG_MEDIUM_DISABLED = "#373c42ff";
export const WARNING_BG_STRONG_DEFAULT = "#35301cff";
export const WARNING_ICON_DEFAULT = "#f0c243ff";
export const WARNING_ICON_HOVER = "#f0c243ff";
export const WARNING_ICON_FOCUS = "#f0c243ff";
export const WARNING_ICON_ACTIVE = "#f0c243ff";
export const WARNING_ICON_DISABLED = "#596069ff";
export const WARNING_ICON_STRONG_DEFAULT = "#f0c243ff";
export const WARNING_ICON_STRONG_HOVER = "#f0c243ff";
export const WARNING_ICON_STRONG_FOCUS = "#f0c243ff";
export const WARNING_ICON_STRONG_ACTIVE = "#f0c243ff";
export const WARNING_ICON_STRONG_DISABLED = "#f0c243ff";
export const WARNING_ICON_IN_DEFAULT = "#23282eff";
export const WARNING_ICON_IN_HOVER = "#23282eff";
export const WARNING_ICON_IN_FOCUS = "#23282eff";
export const WARNING_ICON_IN_ACTIVE = "#23282eff";
export const WARNING_ICON_IN_DISABLED = "#23282eff";
export const WARNING_BORDER_DEFAULT = "#f0c243ff";
export const WARNING_BORDER_HOVER = "#f0c243ff";
export const WARNING_BORDER_FOCUS = "#f0c243ff";
export const WARNING_BORDER_ACTIVE = "#f0c243ff";
export const WARNING_BORDER_DISABLED = "#f0c243ff";
export const WARNING_BORDER_WEAK_DEFAULT = "#23282eff";
export const WARNING_BORDER_WEAK_HOVER = "#23282eff";
export const WARNING_BORDER_WEAK_FOCUS = "#23282eff";
export const WARNING_BORDER_WEAK_ACTIVE = "#23282eff";
export const WARNING_BORDER_WEAK_DISABLED = "#596069ff";
export const WARNING_BORDER_STRONG_DEFAULT = "#f7f7f7ff";
export const WARNING_BORDER_STRONG_HOVER = "#f7f7f7ff";
export const WARNING_BORDER_STRONG_FOCUS = "#f7f7f7ff";
export const WARNING_BORDER_STRONG_ACTIVE = "#f7f7f7ff";
export const WARNING_BORDER_STRONG_DISABLED = "#f7f7f7ff";
export const WARNING_TEXT_IN_DEFAULT = "#23282eff";
export const WARNING_TEXT_IN_HOVER = "#23282eff";
export const WARNING_TEXT_IN_FOCUS = "#23282eff";
export const WARNING_TEXT_IN_ACTIVE = "#23282eff";
export const WARNING_TEXT_IN_DISABLED = "#23282eff";
export const SEVERE_WARNING_BG_DEFAULT = "#f7782fff";
export const SEVERE_WARNING_BG_HOVER = "#f7782fff";
export const SEVERE_WARNING_BG_FOCUS = "#f7782fff";
export const SEVERE_WARNING_BG_ACTIVE = "#f7782fff";
export const SEVERE_WARNING_BG_DISABLED = "#f7782fff";
export const SEVERE_WARNING_BG_WEAK_DEFAULT = "#23282eff";
export const SEVERE_WARNING_BG_WEAK_HOVER = "#23282eff";
export const SEVERE_WARNING_BG_WEAK_FOCUS = "#23282eff";
export const SEVERE_WARNING_BG_WEAK_ACTIVE = "#23282eff";
export const SEVERE_WARNING_BG_WEAK_DISABLED = "#23282eff";
export const SEVERE_WARNING_BG_MEDIUM_DEFAULT = "#644637ff";
export const SEVERE_WARNING_BG_MEDIUM_HOVER = "#644637ff";
export const SEVERE_WARNING_BG_MEDIUM_FOCUS = "#644637ff";
export const SEVERE_WARNING_BG_MEDIUM_ACTIVE = "#644637ff";
export const SEVERE_WARNING_BG_MEDIUM_DISABLED = "#373c42ff";
export const SEVERE_WARNING_BG_STRONG_DEFAULT = "#42281aff";
export const SEVERE_WARNING_ICON_DEFAULT = "#f7782fff";
export const SEVERE_WARNING_ICON_HOVER = "#f7782fff";
export const SEVERE_WARNING_ICON_FOCUS = "#f7782fff";
export const SEVERE_WARNING_ICON_ACTIVE = "#f7782fff";
export const SEVERE_WARNING_ICON_DISABLED = "#596069ff";
export const SEVERE_WARNING_ICON_STRONG_DEFAULT = "#fc8d4cff";
export const SEVERE_WARNING_ICON_STRONG_HOVER = "#fc8d4cff";
export const SEVERE_WARNING_ICON_STRONG_FOCUS = "#fc8d4cff";
export const SEVERE_WARNING_ICON_STRONG_ACTIVE = "#fc8d4cff";
export const SEVERE_WARNING_ICON_STRONG_DISABLED = "#fc8d4cff";
export const SEVERE_WARNING_ICON_IN_DEFAULT = "#23282eff";
export const SEVERE_WARNING_ICON_IN_HOVER = "#23282eff";
export const SEVERE_WARNING_ICON_IN_FOCUS = "#23282eff";
export const SEVERE_WARNING_ICON_IN_ACTIVE = "#23282eff";
export const SEVERE_WARNING_ICON_IN_DISABLED = "#23282eff";
export const SEVERE_WARNING_BORDER_DEFAULT = "#f7782fff";
export const SEVERE_WARNING_BORDER_HOVER = "#f7782fff";
export const SEVERE_WARNING_BORDER_FOCUS = "#f7782fff";
export const SEVERE_WARNING_BORDER_ACTIVE = "#f7782fff";
export const SEVERE_WARNING_BORDER_DISABLED = "#f7782fff";
export const SEVERE_WARNING_BORDER_WEAK_DEFAULT = "#23282eff";
export const SEVERE_WARNING_BORDER_WEAK_HOVER = "#23282eff";
export const SEVERE_WARNING_BORDER_WEAK_FOCUS = "#23282eff";
export const SEVERE_WARNING_BORDER_WEAK_ACTIVE = "#23282eff";
export const SEVERE_WARNING_BORDER_WEAK_DISABLED = "#596069ff";
export const NEGATIVE_BG_DEFAULT = "#fa5762ff";
export const NEGATIVE_BG_HOVER = "#ff6e72ff";
export const NEGATIVE_BG_FOCUS = "#ff6e72ff";
export const NEGATIVE_BG_ACTIVE = "#ff6e72ff";
export const NEGATIVE_BG_DISABLED = "#8f4048ff";
export const NEGATIVE_BG_WEAK_DEFAULT = "#23282eff";
export const NEGATIVE_BG_WEAK_HOVER = "#994a52ff";
export const NEGATIVE_BG_WEAK_FOCUS = "#994a52ff";
export const NEGATIVE_BG_WEAK_ACTIVE = "#994a52ff";
export const NEGATIVE_BG_WEAK_DISABLED = "#373c42ff";
export const NEGATIVE_BG_MEDIUM_DEFAULT = "#64363eff";
export const NEGATIVE_BG_MEDIUM_HOVER = "#64363eff";
export const NEGATIVE_BG_MEDIUM_FOCUS = "#64363eff";
export const NEGATIVE_BG_MEDIUM_ACTIVE = "#64363eff";
export const NEGATIVE_BG_MEDIUM_DISABLED = "#373c42ff";
export const NEGATIVE_BG_STRONG_DEFAULT = "#46272aff";
export const NEGATIVE_TEXT_DEFAULT = "#fa5762ff";
export const NEGATIVE_TEXT_HOVER = "#ff878bff";
export const NEGATIVE_TEXT_FOCUS = "#ff878bff";
export const NEGATIVE_TEXT_ACTIVE = "#ff878bff";
export const NEGATIVE_TEXT_DISABLED = "#8f4048ff";
export const NEGATIVE_TEXT_IN_DEFAULT = "#23282eff";
export const NEGATIVE_TEXT_IN_HOVER = "#23282eff";
export const NEGATIVE_TEXT_IN_FOCUS = "#23282eff";
export const NEGATIVE_TEXT_IN_ACTIVE = "#23282eff";
export const NEGATIVE_TEXT_IN_DISABLED = "#23282eff";
export const NEGATIVE_ICON_DEFAULT = "#fa5762ff";
export const NEGATIVE_ICON_HOVER = "#ff878bff";
export const NEGATIVE_ICON_FOCUS = "#ff878bff";
export const NEGATIVE_ICON_ACTIVE = "#ff878bff";
export const NEGATIVE_ICON_DISABLED = "#8f4048ff";
export const NEGATIVE_ICON_WEAK_DEFAULT = "#ffb2b5ff";
export const NEGATIVE_ICON_WEAK_HOVER = "#ffb2b5ff";
export const NEGATIVE_ICON_WEAK_FOCUS = "#ffb2b5ff";
export const NEGATIVE_ICON_WEAK_ACTIVE = "#ffb2b5ff";
export const NEGATIVE_ICON_WEAK_DISABLED = "#596069ff";
export const NEGATIVE_ICON_IN_DEFAULT = "#23282eff";
export const NEGATIVE_ICON_IN_HOVER = "#23282eff";
export const NEGATIVE_ICON_IN_FOCUS = "#23282eff";
export const NEGATIVE_ICON_IN_ACTIVE = "#23282eff";
export const NEGATIVE_ICON_IN_DISABLED = "#23282eff";
export const NEGATIVE_BORDER_DEFAULT = "#fa5762ff";
export const NEGATIVE_BORDER_HOVER = "#ff878bff";
export const NEGATIVE_BORDER_FOCUS = "#ff878bff";
export const NEGATIVE_BORDER_ACTIVE = "#ff878bff";
export const NEGATIVE_BORDER_DISABLED = "#8f4048ff";
export const NEGATIVE_BORDER_WEAK_DEFAULT = "#23282eff";
export const NEGATIVE_BORDER_WEAK_HOVER = "#23282eff";
export const NEGATIVE_BORDER_WEAK_FOCUS = "#23282eff";
export const NEGATIVE_BORDER_WEAK_ACTIVE = "#23282eff";
export const NEGATIVE_BORDER_WEAK_DISABLED = "#596069ff";
export const NEGATIVE_BORDER_MEDIUM_DEFAULT = "#eb4651ff";
export const NEGATIVE_BORDER_MEDIUM_HOVER = "#ff878bff";
export const NEGATIVE_BORDER_MEDIUM_FOCUS = "#ff878bff";
export const NEGATIVE_BORDER_MEDIUM_ACTIVE = "#ff878bff";
export const NEGATIVE_BORDER_MEDIUM_DISABLED = "#8f4048ff";
export const NEGATIVE_BORDER_STRONG_DEFAULT = "#f7f7f7ff";
export const NEGATIVE_BORDER_STRONG_HOVER = "#ffffffff";
export const NEGATIVE_BORDER_STRONG_FOCUS = "#f7f7f7ff";
export const NEGATIVE_BORDER_STRONG_ACTIVE = "#f7f7f7ff";
export const NEGATIVE_BORDER_STRONG_DISABLED = "#f7f7f7ff";
export const INFO_BG_DEFAULT = "#7cadf7ff";
export const INFO_BG_HOVER = "#7cadf7ff";
export const INFO_BG_FOCUS = "#7cadf7ff";
export const INFO_BG_ACTIVE = "#7cadf7ff";
export const INFO_BG_DISABLED = "#7cadf7ff";
export const INFO_BG_WEAK_DEFAULT = "#23282eff";
export const INFO_BG_WEAK_HOVER = "#23282eff";
export const INFO_BG_WEAK_FOCUS = "#23282eff";
export const INFO_BG_WEAK_ACTIVE = "#23282eff";
export const INFO_BG_WEAK_DISABLED = "#23282eff";
export const INFO_BG_MEDIUM_DEFAULT = "#3e506aff";
export const INFO_BG_MEDIUM_HOVER = "#3e506aff";
export const INFO_BG_MEDIUM_FOCUS = "#3e506aff";
export const INFO_BG_MEDIUM_ACTIVE = "#3e506aff";
export const INFO_BG_MEDIUM_DISABLED = "#373c42ff";
export const INFO_BG_STRONG_DEFAULT = "#283648ff";
export const INFO_ICON_DEFAULT = "#7cadf7ff";
export const INFO_ICON_HOVER = "#7cadf7ff";
export const INFO_ICON_FOCUS = "#7cadf7ff";
export const INFO_ICON_ACTIVE = "#7cadf7ff";
export const INFO_ICON_DISABLED = "#596069ff";
export const INFO_ICON_WEAK_DEFAULT = "#7cadf7ff";
export const INFO_ICON_WEAK_HOVER = "#7cadf7ff";
export const INFO_ICON_WEAK_FOCUS = "#7cadf7ff";
export const INFO_ICON_WEAK_ACTIVE = "#7cadf7ff";
export const INFO_ICON_WEAK_DISABLED = "#596069ff";
export const INFO_ICON_IN_DEFAULT = "#23282eff";
export const INFO_ICON_IN_HOVER = "#23282eff";
export const INFO_ICON_IN_FOCUS = "#23282eff";
export const INFO_ICON_IN_ACTIVE = "#23282eff";
export const INFO_ICON_IN_DISABLED = "#23282eff";
export const INFO_BORDER_DEFAULT = "#7cadf7ff";
export const INFO_BORDER_HOVER = "#7cadf7ff";
export const INFO_BORDER_FOCUS = "#7cadf7ff";
export const INFO_BORDER_ACTIVE = "#7cadf7ff";
export const INFO_BORDER_DISABLED = "#7cadf7ff";
export const INFO_BORDER_WEAK_DEFAULT = "#23282eff";
export const INFO_BORDER_WEAK_HOVER = "#23282eff";
export const INFO_BORDER_WEAK_FOCUS = "#23282eff";
export const INFO_BORDER_WEAK_ACTIVE = "#23282eff";
export const INFO_BORDER_WEAK_DISABLED = "#596069ff";
export const DORMANT_BG_DEFAULT = "#6f7680ff";
export const DORMANT_BG_DISABLED = "#6f7680ff";
export const DORMANT_BG_MEDIUM_DEFAULT = "#464c54ff";
export const DORMANT_BG_MEDIUM_DISABLED = "#373c42ff";
export const DORMANT_ICON_DEFAULT = "#c1c6ccff";
export const DORMANT_ICON_DISABLED = "#596069ff";
export const DORMANT_ICON_MEDIUM_DEFAULT = "#ffffffff";
export const DORMANT_ICON_IN_DEFAULT = "#23282eff";
export const DORMANT_ICON_IN_DISABLED = "#23282eff";
export const DORMANT_ICON_IN_MEDIUM_DEFAULT = "#23282eff";
export const DORMANT_BORDER_DEFAULT = "#a7adb5ff";
export const DORMANT_BORDER_DISABLED = "#596069ff";
export const ACCENT_A_DEFAULT = "#9ca6ffff";
export const ACCENT_A_HOVER = "#8a95ffff";
export const ACCENT_A_WEAK_DEFAULT = "#4d5477ff";
export const ACCENT_B_DEFAULT = "#e3447cff";
export const ACCENT_B_HOVER = "#cf3a7aff";
export const ACCENT_B_WEAK_DEFAULT = "#704959ff";
export const ACCENT_C_DEFAULT = "#fcb3c8ff";
export const ACCENT_C_HOVER = "#ff87a9ff";
export const ACCENT_C_WEAK_DEFAULT = "#6f5964ff";
export const ACCENT_D_DEFAULT = "#9b5ff5ff";
export const ACCENT_D_HOVER = "#8d4eedff";
export const ACCENT_D_WEAK_DEFAULT = "#5b5077ff";
export const ACCENT_E_DEFAULT = "#9dba4cff";
export const ACCENT_E_HOVER = "#89ab2cff";
export const ACCENT_E_WEAK_DEFAULT = "#4e5b39ff";
export const ACCENT_F_DEFAULT = "#d95a1aff";
export const ACCENT_F_HOVER = "#c44f14ff";
export const ACCENT_F_WEAK_DEFAULT = "#6f4b39ff";
export const ACCENT_G_DEFAULT = "#4ad9d9ff";
export const ACCENT_G_HOVER = "#17c2c2ff";
export const ACCENT_G_WEAK_DEFAULT = "#1f5e62ff";
export const ACCENT_H_DEFAULT = "#028e99ff";
export const ACCENT_H_HOVER = "#01818cff";
export const ACCENT_H_WEAK_DEFAULT = "#45696dff";
export const ACCENT_I_DEFAULT = "#f582d8ff";
export const ACCENT_I_HOVER = "#f26dd1ff";
export const ACCENT_I_WEAK_DEFAULT = "#6d486aff";
export const ACCENT_J_DEFAULT = "#767eb2ff";
export const ACCENT_J_HOVER = "#6871a3ff";
export const ACCENT_J_WEAK_DEFAULT = "#505669ff";
export const ACCENT_K_DEFAULT = "#4cbf7fff";
export const ACCENT_K_HOVER = "#36b26eff";
export const ACCENT_K_WEAK_DEFAULT = "#315d4aff";
export const IN_PROGRESS_BG_MEDIUM_DEFAULT = "#3e506aff";
export const IN_PROGRESS_BG_MEDIUM_HOVER = "#3e506aff";
export const IN_PROGRESS_BG_MEDIUM_FOCUS = "#3e506aff";
export const IN_PROGRESS_BG_MEDIUM_ACTIVE = "#3e506aff";
export const IN_PROGRESS_BG_MEDIUM_DISABLED = "#373c42ff";
export const IN_PROGRESS_ICON_DEFAULT = "#7cadf7ff";
export const IN_PROGRESS_ICON_HOVER = "#7cadf7ff";
export const IN_PROGRESS_ICON_FOCUS = "#7cadf7ff";
export const IN_PROGRESS_ICON_ACTIVE = "#7cadf7ff";
export const IN_PROGRESS_ICON_DISABLED = "#596069ff";
export const IN_PROGRESS_ICON_IN_DEFAULT = "#23282eff";
export const IN_PROGRESS_ICON_IN_HOVER = "#23282eff";
export const IN_PROGRESS_ICON_IN_FOCUS = "#23282eff";
export const IN_PROGRESS_ICON_IN_ACTIVE = "#23282eff";
export const IN_PROGRESS_ICON_IN_DISABLED = "#23282eff";
export const IN_PROGRESS_BORDER_DEFAULT = "#7cadf7ff";
export const IN_PROGRESS_BORDER_HOVER = "#7cadf7ff";
export const IN_PROGRESS_BORDER_FOCUS = "#7cadf7ff";
export const IN_PROGRESS_BORDER_ACTIVE = "#7cadf7ff";
export const IN_PROGRESS_BORDER_DISABLED = "#7cadf7ff";
export const DECENT_BG_MEDIUM_DEFAULT = "#485437ff";
export const DECENT_BG_MEDIUM_HOVER = "#485437ff";
export const DECENT_BG_MEDIUM_FOCUS = "#485437ff";
export const DECENT_BG_MEDIUM_ACTIVE = "#485437ff";
export const DECENT_BG_MEDIUM_DISABLED = "#373c42ff";
export const DECENT_ICON_DEFAULT = "#7da11bff";
export const DECENT_ICON_HOVER = "#7da11bff";
export const DECENT_ICON_FOCUS = "#7da11bff";
export const DECENT_ICON_ACTIVE = "#7da11bff";
export const DECENT_ICON_DISABLED = "#596069ff";
export const DECENT_ICON_IN_DEFAULT = "#23282eff";
export const DECENT_ICON_IN_HOVER = "#23282eff";
export const DECENT_ICON_IN_FOCUS = "#23282eff";
export const DECENT_ICON_IN_ACTIVE = "#23282eff";
export const DECENT_ICON_IN_DISABLED = "#23282eff";
export const DECENT_BORDER_DEFAULT = "#6c8c14ff";
export const DECENT_BORDER_HOVER = "#6c8c14ff";
export const DECENT_BORDER_FOCUS = "#6c8c14ff";
export const DECENT_BORDER_ACTIVE = "#6c8c14ff";
export const DECENT_BORDER_DISABLED = "#6c8c14ff";
export const EXCELLENT_BG_MEDIUM_DEFAULT = "#2f5546ff";
export const EXCELLENT_BG_MEDIUM_HOVER = "#2f5546ff";
export const EXCELLENT_BG_MEDIUM_FOCUS = "#2f5546ff";
export const EXCELLENT_BG_MEDIUM_ACTIVE = "#2f5546ff";
export const EXCELLENT_BG_MEDIUM_DISABLED = "#373c42ff";
export const EXCELLENT_ICON_DEFAULT = "#4cbf7fff";
export const EXCELLENT_ICON_HOVER = "#4cbf7fff";
export const EXCELLENT_ICON_FOCUS = "#4cbf7fff";
export const EXCELLENT_ICON_ACTIVE = "#4cbf7fff";
export const EXCELLENT_ICON_DISABLED = "#596069ff";
export const EXCELLENT_ICON_IN_DEFAULT = "#23282eff";
export const EXCELLENT_ICON_IN_HOVER = "#23282eff";
export const EXCELLENT_ICON_IN_FOCUS = "#23282eff";
export const EXCELLENT_ICON_IN_ACTIVE = "#23282eff";
export const EXCELLENT_ICON_IN_DISABLED = "#23282eff";
export const EXCELLENT_BORDER_DEFAULT = "#4cbf7fff";
export const EXCELLENT_BORDER_HOVER = "#4cbf7fff";
export const EXCELLENT_BORDER_FOCUS = "#4cbf7fff";
export const EXCELLENT_BORDER_ACTIVE = "#4cbf7fff";
export const EXCELLENT_BORDER_DISABLED = "#4cbf7fff";
