// @ciscodesignsystems/cds-magnetic-theme-web/js/dist/token-theme-classic-light-hex8-es6.js
/**
 * This file is auto-generated. Do not edit it directly.
 */

export const BRAND_BG_DEFAULT = "#373c42ff";
export const BRAND_BG_HOVER = "#373c42ff";
export const BRAND_BG_FOCUS = "#464c54ff";
export const BRAND_BG_ACTIVE = "#464c54ff";
export const BRAND_BG_DISABLED = "#e1e4e8ff";
export const BRAND_BG_GRADIENT_START = "#373c42ff";
export const BRAND_BG_GRADIENT_END = "#0f1214ff";
export const BRAND_BG_STRONG_DEFAULT = "#373c42ff";
export const BRAND_BG_STRONG_HOVER = "#373c42ff";
export const BRAND_BG_STRONG_FOCUS = "#373c42ff";
export const BRAND_BG_STRONG_ACTIVE = "#373c42ff";
export const BRAND_TEXT_DEFAULT = "#f7f7f7ff";
export const BRAND_TEXT_HOVER = "#f7f7f7ff";
export const BRAND_TEXT_ACTIVE = "#f7f7f7ff";
export const BRAND_TEXT_WEAK_DEFAULT = "#889099ff";
export const BRAND_TEXT_WEAK_HOVER = "#f7f7f7ff";
export const BRAND_TEXT_WEAK_FOCUS = "#f7f7f7ff";
export const BRAND_TEXT_WEAK_ACTIVE = "#f7f7f7ff";
export const BRAND_TEXT_STRONG_DEFAULT = "#f0f1f2ff";
export const BRAND_ICON_DEFAULT = "#f7f7f7ff";
export const BRAND_ICON_HOVER = "#ffffffff";
export const BRAND_ICON_ACTIVE = "#ffffffff";
export const BRAND_ICON_WEAK_DEFAULT = "#f7f7f7ff";
export const BRAND_ICON_WEAK_HOVER = "#f7f7f7ff";
export const BRAND_ICON_WEAK_FOCUS = "#f7f7f7ff";
export const BRAND_ICON_WEAK_ACTIVE = "#f7f7f7ff";
export const BRAND_BORDER_ACTIVE = "#cce1ffff";
export const BRAND_ACCENT_ICON_DEFAULT = "#7e868fff";
export const BRAND_ACCENT_ICON_FOCUS = "#ffffffff";
export const BRAND_ACCENT_ICON_ACTIVE = "#45991fff";
export const BRAND_ACCENT_ICON_DISABLED = "#d0d4d9ff";
export const BRAND_ACCENT_ICON_WEAK_DEFAULT = "#c1c6ccff";
export const BRAND_ACCENT_ICON_WEAK_FOCUS = "#ffffffff";
export const BRAND_ACCENT_ICON_WEAK_ACTIVE = "#6bbf41ff";
export const BRAND_ACCENT_ICON_WEAK_DISABLED = "#d0d4d9ff";
export const BRAND_ACCENT_ICON_MEDIUM_DEFAULT = "#a7adb5ff";
export const BRAND_ACCENT_ICON_MEDIUM_FOCUS = "#ffffffff";
export const BRAND_ACCENT_ICON_MEDIUM_ACTIVE = "#52a62bff";
export const BRAND_ACCENT_ICON_MEDIUM_DISABLED = "#d0d4d9ff";
export const BRAND_ACCENT_BORDER_DEFAULT = "#52a62bff";
export const BRAND_ACCENT_BORDER_HOVER = "#e1e4e8ff";
export const BRAND_ACCENT_BORDER_FOCUS = "#45991fff";
export const BRAND_ACCENT_BORDER_ACTIVE = "#45991fff";
export const BRAND_ACCENT_BORDER_WEAK_DEFAULT = "#d0d4d9ff";
export const BRAND_ACCENT_BORDER_WEAK_HOVER = "#ffffffff";
export const BRAND_ACCENT_BORDER_WEAK_FOCUS = "#ffffffff";
export const BRAND_ACCENT_BORDER_WEAK_ACTIVE = "#6bbf41ff";
export const BRAND_ACCENT_BORDER_WEAK_DISABLED = "#ffffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_DEFAULT = "#f0f1f2ff";
export const BRAND_ACCENT_BORDER_MEDIUM_HOVER = "#ffffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_FOCUS = "#ffffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_ACTIVE = "#e0f5d5ff";
export const BRAND_ACCENT_BORDER_MEDIUM_DISABLED = "#ffffffff";
export const BRAND_ACCENT_BORDER_STRONG_DEFAULT = "#f7f7f7ff";
export const BRAND_ACCENT_BORDER_STRONG_HOVER = "#f7f7f7ff";
export const BRAND_ACCENT_BORDER_STRONG_FOCUS = "#f7f7f7ff";
export const BRAND_ACCENT_BORDER_STRONG_ACTIVE = "#f7f7f7ff";
export const BRAND_ACCENT_BORDER_STRONG_DISABLED = "#ffffffff";
export const BASE_BG_DEFAULT = "#f7f7f7ff";
export const BASE_BG_HOVER = "#f7f7f7ff";
export const BASE_BG_ACTIVE = "#f7f7f7ff";
export const BASE_BG_DISABLED = "#f0f1f2ff";
export const BASE_BG_WEAK_DEFAULT = "#ffffffff";
export const BASE_BG_WEAK_HOVER = "#e1e4e8cc";
export const BASE_BG_WEAK_ACTIVE = "#f0f1f2ff";
export const BASE_BG_WEAK_DISABLED = "#f0f1f2ff";
export const BASE_BG_MEDIUM_DEFAULT = "#f7f7f7ff";
export const BASE_BG_MEDIUM_HOVER = "#d0d4d9ff";
export const BASE_BG_MEDIUM_ACTIVE = "#e1e4e8ff";
export const BASE_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const BASE_BG_STRONG_DEFAULT = "#656c75ff";
export const BASE_BG_STRONG_HOVER = "#23282eff";
export const BASE_BG_STRONG_FOCUS = "#23282eff";
export const BASE_BG_STRONG_ACTIVE = "#23282eff";
export const BASE_BG_STRONG_DISABLED = "#d0d4d9ff";
export const BASE_ICON_IN_DEFAULT = "#ffffffff";
export const BASE_ICON_IN_HOVER = "#ffffffff";
export const BASE_ICON_IN_FOCUS = "#ffffffff";
export const BASE_ICON_IN_ACTIVE = "#ffffffff";
export const BASE_ICON_IN_DISABLED = "#ffffffff";
export const BASE_TEXT_DEFAULT = "#23282eff";
export const BASE_TEXT_DISABLED = "#c1c6ccff";
export const BASE_TEXT_WEAK_DEFAULT = "#889099ff";
export const BASE_TEXT_WEAK_DISABLED = "#c1c6ccff";
export const BASE_TEXT_MEDIUM_DEFAULT = "#596069ff";
export const BASE_TEXT_MEDIUM_DISABLED = "#c1c6ccff";
export const BASE_TEXT_STRONG_DEFAULT = "#373c42ff";
export const BASE_TEXT_STRONG_DISABLED = "#c1c6ccff";
export const BASE_TEXT_IN_DEFAULT = "#ffffffff";
export const BASE_TEXT_IN_HOVER = "#ffffffff";
export const BASE_TEXT_IN_FOCUS = "#ffffffff";
export const BASE_TEXT_IN_ACTIVE = "#ffffffff";
export const BASE_TEXT_IN_DISABLED = "#ffffffff";
export const BASE_TEXT_IN_WEAK_DEFAULT = "#f7f7f7ff";
export const BASE_BORDER_DEFAULT = "#e1e4e8ff";
export const BASE_BORDER_WEAK_DEFAULT = "#ffffffff";
export const BASE_BORDER_MEDIUM_DEFAULT = "#e1e4e8ff";
export const BASE_BORDER_STRONG_DEFAULT = "#656c75ff";
export const INVERSE_BG_DEFAULT = "#373c42ff";
export const INVERSE_BG_HOVER = "#373c42ff";
export const INVERSE_BG_FOCUS = "#464c54ff";
export const INVERSE_BG_ACTIVE = "#464c54ff";
export const INVERSE_BG_DISABLED = "#e1e4e8ff";
export const INVERSE_BG_GRADIENT_START = "#373c42ff";
export const INVERSE_BG_GRADIENT_END = "#0f1214ff";
export const INVERSE_BG_WEAK_DEFAULT = "#373c42ff";
export const INVERSE_TEXT_DEFAULT = "#f7f7f7ff";
export const INVERSE_TEXT_HOVER = "#f7f7f7ff";
export const INVERSE_TEXT_FOCUS = "#f7f7f7ff";
export const INVERSE_TEXT_ACTIVE = "#f7f7f7ff";
export const INVERSE_TEXT_DISABLED = "#656c75ff";
export const INVERSE_TEXT_MEDIUM_DEFAULT = "#c1c6ccff";
export const INVERSE_TEXT_MEDIUM_DISABLED = "#656c75ff";
export const INVERSE_ICON_DEFAULT = "#f7f7f7ff";
export const INVERSE_ICON_HOVER = "#ffffffff";
export const INVERSE_ICON_FOCUS = "#ffffffff";
export const INVERSE_ICON_ACTIVE = "#ffffffff";
export const INVERSE_ICON_DISABLED = "#6f7680ff";
export const INVERSE_ICON_STRONG_DEFAULT = "#ffffffff";
export const INVERSE_ICON_STRONG_HOVER = "#ffffffff";
export const INVERSE_ICON_STRONG_FOCUS = "#ffffffff";
export const INVERSE_ICON_STRONG_ACTIVE = "#ffffffff";
export const INVERSE_ICON_STRONG_DISABLED = "#ffffffff";
export const INVERSE_BORDER_DEFAULT = "#6f7680ff";
export const INVERSE_BORDER_HOVER = "#649ef5ff";
export const INVERSE_BORDER_FOCUS = "#cce1ffff";
export const INVERSE_BORDER_ACTIVE = "#cce1ffff";
export const INVERSE_BORDER_DISABLED = "#e1e4e8ff";
export const CONTROL_BG_DEFAULT = "#ffffffff";
export const CONTROL_BG_HOVER = "#e3eeffff";
export const CONTROL_BG_FOCUS = "#f0f1f2ff";
export const CONTROL_BG_ACTIVE = "#f0f1f2ff";
export const CONTROL_BG_DISABLED = "#f7f7f7ff";
export const CONTROL_BG_WEAK_DEFAULT = "#ffffffff";
export const CONTROL_BG_WEAK_HOVER = "#cce1ffff";
export const CONTROL_BG_WEAK_FOCUS = "#e3eeffff";
export const CONTROL_BG_WEAK_ACTIVE = "#e3eeffff";
export const CONTROL_BG_WEAK_DISABLED = "#f7f7f7ff";
export const CONTROL_BG_MEDIUM_DEFAULT = "#ffffffff";
export const CONTROL_BG_MEDIUM_HOVER = "#f7f7f7ff";
export const CONTROL_BG_MEDIUM_FOCUS = "#e3eeffff";
export const CONTROL_BG_MEDIUM_ACTIVE = "#e3eeffff";
export const CONTROL_BG_MEDIUM_DISABLED = "#e1e4e8ff";
export const CONTROL_BG_STRONG_DEFAULT = "#2774d9ff";
export const CONTROL_BG_STRONG_HOVER = "#0d5cbdff";
export const CONTROL_BG_STRONG_FOCUS = "#0d5cbdff";
export const CONTROL_BG_STRONG_ACTIVE = "#0d5cbdff";
export const CONTROL_BG_STRONG_DISABLED = "#bad6ffff";
export const CONTROL_ICON_DEFAULT = "#1d69ccff";
export const CONTROL_ICON_HOVER = "#0d5cbdff";
export const CONTROL_ICON_FOCUS = "#0d5cbdff";
export const CONTROL_ICON_ACTIVE = "#0d5cbdff";
export const CONTROL_ICON_DISABLED = "#a3c8ffff";
export const CONTROL_ICON_WEAK_DEFAULT = "#7e868fff";
export const CONTROL_ICON_WEAK_HOVER = "#2774d9ff";
export const CONTROL_ICON_WEAK_FOCUS = "#2774d9ff";
export const CONTROL_ICON_WEAK_ACTIVE = "#2774d9ff";
export const CONTROL_ICON_WEAK_DISABLED = "#d0d4d9ff";
export const CONTROL_ICON_MEDIUM_DEFAULT = "#2774d9ff";
export const CONTROL_ICON_MEDIUM_HOVER = "#0d5cbdff";
export const CONTROL_ICON_MEDIUM_FOCUS = "#0d5cbdff";
export const CONTROL_ICON_MEDIUM_ACTIVE = "#0d5cbdff";
export const CONTROL_ICON_MEDIUM_DISABLED = "#a3c8ffff";
export const CONTROL_ICON_STRONG_DEFAULT = "#656c75ff";
export const CONTROL_ICON_STRONG_HOVER = "#464c54ff";
export const CONTROL_ICON_STRONG_FOCUS = "#1d69ccff";
export const CONTROL_ICON_STRONG_ACTIVE = "#1d69ccff";
export const CONTROL_ICON_STRONG_DISABLED = "#c1c6ccff";
export const CONTROL_ICON_IN_DEFAULT = "#ffffffff";
export const CONTROL_ICON_IN_HOVER = "#ffffffff";
export const CONTROL_ICON_IN_FOCUS = "#ffffffff";
export const CONTROL_ICON_IN_ACTIVE = "#ffffffff";
export const CONTROL_ICON_IN_DISABLED = "#c1c6ccff";
export const CONTROL_ICON_IN_STRONG_DEFAULT = "#2774d9ff";
export const CONTROL_ICON_IN_STRONG_HOVER = "#2774d9ff";
export const CONTROL_ICON_IN_STRONG_FOCUS = "#2774d9ff";
export const CONTROL_ICON_IN_STRONG_ACTIVE = "#2774d9ff";
export const CONTROL_ICON_IN_STRONG_DISABLED = "#c1c6ccff";
export const CONTROL_BORDER_DEFAULT = "#889099ff";
export const CONTROL_BORDER_HOVER = "#3e84e5ff";
export const CONTROL_BORDER_FOCUS = "#3e84e5ff";
export const CONTROL_BORDER_ACTIVE = "#3e84e5ff";
export const CONTROL_BORDER_DISABLED = "#d0d4d9ff";
export const CONTROL_BORDER_WEAK_DEFAULT = "#e1e4e8ff";
export const CONTROL_BORDER_WEAK_HOVER = "#7cadf7ff";
export const CONTROL_BORDER_WEAK_FOCUS = "#7cadf7ff";
export const CONTROL_BORDER_WEAK_ACTIVE = "#7cadf7ff";
export const CONTROL_BORDER_WEAK_DISABLED = "#d0d4d9ff";
export const CONTROL_BORDER_MEDIUM_DEFAULT = "#2774d9ff";
export const CONTROL_BORDER_MEDIUM_HOVER = "#bad6ffff";
export const CONTROL_BORDER_MEDIUM_FOCUS = "#2774d9ff";
export const CONTROL_BORDER_MEDIUM_ACTIVE = "#2774d9ff";
export const CONTROL_BORDER_MEDIUM_DISABLED = "#d0d4d9ff";
export const CONTROL_BORDER_STRONG_DEFAULT = "#656c75ff";
export const CONTROL_BORDER_STRONG_HOVER = "#656c75ff";
export const CONTROL_BORDER_STRONG_FOCUS = "#2774d9ff";
export const CONTROL_BORDER_STRONG_ACTIVE = "#2774d9ff";
export const CONTROL_BORDER_STRONG_DISABLED = "#d0d4d9ff";
export const INTERACT_BG_DEFAULT = "#1d69ccff";
export const INTERACT_BG_HOVER = "#0d5cbdff";
export const INTERACT_BG_FOCUS = "#0d5cbdff";
export const INTERACT_BG_ACTIVE = "#0d5cbdff";
export const INTERACT_BG_DISABLED = "#a3c8ffff";
export const INTERACT_BG_WEAK_DEFAULT = "#ffffffff";
export const INTERACT_BG_WEAK_HOVER = "#cce1ffff";
export const INTERACT_BG_WEAK_FOCUS = "#cce1ffff";
export const INTERACT_BG_WEAK_ACTIVE = "#cce1ffff";
export const INTERACT_BG_WEAK_DISABLED = "#ffffffff";
export const INTERACT_BG_MEDIUM_DEFAULT = "#2774d9ff";
export const INTERACT_BG_MEDIUM_ACTIVE = "#cce1ffff";
export const INTERACT_BG_STRONG_ACTIVE = "#f7f7f7ff";
export const INTERACT_BG_STRONG_HOVER = "#f7f7f7ff";
export const INTERACT_TEXT_DEFAULT = "#1d69ccff";
export const INTERACT_TEXT_HOVER = "#0d5cbdff";
export const INTERACT_TEXT_FOCUS = "#0d5cbdff";
export const INTERACT_TEXT_ACTIVE = "#0d5cbdff";
export const INTERACT_TEXT_DISABLED = "#a3c8ffff";
export const INTERACT_TEXT_WEAK_DEFAULT = "#656c75ff";
export const INTERACT_TEXT_WEAK_HOVER = "#1d69ccff";
export const INTERACT_TEXT_WEAK_FOCUS = "#1d69ccff";
export const INTERACT_TEXT_WEAK_ACTIVE = "#1d69ccff";
export const INTERACT_TEXT_WEAK_DISABLED = "#c1c6ccff";
export const INTERACT_TEXT_IN_DEFAULT = "#ffffffff";
export const INTERACT_TEXT_IN_HOVER = "#ffffffff";
export const INTERACT_TEXT_IN_FOCUS = "#ffffffff";
export const INTERACT_TEXT_IN_ACTIVE = "#ffffffff";
export const INTERACT_TEXT_IN_DISABLED = "#ffffffff";
export const INTERACT_ICON_DEFAULT = "#1d69ccff";
export const INTERACT_ICON_HOVER = "#0d5cbdff";
export const INTERACT_ICON_FOCUS = "#0d5cbdff";
export const INTERACT_ICON_ACTIVE = "#0d5cbdff";
export const INTERACT_ICON_DISABLED = "#a3c8ffff";
export const INTERACT_ICON_WEAK_DEFAULT = "#7e868fff";
export const INTERACT_ICON_WEAK_HOVER = "#6f7680ff";
export const INTERACT_ICON_WEAK_FOCUS = "#6f7680ff";
export const INTERACT_ICON_WEAK_ACTIVE = "#6f7680ff";
export const INTERACT_ICON_WEAK_DISABLED = "#d0d4d9ff";
export const INTERACT_ICON_MEDIUM_DEFAULT = "#7e868fff";
export const INTERACT_ICON_MEDIUM_HOVER = "#2774d9ff";
export const INTERACT_ICON_MEDIUM_FOCUS = "#2774d9ff";
export const INTERACT_ICON_MEDIUM_ACTIVE = "#2774d9ff";
export const INTERACT_ICON_MEDIUM_DISABLED = "#c1c6ccff";
export const INTERACT_ICON_STRONG_DEFAULT = "#656c75ff";
export const INTERACT_ICON_STRONG_HOVER = "#464c54ff";
export const INTERACT_ICON_STRONG_FOCUS = "#464c54ff";
export const INTERACT_ICON_STRONG_ACTIVE = "#464c54ff";
export const INTERACT_ICON_STRONG_DISABLED = "#a7adb5ff";
export const INTERACT_ICON_IN_DEFAULT = "#ffffffff";
export const INTERACT_ICON_IN_HOVER = "#ffffffff";
export const INTERACT_ICON_IN_FOCUS = "#ffffffff";
export const INTERACT_ICON_IN_ACTIVE = "#ffffffff";
export const INTERACT_ICON_IN_DISABLED = "#ffffffff";
export const INTERACT_BORDER_DEFAULT = "#1d69ccff";
export const INTERACT_BORDER_HOVER = "#0d5cbdff";
export const INTERACT_BORDER_FOCUS = "#0d5cbdff";
export const INTERACT_BORDER_ACTIVE = "#0d5cbdff";
export const INTERACT_BORDER_DISABLED = "#a3c8ffff";
export const INTERACT_BORDER_WEAK_DEFAULT = "#e1e4e8ff";
export const INTERACT_BORDER_WEAK_HOVER = "#cce1ffff";
export const INTERACT_BORDER_WEAK_FOCUS = "#e1e4e8ff";
export const INTERACT_BORDER_WEAK_ACTIVE = "#e1e4e8ff";
export const INTERACT_BORDER_WEAK_DISABLED = "#e1e4e8ff";
export const INTERACT_BORDER_MEDIUM_DEFAULT = "#2774d9ff";
export const INTERACT_BORDER_MEDIUM_HOVER = "#bad6ffff";
export const INTERACT_BORDER_MEDIUM_FOCUS = "#2774d9ff";
export const INTERACT_BORDER_MEDIUM_ACTIVE = "#2774d9ff";
export const INTERACT_BORDER_MEDIUM_DISABLED = "#e1e4e8ff";
export const POSITIVE_BG_DEFAULT = "#45991fff";
export const POSITIVE_BG_HOVER = "#45991fff";
export const POSITIVE_BG_FOCUS = "#45991fff";
export const POSITIVE_BG_ACTIVE = "#45991fff";
export const POSITIVE_BG_DISABLED = "#b0e396ff";
export const POSITIVE_BG_WEAK_DEFAULT = "#ffffffff";
export const POSITIVE_BG_WEAK_HOVER = "#ffffffff";
export const POSITIVE_BG_WEAK_FOCUS = "#ffffffff";
export const POSITIVE_BG_WEAK_ACTIVE = "#ffffffff";
export const POSITIVE_BG_WEAK_DISABLED = "#ffffffff";
export const POSITIVE_BG_MEDIUM_DEFAULT = "#e0f5d5ff";
export const POSITIVE_BG_MEDIUM_HOVER = "#e0f5d5ff";
export const POSITIVE_BG_MEDIUM_FOCUS = "#e0f5d5ff";
export const POSITIVE_BG_MEDIUM_ACTIVE = "#e0f5d5ff";
export const POSITIVE_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const POSITIVE_BG_STRONG_DEFAULT = "#e0f5d5ff";
export const POSITIVE_ICON_DEFAULT = "#45991fff";
export const POSITIVE_ICON_HOVER = "#45991fff";
export const POSITIVE_ICON_FOCUS = "#45991fff";
export const POSITIVE_ICON_ACTIVE = "#45991fff";
export const POSITIVE_ICON_DISABLED = "#d0d4d9ff";
export const POSITIVE_ICON_WEAK_DEFAULT = "#b0e396ff";
export const POSITIVE_ICON_WEAK_HOVER = "#b0e396ff";
export const POSITIVE_ICON_WEAK_FOCUS = "#b0e396ff";
export const POSITIVE_ICON_WEAK_ACTIVE = "#b0e396ff";
export const POSITIVE_ICON_WEAK_DISABLED = "#b0e396ff";
export const POSITIVE_ICON_STRONG_DEFAULT = "#45991fff";
export const POSITIVE_ICON_STRONG_HOVER = "#45991fff";
export const POSITIVE_ICON_STRONG_FOCUS = "#45991fff";
export const POSITIVE_ICON_STRONG_ACTIVE = "#45991fff";
export const POSITIVE_ICON_STRONG_DISABLED = "#d0d4d9ff";
export const POSITIVE_ICON_IN_DEFAULT = "#ffffffff";
export const POSITIVE_ICON_IN_HOVER = "#ffffffff";
export const POSITIVE_ICON_IN_FOCUS = "#ffffffff";
export const POSITIVE_ICON_IN_ACTIVE = "#ffffffff";
export const POSITIVE_ICON_IN_DISABLED = "#ffffffff";
export const POSITIVE_BORDER_DEFAULT = "#45991fff";
export const POSITIVE_BORDER_HOVER = "#45991fff";
export const POSITIVE_BORDER_FOCUS = "#45991fff";
export const POSITIVE_BORDER_ACTIVE = "#45991fff";
export const POSITIVE_BORDER_DISABLED = "#d0d4d9ff";
export const POSITIVE_BORDER_WEAK_DEFAULT = "#ffffffff";
export const POSITIVE_BORDER_WEAK_HOVER = "#ffffffff";
export const POSITIVE_BORDER_WEAK_FOCUS = "#ffffffff";
export const POSITIVE_BORDER_WEAK_ACTIVE = "#ffffffff";
export const POSITIVE_BORDER_WEAK_DISABLED = "#d0d4d9ff";
export const POSITIVE_BORDER_STRONG_DEFAULT = "#398519ff";
export const POSITIVE_BORDER_STRONG_HOVER = "#398519ff";
export const POSITIVE_BORDER_STRONG_FOCUS = "#398519ff";
export const POSITIVE_BORDER_STRONG_ACTIVE = "#398519ff";
export const POSITIVE_BORDER_STRONG_DISABLED = "#398519ff";
export const POSITIVE_TEXT_IN_DEFAULT = "#ffffffff";
export const POSITIVE_TEXT_IN_HOVER = "#ffffffff";
export const POSITIVE_TEXT_IN_FOCUS = "#ffffffff";
export const POSITIVE_TEXT_IN_ACTIVE = "#ffffffff";
export const POSITIVE_TEXT_IN_DISABLED = "#ffffffff";
export const WARNING_BG_DEFAULT = "#cc8604ff";
export const WARNING_BG_HOVER = "#cc8604ff";
export const WARNING_BG_FOCUS = "#cc8604ff";
export const WARNING_BG_ACTIVE = "#cc8604ff";
export const WARNING_BG_DISABLED = "#f2d268ff";
export const WARNING_BG_WEAK_DEFAULT = "#ffffffff";
export const WARNING_BG_WEAK_HOVER = "#ffffffff";
export const WARNING_BG_WEAK_FOCUS = "#ffffffff";
export const WARNING_BG_WEAK_ACTIVE = "#ffffffff";
export const WARNING_BG_WEAK_DISABLED = "#ffffffff";
export const WARNING_BG_MEDIUM_DEFAULT = "#faefb9ff";
export const WARNING_BG_MEDIUM_HOVER = "#faefb9ff";
export const WARNING_BG_MEDIUM_FOCUS = "#faefb9ff";
export const WARNING_BG_MEDIUM_ACTIVE = "#faefb9ff";
export const WARNING_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const WARNING_BG_STRONG_DEFAULT = "#faefb9ff";
export const WARNING_ICON_DEFAULT = "#cc8604ff";
export const WARNING_ICON_HOVER = "#cc8604ff";
export const WARNING_ICON_FOCUS = "#cc8604ff";
export const WARNING_ICON_ACTIVE = "#cc8604ff";
export const WARNING_ICON_DISABLED = "#d0d4d9ff";
export const WARNING_ICON_STRONG_DEFAULT = "#bd7202ff";
export const WARNING_ICON_STRONG_HOVER = "#bd7202ff";
export const WARNING_ICON_STRONG_FOCUS = "#bd7202ff";
export const WARNING_ICON_STRONG_ACTIVE = "#bd7202ff";
export const WARNING_ICON_STRONG_DISABLED = "#d0d4d9ff";
export const WARNING_ICON_IN_DEFAULT = "#ffffffff";
export const WARNING_ICON_IN_HOVER = "#ffffffff";
export const WARNING_ICON_IN_FOCUS = "#ffffffff";
export const WARNING_ICON_IN_ACTIVE = "#ffffffff";
export const WARNING_ICON_IN_DISABLED = "#ffffffff";
export const WARNING_BORDER_DEFAULT = "#cc8604ff";
export const WARNING_BORDER_HOVER = "#cc8604ff";
export const WARNING_BORDER_FOCUS = "#cc8604ff";
export const WARNING_BORDER_ACTIVE = "#cc8604ff";
export const WARNING_BORDER_DISABLED = "#d0d4d9ff";
export const WARNING_BORDER_WEAK_DEFAULT = "#ffffffff";
export const WARNING_BORDER_WEAK_HOVER = "#ffffffff";
export const WARNING_BORDER_WEAK_FOCUS = "#ffffffff";
export const WARNING_BORDER_WEAK_ACTIVE = "#ffffffff";
export const WARNING_BORDER_WEAK_DISABLED = "#d0d4d9ff";
export const WARNING_BORDER_STRONG_DEFAULT = "#a65503ff";
export const WARNING_BORDER_STRONG_HOVER = "#a65503ff";
export const WARNING_BORDER_STRONG_FOCUS = "#a65503ff";
export const WARNING_BORDER_STRONG_ACTIVE = "#a65503ff";
export const WARNING_BORDER_STRONG_DISABLED = "#a65503ff";
export const WARNING_TEXT_IN_DEFAULT = "#ffffffff";
export const WARNING_TEXT_IN_HOVER = "#ffffffff";
export const WARNING_TEXT_IN_FOCUS = "#ffffffff";
export const WARNING_TEXT_IN_ACTIVE = "#ffffffff";
export const WARNING_TEXT_IN_DISABLED = "#ffffffff";
export const SEVERE_WARNING_BG_DEFAULT = "#f26722ff";
export const SEVERE_WARNING_BG_HOVER = "#f26722ff";
export const SEVERE_WARNING_BG_FOCUS = "#f26722ff";
export const SEVERE_WARNING_BG_ACTIVE = "#f26722ff";
export const SEVERE_WARNING_BG_DISABLED = "#fcc9a7ff";
export const SEVERE_WARNING_BG_WEAK_DEFAULT = "#ffffffff";
export const SEVERE_WARNING_BG_WEAK_HOVER = "#ffffffff";
export const SEVERE_WARNING_BG_WEAK_FOCUS = "#ffffffff";
export const SEVERE_WARNING_BG_WEAK_ACTIVE = "#ffffffff";
export const SEVERE_WARNING_BG_WEAK_DISABLED = "#ffffffff";
export const SEVERE_WARNING_BG_MEDIUM_DEFAULT = "#ffeadbff";
export const SEVERE_WARNING_BG_MEDIUM_HOVER = "#ffeadbff";
export const SEVERE_WARNING_BG_MEDIUM_FOCUS = "#ffeadbff";
export const SEVERE_WARNING_BG_MEDIUM_ACTIVE = "#ffeadbff";
export const SEVERE_WARNING_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const SEVERE_WARNING_BG_STRONG_DEFAULT = "#ffeadbff";
export const SEVERE_WARNING_ICON_DEFAULT = "#f26722ff";
export const SEVERE_WARNING_ICON_HOVER = "#d95a1aff";
export const SEVERE_WARNING_ICON_FOCUS = "#c44f14ff";
export const SEVERE_WARNING_ICON_ACTIVE = "#c44f14ff";
export const SEVERE_WARNING_ICON_DISABLED = "#d0d4d9ff";
export const SEVERE_WARNING_ICON_STRONG_DEFAULT = "#d95a1aff";
export const SEVERE_WARNING_ICON_STRONG_HOVER = "#d95a1aff";
export const SEVERE_WARNING_ICON_STRONG_FOCUS = "#d95a1aff";
export const SEVERE_WARNING_ICON_STRONG_ACTIVE = "#d95a1aff";
export const SEVERE_WARNING_ICON_STRONG_DISABLED = "#d95a1aff";
export const SEVERE_WARNING_ICON_IN_DEFAULT = "#ffffffff";
export const SEVERE_WARNING_ICON_IN_HOVER = "#ffffffff";
export const SEVERE_WARNING_ICON_IN_FOCUS = "#ffffffff";
export const SEVERE_WARNING_ICON_IN_ACTIVE = "#ffffffff";
export const SEVERE_WARNING_ICON_IN_DISABLED = "#ffffffff";
export const SEVERE_WARNING_BORDER_DEFAULT = "#f26722ff";
export const SEVERE_WARNING_BORDER_HOVER = "#f26722ff";
export const SEVERE_WARNING_BORDER_FOCUS = "#f26722ff";
export const SEVERE_WARNING_BORDER_ACTIVE = "#f26722ff";
export const SEVERE_WARNING_BORDER_DISABLED = "#f26722ff";
export const SEVERE_WARNING_BORDER_WEAK_DEFAULT = "#ffffffff";
export const SEVERE_WARNING_BORDER_WEAK_HOVER = "#ffffffff";
export const SEVERE_WARNING_BORDER_WEAK_FOCUS = "#ffffffff";
export const SEVERE_WARNING_BORDER_WEAK_ACTIVE = "#ffffffff";
export const SEVERE_WARNING_BORDER_WEAK_DISABLED = "#d0d4d9ff";
export const NEGATIVE_BG_DEFAULT = "#cc2d37ff";
export const NEGATIVE_BG_HOVER = "#cc2d37ff";
export const NEGATIVE_BG_FOCUS = "#cc2d37ff";
export const NEGATIVE_BG_ACTIVE = "#cc2d37ff";
export const NEGATIVE_BG_DISABLED = "#ffb2b5ff";
export const NEGATIVE_BG_WEAK_DEFAULT = "#ffffffff";
export const NEGATIVE_BG_WEAK_HOVER = "#ffd4d5ff";
export const NEGATIVE_BG_WEAK_FOCUS = "#ffd4d5ff";
export const NEGATIVE_BG_WEAK_ACTIVE = "#ffd4d5ff";
export const NEGATIVE_BG_WEAK_DISABLED = "#ffffffff";
export const NEGATIVE_BG_MEDIUM_DEFAULT = "#ffe8e9ff";
export const NEGATIVE_BG_MEDIUM_HOVER = "#ffe8e9ff";
export const NEGATIVE_BG_MEDIUM_FOCUS = "#ffe8e9ff";
export const NEGATIVE_BG_MEDIUM_ACTIVE = "#ffe8e9ff";
export const NEGATIVE_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const NEGATIVE_BG_STRONG_DEFAULT = "#ffe8e9ff";
export const NEGATIVE_TEXT_DEFAULT = "#cc2d37ff";
export const NEGATIVE_TEXT_HOVER = "#b2242dff";
export const NEGATIVE_TEXT_FOCUS = "#b2242dff";
export const NEGATIVE_TEXT_ACTIVE = "#b2242dff";
export const NEGATIVE_TEXT_DISABLED = "#ffb2b5ff";
export const NEGATIVE_TEXT_IN_DEFAULT = "#ffffffff";
export const NEGATIVE_TEXT_IN_HOVER = "#ffffffff";
export const NEGATIVE_TEXT_IN_FOCUS = "#ffffffff";
export const NEGATIVE_TEXT_IN_ACTIVE = "#ffffffff";
export const NEGATIVE_TEXT_IN_DISABLED = "#ffffffff";
export const NEGATIVE_ICON_DEFAULT = "#cc2d37ff";
export const NEGATIVE_ICON_HOVER = "#b2242dff";
export const NEGATIVE_ICON_FOCUS = "#b2242dff";
export const NEGATIVE_ICON_ACTIVE = "#b2242dff";
export const NEGATIVE_ICON_DISABLED = "#ffb2b5ff";
export const NEGATIVE_ICON_WEAK_DEFAULT = "#ffc7c9ff";
export const NEGATIVE_ICON_WEAK_HOVER = "#ffc7c9ff";
export const NEGATIVE_ICON_WEAK_FOCUS = "#ffc7c9ff";
export const NEGATIVE_ICON_WEAK_ACTIVE = "#ffc7c9ff";
export const NEGATIVE_ICON_WEAK_DISABLED = "#d0d4d9ff";
export const NEGATIVE_ICON_IN_DEFAULT = "#ffffffff";
export const NEGATIVE_ICON_IN_HOVER = "#ffffffff";
export const NEGATIVE_ICON_IN_FOCUS = "#ffffffff";
export const NEGATIVE_ICON_IN_ACTIVE = "#ffffffff";
export const NEGATIVE_ICON_IN_DISABLED = "#ffffffff";
export const NEGATIVE_BORDER_DEFAULT = "#cc2d37ff";
export const NEGATIVE_BORDER_HOVER = "#b2242dff";
export const NEGATIVE_BORDER_FOCUS = "#b2242dff";
export const NEGATIVE_BORDER_ACTIVE = "#b2242dff";
export const NEGATIVE_BORDER_DISABLED = "#ffb2b5ff";
export const NEGATIVE_BORDER_WEAK_DEFAULT = "#ffffffff";
export const NEGATIVE_BORDER_WEAK_HOVER = "#ffffffff";
export const NEGATIVE_BORDER_WEAK_FOCUS = "#ffffffff";
export const NEGATIVE_BORDER_WEAK_ACTIVE = "#ffffffff";
export const NEGATIVE_BORDER_WEAK_DISABLED = "#d0d4d9ff";
export const NEGATIVE_BORDER_MEDIUM_DEFAULT = "#d93843ff";
export const NEGATIVE_BORDER_MEDIUM_HOVER = "#d93843ff";
export const NEGATIVE_BORDER_MEDIUM_FOCUS = "#d93843ff";
export const NEGATIVE_BORDER_MEDIUM_ACTIVE = "#d93843ff";
export const NEGATIVE_BORDER_MEDIUM_DISABLED = "#ffb2b5ff";
export const NEGATIVE_BORDER_STRONG_DEFAULT = "#b2242dff";
export const NEGATIVE_BORDER_STRONG_HOVER = "#b2242dff";
export const NEGATIVE_BORDER_STRONG_FOCUS = "#b2242dff";
export const NEGATIVE_BORDER_STRONG_ACTIVE = "#b2242dff";
export const NEGATIVE_BORDER_STRONG_DISABLED = "#ffb2b5ff";
export const INFO_BG_DEFAULT = "#2774d9ff";
export const INFO_BG_HOVER = "#2774d9ff";
export const INFO_BG_FOCUS = "#2774d9ff";
export const INFO_BG_ACTIVE = "#2774d9ff";
export const INFO_BG_DISABLED = "#2774d9ff";
export const INFO_BG_WEAK_DEFAULT = "#ffffffff";
export const INFO_BG_WEAK_HOVER = "#ffffffff";
export const INFO_BG_WEAK_FOCUS = "#ffffffff";
export const INFO_BG_WEAK_ACTIVE = "#ffffffff";
export const INFO_BG_WEAK_DISABLED = "#ffffffff";
export const INFO_BG_MEDIUM_DEFAULT = "#e3eeffff";
export const INFO_BG_MEDIUM_HOVER = "#e3eeffff";
export const INFO_BG_MEDIUM_FOCUS = "#e3eeffff";
export const INFO_BG_MEDIUM_ACTIVE = "#e3eeffff";
export const INFO_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const INFO_BG_STRONG_DEFAULT = "#e3eeffff";
export const INFO_ICON_DEFAULT = "#2774d9ff";
export const INFO_ICON_HOVER = "#2774d9ff";
export const INFO_ICON_FOCUS = "#2774d9ff";
export const INFO_ICON_ACTIVE = "#2774d9ff";
export const INFO_ICON_DISABLED = "#d0d4d9ff";
export const INFO_ICON_WEAK_DEFAULT = "#a3c8ffff";
export const INFO_ICON_WEAK_HOVER = "#a3c8ffff";
export const INFO_ICON_WEAK_FOCUS = "#a3c8ffff";
export const INFO_ICON_WEAK_ACTIVE = "#a3c8ffff";
export const INFO_ICON_WEAK_DISABLED = "#a3c8ffff";
export const INFO_ICON_IN_DEFAULT = "#ffffffff";
export const INFO_ICON_IN_HOVER = "#ffffffff";
export const INFO_ICON_IN_FOCUS = "#ffffffff";
export const INFO_ICON_IN_ACTIVE = "#ffffffff";
export const INFO_ICON_IN_DISABLED = "#ffffffff";
export const INFO_BORDER_DEFAULT = "#2774d9ff";
export const INFO_BORDER_HOVER = "#2774d9ff";
export const INFO_BORDER_FOCUS = "#2774d9ff";
export const INFO_BORDER_ACTIVE = "#2774d9ff";
export const INFO_BORDER_DISABLED = "#d0d4d9ff";
export const INFO_BORDER_WEAK_DEFAULT = "#ffffffff";
export const INFO_BORDER_WEAK_HOVER = "#ffffffff";
export const INFO_BORDER_WEAK_FOCUS = "#ffffffff";
export const INFO_BORDER_WEAK_ACTIVE = "#ffffffff";
export const INFO_BORDER_WEAK_DISABLED = "#d0d4d9ff";
export const DORMANT_BG_DEFAULT = "#6f7680ff";
export const DORMANT_BG_DISABLED = "#6f7680ff";
export const DORMANT_BG_MEDIUM_DEFAULT = "#e1e4e8ff";
export const DORMANT_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const DORMANT_ICON_DEFAULT = "#6f7680ff";
export const DORMANT_ICON_DISABLED = "#d0d4d9ff";
export const DORMANT_ICON_MEDIUM_DEFAULT = "#ffffffff";
export const DORMANT_ICON_IN_DEFAULT = "#ffffffff";
export const DORMANT_ICON_IN_DISABLED = "#ffffffff";
export const DORMANT_ICON_IN_MEDIUM_DEFAULT = "#ffffffff";
export const DORMANT_BORDER_DEFAULT = "#889099ff";
export const DORMANT_BORDER_DISABLED = "#d0d4d9ff";
export const ACCENT_A_DEFAULT = "#7d8affff";
export const ACCENT_A_HOVER = "#6977f0ff";
export const ACCENT_A_WEAK_DEFAULT = "#ebedffff";
export const ACCENT_B_DEFAULT = "#b02863ff";
export const ACCENT_B_HOVER = "#991d53ff";
export const ACCENT_B_WEAK_DEFAULT = "#ffe8efff";
export const ACCENT_C_DEFAULT = "#f2638cff";
export const ACCENT_C_HOVER = "#e3447cff";
export const ACCENT_C_WEAK_DEFAULT = "#ffd4e0ff";
export const ACCENT_D_DEFAULT = "#753bccff";
export const ACCENT_D_HOVER = "#6732b8ff";
export const ACCENT_D_WEAK_DEFAULT = "#f3ebffff";
export const ACCENT_E_DEFAULT = "#7da11bff";
export const ACCENT_E_HOVER = "#6c8c14ff";
export const ACCENT_E_WEAK_DEFAULT = "#eaf2d3ff";
export const ACCENT_F_DEFAULT = "#ad3907ff";
export const ACCENT_F_HOVER = "#942e03ff";
export const ACCENT_F_WEAK_DEFAULT = "#ffeadbff";
export const ACCENT_G_DEFAULT = "#04a4b0ff";
export const ACCENT_G_HOVER = "#028e99ff";
export const ACCENT_G_WEAK_DEFAULT = "#d5f5f5ff";
export const ACCENT_H_DEFAULT = "#006773ff";
export const ACCENT_H_HOVER = "#005c66ff";
export const ACCENT_H_WEAK_DEFAULT = "#a9ebebff";
export const ACCENT_I_DEFAULT = "#e85fc6ff";
export const ACCENT_I_HOVER = "#d649b3ff";
export const ACCENT_I_WEAK_DEFAULT = "#ffe8f9ff";
export const ACCENT_J_DEFAULT = "#545c8aff";
export const ACCENT_J_HOVER = "#484f7aff";
export const ACCENT_J_WEAK_DEFAULT = "#ebedffff";
export const ACCENT_K_DEFAULT = "#21a65fff";
export const ACCENT_K_HOVER = "#169855ff";
export const ACCENT_K_WEAK_DEFAULT = "#d4f5e1ff";
export const IN_PROGRESS_BG_MEDIUM_DEFAULT = "#e3eeffff";
export const IN_PROGRESS_BG_MEDIUM_HOVER = "#e3eeffff";
export const IN_PROGRESS_BG_MEDIUM_FOCUS = "#e3eeffff";
export const IN_PROGRESS_BG_MEDIUM_ACTIVE = "#e3eeffff";
export const IN_PROGRESS_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const IN_PROGRESS_ICON_DEFAULT = "#2774d9ff";
export const IN_PROGRESS_ICON_HOVER = "#2774d9ff";
export const IN_PROGRESS_ICON_FOCUS = "#2774d9ff";
export const IN_PROGRESS_ICON_ACTIVE = "#2774d9ff";
export const IN_PROGRESS_ICON_DISABLED = "#d0d4d9ff";
export const IN_PROGRESS_ICON_IN_DEFAULT = "#ffffffff";
export const IN_PROGRESS_ICON_IN_HOVER = "#ffffffff";
export const IN_PROGRESS_ICON_IN_FOCUS = "#ffffffff";
export const IN_PROGRESS_ICON_IN_ACTIVE = "#ffffffff";
export const IN_PROGRESS_ICON_IN_DISABLED = "#ffffffff";
export const IN_PROGRESS_BORDER_DEFAULT = "#2774d9ff";
export const IN_PROGRESS_BORDER_HOVER = "#2774d9ff";
export const IN_PROGRESS_BORDER_FOCUS = "#2774d9ff";
export const IN_PROGRESS_BORDER_ACTIVE = "#2774d9ff";
export const IN_PROGRESS_BORDER_DISABLED = "#d0d4d9ff";
export const DECENT_BG_MEDIUM_DEFAULT = "#eaf2d3ff";
export const DECENT_BG_MEDIUM_HOVER = "#eaf2d3ff";
export const DECENT_BG_MEDIUM_FOCUS = "#eaf2d3ff";
export const DECENT_BG_MEDIUM_ACTIVE = "#eaf2d3ff";
export const DECENT_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const DECENT_ICON_DEFAULT = "#6c8c14ff";
export const DECENT_ICON_HOVER = "#6c8c14ff";
export const DECENT_ICON_FOCUS = "#6c8c14ff";
export const DECENT_ICON_ACTIVE = "#6c8c14ff";
export const DECENT_ICON_DISABLED = "#d0d4d9ff";
export const DECENT_ICON_IN_DEFAULT = "#ffffffff";
export const DECENT_ICON_IN_HOVER = "#ffffffff";
export const DECENT_ICON_IN_FOCUS = "#ffffffff";
export const DECENT_ICON_IN_ACTIVE = "#ffffffff";
export const DECENT_ICON_IN_DISABLED = "#ffffffff";
export const DECENT_BORDER_DEFAULT = "#7da11bff";
export const DECENT_BORDER_HOVER = "#7da11bff";
export const DECENT_BORDER_FOCUS = "#7da11bff";
export const DECENT_BORDER_ACTIVE = "#7da11bff";
export const DECENT_BORDER_DISABLED = "#d0d4d9ff";
export const EXCELLENT_BG_MEDIUM_DEFAULT = "#d4f5e1ff";
export const EXCELLENT_BG_MEDIUM_HOVER = "#d4f5e1ff";
export const EXCELLENT_BG_MEDIUM_FOCUS = "#d4f5e1ff";
export const EXCELLENT_BG_MEDIUM_ACTIVE = "#d4f5e1ff";
export const EXCELLENT_BG_MEDIUM_DISABLED = "#f7f7f7ff";
export const EXCELLENT_ICON_DEFAULT = "#087041ff";
export const EXCELLENT_ICON_HOVER = "#087041ff";
export const EXCELLENT_ICON_FOCUS = "#087041ff";
export const EXCELLENT_ICON_ACTIVE = "#087041ff";
export const EXCELLENT_ICON_DISABLED = "#d0d4d9ff";
export const EXCELLENT_ICON_IN_DEFAULT = "#ffffffff";
export const EXCELLENT_ICON_IN_HOVER = "#ffffffff";
export const EXCELLENT_ICON_IN_FOCUS = "#ffffffff";
export const EXCELLENT_ICON_IN_ACTIVE = "#ffffffff";
export const EXCELLENT_ICON_IN_DISABLED = "#ffffffff";
export const EXCELLENT_BORDER_DEFAULT = "#075e39ff";
export const EXCELLENT_BORDER_HOVER = "#075e39ff";
export const EXCELLENT_BORDER_FOCUS = "#075e39ff";
export const EXCELLENT_BORDER_ACTIVE = "#075e39ff";
export const EXCELLENT_BORDER_DISABLED = "#d0d4d9ff";
