// @ciscodesignsystems/cds-magnetic-theme-web/js/dist/token-theme-dark-es6.js
/**
 * This file is auto-generated. Do not edit it directly.
 */

export const BRAND_BG_DEFAULT = "#373c42";
export const BRAND_BG_HOVER = "#373c42";
export const BRAND_BG_FOCUS = "#464c54";
export const BRAND_BG_ACTIVE = "#464c54";
export const BRAND_BG_DISABLED = "#e1e4e8";
export const BRAND_BG_GRADIENT_START = "#373c42";
export const BRAND_BG_GRADIENT_END = "#0f1214";
export const BRAND_BG_STRONG_DEFAULT = "#373c42";
export const BRAND_BG_STRONG_HOVER = "#373c42";
export const BRAND_BG_STRONG_FOCUS = "#373c42";
export const BRAND_BG_STRONG_ACTIVE = "#373c42";
export const BRAND_TEXT_DEFAULT = "#f7f7f7";
export const BRAND_TEXT_HOVER = "#f7f7f7";
export const BRAND_TEXT_ACTIVE = "#f7f7f7";
export const BRAND_TEXT_WEAK_DEFAULT = "#889099";
export const BRAND_TEXT_WEAK_HOVER = "#f7f7f7";
export const BRAND_TEXT_WEAK_FOCUS = "#f7f7f7";
export const BRAND_TEXT_WEAK_ACTIVE = "#f7f7f7";
export const BRAND_TEXT_STRONG_DEFAULT = "#f0f1f2";
export const BRAND_ICON_DEFAULT = "#f7f7f7";
export const BRAND_ICON_HOVER = "#ffffff";
export const BRAND_ICON_ACTIVE = "#ffffff";
export const BRAND_ICON_WEAK_DEFAULT = "#f7f7f7";
export const BRAND_ICON_WEAK_HOVER = "#f7f7f7";
export const BRAND_ICON_WEAK_FOCUS = "#f7f7f7";
export const BRAND_ICON_WEAK_ACTIVE = "#f7f7f7";
export const BRAND_BORDER_ACTIVE = "#cce1ff";
export const BRAND_ACCENT_ICON_DEFAULT = "#c1c6cc";
export const BRAND_ACCENT_ICON_FOCUS = "#ffffff";
export const BRAND_ACCENT_ICON_ACTIVE = "#6bbf41";
export const BRAND_ACCENT_ICON_DISABLED = "#6f7680";
export const BRAND_ACCENT_ICON_WEAK_DEFAULT = "#6f7680";
export const BRAND_ACCENT_ICON_WEAK_FOCUS = "#ffffff";
export const BRAND_ACCENT_ICON_WEAK_ACTIVE = "#398519";
export const BRAND_ACCENT_ICON_WEAK_DISABLED = "#ffffff";
export const BRAND_ACCENT_ICON_MEDIUM_DEFAULT = "#979fa8";
export const BRAND_ACCENT_ICON_MEDIUM_FOCUS = "#ffffff";
export const BRAND_ACCENT_ICON_MEDIUM_ACTIVE = "#52a62b";
export const BRAND_ACCENT_ICON_MEDIUM_DISABLED = "#ffffff";
export const BRAND_ACCENT_BORDER_DEFAULT = "#52a62b";
export const BRAND_ACCENT_BORDER_HOVER = "#7e868f";
export const BRAND_ACCENT_BORDER_FOCUS = "#52a62b";
export const BRAND_ACCENT_BORDER_ACTIVE = "#52a62b";
export const BRAND_ACCENT_BORDER_WEAK_DEFAULT = "#d0d4d9";
export const BRAND_ACCENT_BORDER_WEAK_HOVER = "#ffffff";
export const BRAND_ACCENT_BORDER_WEAK_FOCUS = "#ffffff";
export const BRAND_ACCENT_BORDER_WEAK_ACTIVE = "#398519";
export const BRAND_ACCENT_BORDER_WEAK_DISABLED = "#ffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_DEFAULT = "#596069";
export const BRAND_ACCENT_BORDER_MEDIUM_HOVER = "#ffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_FOCUS = "#ffffff";
export const BRAND_ACCENT_BORDER_MEDIUM_ACTIVE = "#2b7a0c";
export const BRAND_ACCENT_BORDER_MEDIUM_DISABLED = "#ffffff";
export const BRAND_ACCENT_BORDER_STRONG_DEFAULT = "#0f1214";
export const BRAND_ACCENT_BORDER_STRONG_HOVER = "#ffffff";
export const BRAND_ACCENT_BORDER_STRONG_FOCUS = "#ffffff";
export const BRAND_ACCENT_BORDER_STRONG_ACTIVE = "#0f1214";
export const BRAND_ACCENT_BORDER_STRONG_DISABLED = "#ffffff";
export const BASE_BG_DEFAULT = "#0f1214";
export const BASE_BG_HOVER = "#0f1214";
export const BASE_BG_ACTIVE = "#0f1214";
export const BASE_BG_DISABLED = "#464c54";
export const BASE_BG_WEAK_DEFAULT = "#23282e";
export const BASE_BG_WEAK_HOVER = "#373c42";
export const BASE_BG_WEAK_ACTIVE = "#23282e";
export const BASE_BG_WEAK_DISABLED = "#464c54";
export const BASE_BG_MEDIUM_DEFAULT = "#373c42";
export const BASE_BG_MEDIUM_HOVER = "#373c42";
export const BASE_BG_MEDIUM_ACTIVE = "#23282e";
export const BASE_BG_MEDIUM_DISABLED = "#e85fc6";
export const BASE_BG_STRONG_DEFAULT = "#a7adb5";
export const BASE_BG_STRONG_HOVER = "#d0d4d9";
export const BASE_BG_STRONG_FOCUS = "#d0d4d9";
export const BASE_BG_STRONG_ACTIVE = "#d0d4d9";
export const BASE_BG_STRONG_DISABLED = "#596069";
export const BASE_ICON_IN_DEFAULT = "#23282e";
export const BASE_ICON_IN_HOVER = "#23282e";
export const BASE_ICON_IN_FOCUS = "#23282e";
export const BASE_ICON_IN_ACTIVE = "#23282e";
export const BASE_ICON_IN_DISABLED = "#6f7680";
export const BASE_TEXT_DEFAULT = "#f7f7f7";
export const BASE_TEXT_DISABLED = "#6f7680";
export const BASE_TEXT_WEAK_DEFAULT = "#889099";
export const BASE_TEXT_WEAK_DISABLED = "#6f7680";
export const BASE_TEXT_MEDIUM_DEFAULT = "#d0d4d9";
export const BASE_TEXT_MEDIUM_DISABLED = "#6f7680";
export const BASE_TEXT_STRONG_DEFAULT = "#f0f1f2";
export const BASE_TEXT_STRONG_DISABLED = "#6f7680";
export const BASE_TEXT_IN_DEFAULT = "#23282e";
export const BASE_TEXT_IN_HOVER = "#23282e";
export const BASE_TEXT_IN_FOCUS = "#23282e";
export const BASE_TEXT_IN_ACTIVE = "#23282e";
export const BASE_TEXT_IN_DISABLED = "#23282e";
export const BASE_TEXT_IN_WEAK_DEFAULT = "#ffffff";
export const BASE_BORDER_DEFAULT = "#596069";
export const BASE_BORDER_WEAK_DEFAULT = "#23282e";
export const BASE_BORDER_MEDIUM_DEFAULT = "#464c54";
export const BASE_BORDER_STRONG_DEFAULT = "#c1c6cc";
export const INVERSE_BG_DEFAULT = "#373c42";
export const INVERSE_BG_HOVER = "#373c42";
export const INVERSE_BG_FOCUS = "#464c54";
export const INVERSE_BG_ACTIVE = "#464c54";
export const INVERSE_BG_DISABLED = "#e1e4e8";
export const INVERSE_BG_GRADIENT_START = "#373c42";
export const INVERSE_BG_GRADIENT_END = "#0f1214";
export const INVERSE_BG_WEAK_DEFAULT = "#464c54";
export const INVERSE_TEXT_DEFAULT = "#f7f7f7";
export const INVERSE_TEXT_HOVER = "#f7f7f7";
export const INVERSE_TEXT_FOCUS = "#f7f7f7";
export const INVERSE_TEXT_ACTIVE = "#f7f7f7";
export const INVERSE_TEXT_DISABLED = "#656c75";
export const INVERSE_TEXT_MEDIUM_DEFAULT = "#c1c6cc";
export const INVERSE_TEXT_MEDIUM_DISABLED = "#656c75";
export const INVERSE_ICON_DEFAULT = "#f7f7f7";
export const INVERSE_ICON_HOVER = "#ffffff";
export const INVERSE_ICON_FOCUS = "#ffffff";
export const INVERSE_ICON_ACTIVE = "#ffffff";
export const INVERSE_ICON_DISABLED = "#6f7680";
export const INVERSE_ICON_STRONG_DEFAULT = "#ffffff";
export const INVERSE_ICON_STRONG_HOVER = "#ffffff";
export const INVERSE_ICON_STRONG_FOCUS = "#ffffff";
export const INVERSE_ICON_STRONG_ACTIVE = "#ffffff";
export const INVERSE_ICON_STRONG_DISABLED = "#6f7680";
export const INVERSE_BORDER_DEFAULT = "#6f7680";
export const INVERSE_BORDER_HOVER = "#649ef5";
export const INVERSE_BORDER_FOCUS = "#cce1ff";
export const INVERSE_BORDER_ACTIVE = "#cce1ff";
export const INVERSE_BORDER_DISABLED = "#e1e4e8";
export const CONTROL_BG_DEFAULT = "#23282e";
export const CONTROL_BG_HOVER = "#2a3442";
export const CONTROL_BG_FOCUS = "#2a3442";
export const CONTROL_BG_ACTIVE = "#2a3442";
export const CONTROL_BG_DISABLED = "#373c42";
export const CONTROL_BG_WEAK_DEFAULT = "#23282e";
export const CONTROL_BG_WEAK_HOVER = "#373c42";
export const CONTROL_BG_WEAK_FOCUS = "#23282e";
export const CONTROL_BG_WEAK_ACTIVE = "#2a3442";
export const CONTROL_BG_WEAK_DISABLED = "#373c42";
export const CONTROL_BG_MEDIUM_DEFAULT = "#23282e";
export const CONTROL_BG_MEDIUM_HOVER = "#373c42";
export const CONTROL_BG_MEDIUM_FOCUS = "#979fa8";
export const CONTROL_BG_MEDIUM_ACTIVE = "#2a3442";
export const CONTROL_BG_MEDIUM_DISABLED = "#373c42";
export const CONTROL_BG_STRONG_DEFAULT = "#649ef5";
export const CONTROL_BG_STRONG_HOVER = "#7cadf7";
export const CONTROL_BG_STRONG_FOCUS = "#7cadf7";
export const CONTROL_BG_STRONG_ACTIVE = "#7cadf7";
export const CONTROL_BG_STRONG_DISABLED = "#596069";
export const CONTROL_ICON_DEFAULT = "#649ef5";
export const CONTROL_ICON_HOVER = "#7cadf7";
export const CONTROL_ICON_FOCUS = "#7cadf7";
export const CONTROL_ICON_ACTIVE = "#7cadf7";
export const CONTROL_ICON_DISABLED = "#cce1ff";
export const CONTROL_ICON_WEAK_DEFAULT = "#c1c6cc";
export const CONTROL_ICON_WEAK_HOVER = "#649ef5";
export const CONTROL_ICON_WEAK_FOCUS = "#649ef5";
export const CONTROL_ICON_WEAK_ACTIVE = "#649ef5";
export const CONTROL_ICON_WEAK_DISABLED = "#6f7680";
export const CONTROL_ICON_MEDIUM_DEFAULT = "#649ef5";
export const CONTROL_ICON_MEDIUM_HOVER = "#979fa8";
export const CONTROL_ICON_MEDIUM_FOCUS = "#979fa8";
export const CONTROL_ICON_MEDIUM_ACTIVE = "#889099";
export const CONTROL_ICON_MEDIUM_DISABLED = "#e1e4e8";
export const CONTROL_ICON_STRONG_DEFAULT = "#d0d4d9";
export const CONTROL_ICON_STRONG_HOVER = "#f0f1f2";
export const CONTROL_ICON_STRONG_FOCUS = "#649ef5";
export const CONTROL_ICON_STRONG_ACTIVE = "#649ef5";
export const CONTROL_ICON_STRONG_DISABLED = "#6f7680";
export const CONTROL_ICON_IN_DEFAULT = "#23282e";
export const CONTROL_ICON_IN_HOVER = "#23282e";
export const CONTROL_ICON_IN_FOCUS = "#23282e";
export const CONTROL_ICON_IN_ACTIVE = "#23282e";
export const CONTROL_ICON_IN_DISABLED = "#6f7680";
export const CONTROL_ICON_IN_STRONG_DEFAULT = "#649ef5";
export const CONTROL_ICON_IN_STRONG_HOVER = "#7cadf7";
export const CONTROL_ICON_IN_STRONG_FOCUS = "#7cadf7";
export const CONTROL_ICON_IN_STRONG_ACTIVE = "#7cadf7";
export const CONTROL_ICON_IN_STRONG_DISABLED = "#6f7680";
export const CONTROL_BORDER_DEFAULT = "#7e868f";
export const CONTROL_BORDER_HOVER = "#7cadf7";
export const CONTROL_BORDER_FOCUS = "#7cadf7";
export const CONTROL_BORDER_ACTIVE = "#7cadf7";
export const CONTROL_BORDER_DISABLED = "#596069";
export const CONTROL_BORDER_WEAK_DEFAULT = "#656c75";
export const CONTROL_BORDER_WEAK_HOVER = "#7cadf7";
export const CONTROL_BORDER_WEAK_FOCUS = "#7cadf7";
export const CONTROL_BORDER_WEAK_ACTIVE = "#7cadf7";
export const CONTROL_BORDER_WEAK_DISABLED = "#596069";
export const CONTROL_BORDER_MEDIUM_DEFAULT = "#649ef5";
export const CONTROL_BORDER_MEDIUM_HOVER = "#649ef5";
export const CONTROL_BORDER_MEDIUM_FOCUS = "#649ef5";
export const CONTROL_BORDER_MEDIUM_ACTIVE = "#649ef5";
export const CONTROL_BORDER_MEDIUM_DISABLED = "#596069";
export const CONTROL_BORDER_STRONG_DEFAULT = "#c1c6cc";
export const CONTROL_BORDER_STRONG_HOVER = "#656c75";
export const CONTROL_BORDER_STRONG_FOCUS = "#656c75";
export const CONTROL_BORDER_STRONG_ACTIVE = "#649ef5";
export const CONTROL_BORDER_STRONG_DISABLED = "#596069";
export const INTERACT_BG_DEFAULT = "#649ef5";
export const INTERACT_BG_HOVER = "#7cadf7";
export const INTERACT_BG_FOCUS = "#7cadf7";
export const INTERACT_BG_ACTIVE = "#7cadf7";
export const INTERACT_BG_DISABLED = "#446392";
export const INTERACT_BG_WEAK_DEFAULT = "#23282e";
export const INTERACT_BG_WEAK_HOVER = "#2f3d52";
export const INTERACT_BG_WEAK_FOCUS = "#2f3d52";
export const INTERACT_BG_WEAK_ACTIVE = "#2f3d52";
export const INTERACT_BG_WEAK_DISABLED = "#464c54";
export const INTERACT_BG_MEDIUM_DEFAULT = "#649ef5";
export const INTERACT_BG_MEDIUM_HOVER = "#3a5276";
export const INTERACT_BG_MEDIUM_ACTIVE = "#3a5276";
export const INTERACT_BG_STRONG_HOVER = "#596069";
export const INTERACT_BG_STRONG_ACTIVE = "#596069";
export const INTERACT_TEXT_DEFAULT = "#649ef5";
export const INTERACT_TEXT_HOVER = "#7cadf7";
export const INTERACT_TEXT_FOCUS = "#7cadf7";
export const INTERACT_TEXT_ACTIVE = "#7cadf7";
export const INTERACT_TEXT_DISABLED = "#446392";
export const INTERACT_TEXT_WEAK_DEFAULT = "#c1c6cc";
export const INTERACT_TEXT_WEAK_HOVER = "#649ef5";
export const INTERACT_TEXT_WEAK_FOCUS = "#649ef5";
export const INTERACT_TEXT_WEAK_ACTIVE = "#649ef5";
export const INTERACT_TEXT_WEAK_DISABLED = "#6f7680";
export const INTERACT_TEXT_IN_DEFAULT = "#23282e";
export const INTERACT_TEXT_IN_HOVER = "#23282e";
export const INTERACT_TEXT_IN_FOCUS = "#23282e";
export const INTERACT_TEXT_IN_ACTIVE = "#23282e";
export const INTERACT_TEXT_IN_DISABLED = "#23282e";
export const INTERACT_ICON_DEFAULT = "#649ef5";
export const INTERACT_ICON_HOVER = "#7cadf7";
export const INTERACT_ICON_FOCUS = "#7cadf7";
export const INTERACT_ICON_ACTIVE = "#7cadf7";
export const INTERACT_ICON_DISABLED = "#446392";
export const INTERACT_ICON_WEAK_DEFAULT = "#c1c6cc";
export const INTERACT_ICON_WEAK_HOVER = "#d0d4d9";
export const INTERACT_ICON_WEAK_FOCUS = "#d0d4d9";
export const INTERACT_ICON_WEAK_ACTIVE = "#d0d4d9";
export const INTERACT_ICON_WEAK_DISABLED = "#6f7680";
export const INTERACT_ICON_MEDIUM_DEFAULT = "#c1c6cc";
export const INTERACT_ICON_MEDIUM_HOVER = "#649ef5";
export const INTERACT_ICON_MEDIUM_FOCUS = "#649ef5";
export const INTERACT_ICON_MEDIUM_ACTIVE = "#649ef5";
export const INTERACT_ICON_MEDIUM_DISABLED = "#6f7680";
export const INTERACT_ICON_STRONG_DEFAULT = "#d0d4d9";
export const INTERACT_ICON_STRONG_HOVER = "#f0f1f2";
export const INTERACT_ICON_STRONG_FOCUS = "#f0f1f2";
export const INTERACT_ICON_STRONG_ACTIVE = "#f0f1f2";
export const INTERACT_ICON_STRONG_DISABLED = "#6f7680";
export const INTERACT_ICON_IN_DEFAULT = "#23282e";
export const INTERACT_ICON_IN_HOVER = "#23282e";
export const INTERACT_ICON_IN_FOCUS = "#23282e";
export const INTERACT_ICON_IN_ACTIVE = "#23282e";
export const INTERACT_ICON_IN_DISABLED = "#23282e";
export const INTERACT_BORDER_DEFAULT = "#649ef5";
export const INTERACT_BORDER_HOVER = "#7cadf7";
export const INTERACT_BORDER_FOCUS = "#7cadf7";
export const INTERACT_BORDER_ACTIVE = "#7cadf7";
export const INTERACT_BORDER_DISABLED = "#446392";
export const INTERACT_BORDER_WEAK_DEFAULT = "#656c75";
export const INTERACT_BORDER_WEAK_HOVER = "#cce1ff";
export const INTERACT_BORDER_WEAK_FOCUS = "#e1e4e8";
export const INTERACT_BORDER_WEAK_ACTIVE = "#e1e4e8";
export const INTERACT_BORDER_WEAK_DISABLED = "#596069";
export const INTERACT_BORDER_MEDIUM_DEFAULT = "#649ef5";
export const INTERACT_BORDER_MEDIUM_HOVER = "#a3c8ff";
export const INTERACT_BORDER_MEDIUM_FOCUS = "#649ef5";
export const INTERACT_BORDER_MEDIUM_ACTIVE = "#649ef5";
export const INTERACT_BORDER_MEDIUM_DISABLED = "#e1e4e8";
export const POSITIVE_BG_DEFAULT = "#33bbf5";
export const POSITIVE_BG_HOVER = "#33bbf5";
export const POSITIVE_BG_FOCUS = "#33bbf5";
export const POSITIVE_BG_ACTIVE = "#33bbf5";
export const POSITIVE_BG_DISABLED = "#33bbf5";
export const POSITIVE_BG_WEAK_DEFAULT = "#23282e";
export const POSITIVE_BG_WEAK_HOVER = "#23282e";
export const POSITIVE_BG_WEAK_FOCUS = "#23282e";
export const POSITIVE_BG_WEAK_ACTIVE = "#23282e";
export const POSITIVE_BG_WEAK_DISABLED = "#23282e";
export const POSITIVE_BG_MEDIUM_DEFAULT = "#28546a";
export const POSITIVE_BG_MEDIUM_HOVER = "#28546a";
export const POSITIVE_BG_MEDIUM_FOCUS = "#28546a";
export const POSITIVE_BG_MEDIUM_ACTIVE = "#28546a";
export const POSITIVE_BG_MEDIUM_DISABLED = "#373c42";
export const POSITIVE_BG_STRONG_DEFAULT = "#173746";
export const POSITIVE_ICON_DEFAULT = "#33bbf5";
export const POSITIVE_ICON_HOVER = "#33bbf5";
export const POSITIVE_ICON_FOCUS = "#33bbf5";
export const POSITIVE_ICON_ACTIVE = "#33bbf5";
export const POSITIVE_ICON_DISABLED = "#596069";
export const POSITIVE_ICON_WEAK_DEFAULT = "#6fd2fc";
export const POSITIVE_ICON_WEAK_HOVER = "#6fd2fc";
export const POSITIVE_ICON_WEAK_FOCUS = "#6fd2fc";
export const POSITIVE_ICON_WEAK_ACTIVE = "#6fd2fc";
export const POSITIVE_ICON_WEAK_DISABLED = "#596069";
export const POSITIVE_ICON_STRONG_DEFAULT = "#33bbf5";
export const POSITIVE_ICON_STRONG_HOVER = "#33bbf5";
export const POSITIVE_ICON_STRONG_FOCUS = "#33bbf5";
export const POSITIVE_ICON_STRONG_ACTIVE = "#33bbf5";
export const POSITIVE_ICON_STRONG_DISABLED = "#2b7292";
export const POSITIVE_ICON_IN_DEFAULT = "#23282e";
export const POSITIVE_ICON_IN_HOVER = "#23282e";
export const POSITIVE_ICON_IN_FOCUS = "#23282e";
export const POSITIVE_ICON_IN_ACTIVE = "#23282e";
export const POSITIVE_ICON_IN_DISABLED = "#23282e";
export const POSITIVE_BORDER_DEFAULT = "#33bbf5";
export const POSITIVE_BORDER_HOVER = "#33bbf5";
export const POSITIVE_BORDER_FOCUS = "#33bbf5";
export const POSITIVE_BORDER_ACTIVE = "#33bbf5";
export const POSITIVE_BORDER_DISABLED = "#33bbf5";
export const POSITIVE_BORDER_WEAK_DEFAULT = "#23282e";
export const POSITIVE_BORDER_WEAK_HOVER = "#33bbf5";
export const POSITIVE_BORDER_WEAK_FOCUS = "#33bbf5";
export const POSITIVE_BORDER_WEAK_ACTIVE = "#33bbf5";
export const POSITIVE_BORDER_WEAK_DISABLED = "#596069";
export const POSITIVE_BORDER_STRONG_DEFAULT = "#f7f7f7";
export const POSITIVE_BORDER_STRONG_HOVER = "#f7f7f7";
export const POSITIVE_BORDER_STRONG_FOCUS = "#f7f7f7";
export const POSITIVE_BORDER_STRONG_ACTIVE = "#f7f7f7";
export const POSITIVE_BORDER_STRONG_DISABLED = "#f7f7f7";
export const POSITIVE_TEXT_IN_DEFAULT = "#23282e";
export const POSITIVE_TEXT_IN_HOVER = "#23282e";
export const POSITIVE_TEXT_IN_FOCUS = "#23282e";
export const POSITIVE_TEXT_IN_ACTIVE = "#23282e";
export const POSITIVE_TEXT_IN_DISABLED = "#23282e";
export const WARNING_BG_DEFAULT = "#f0c243";
export const WARNING_BG_HOVER = "#f0c243";
export const WARNING_BG_FOCUS = "#f0c243";
export const WARNING_BG_ACTIVE = "#f0c243";
export const WARNING_BG_DISABLED = "#f0c243";
export const WARNING_BG_WEAK_DEFAULT = "#23282e";
export const WARNING_BG_WEAK_HOVER = "#23282e";
export const WARNING_BG_WEAK_FOCUS = "#23282e";
export const WARNING_BG_WEAK_ACTIVE = "#23282e";
export const WARNING_BG_WEAK_DISABLED = "#23282e";
export const WARNING_BG_MEDIUM_DEFAULT = "#5c4d28";
export const WARNING_BG_MEDIUM_HOVER = "#5c4d28";
export const WARNING_BG_MEDIUM_FOCUS = "#5c4d28";
export const WARNING_BG_MEDIUM_ACTIVE = "#5c4d28";
export const WARNING_BG_MEDIUM_DISABLED = "#373c42";
export const WARNING_BG_STRONG_DEFAULT = "#35301c";
export const WARNING_ICON_DEFAULT = "#f0c243";
export const WARNING_ICON_HOVER = "#f0c243";
export const WARNING_ICON_FOCUS = "#f0c243";
export const WARNING_ICON_ACTIVE = "#f0c243";
export const WARNING_ICON_DISABLED = "#596069";
export const WARNING_ICON_STRONG_DEFAULT = "#f0c243";
export const WARNING_ICON_STRONG_HOVER = "#f0c243";
export const WARNING_ICON_STRONG_FOCUS = "#f0c243";
export const WARNING_ICON_STRONG_ACTIVE = "#f0c243";
export const WARNING_ICON_STRONG_DISABLED = "#f0c243";
export const WARNING_ICON_IN_DEFAULT = "#23282e";
export const WARNING_ICON_IN_HOVER = "#23282e";
export const WARNING_ICON_IN_FOCUS = "#23282e";
export const WARNING_ICON_IN_ACTIVE = "#23282e";
export const WARNING_ICON_IN_DISABLED = "#23282e";
export const WARNING_BORDER_DEFAULT = "#f0c243";
export const WARNING_BORDER_HOVER = "#f0c243";
export const WARNING_BORDER_FOCUS = "#f0c243";
export const WARNING_BORDER_ACTIVE = "#f0c243";
export const WARNING_BORDER_DISABLED = "#f0c243";
export const WARNING_BORDER_WEAK_DEFAULT = "#23282e";
export const WARNING_BORDER_WEAK_HOVER = "#23282e";
export const WARNING_BORDER_WEAK_FOCUS = "#23282e";
export const WARNING_BORDER_WEAK_ACTIVE = "#23282e";
export const WARNING_BORDER_WEAK_DISABLED = "#596069";
export const WARNING_BORDER_STRONG_DEFAULT = "#f7f7f7";
export const WARNING_BORDER_STRONG_HOVER = "#f7f7f7";
export const WARNING_BORDER_STRONG_FOCUS = "#f7f7f7";
export const WARNING_BORDER_STRONG_ACTIVE = "#f7f7f7";
export const WARNING_BORDER_STRONG_DISABLED = "#f7f7f7";
export const WARNING_TEXT_IN_DEFAULT = "#23282e";
export const WARNING_TEXT_IN_HOVER = "#23282e";
export const WARNING_TEXT_IN_FOCUS = "#23282e";
export const WARNING_TEXT_IN_ACTIVE = "#23282e";
export const WARNING_TEXT_IN_DISABLED = "#23282e";
export const SEVERE_WARNING_BG_DEFAULT = "#f7782f";
export const SEVERE_WARNING_BG_HOVER = "#f7782f";
export const SEVERE_WARNING_BG_FOCUS = "#f7782f";
export const SEVERE_WARNING_BG_ACTIVE = "#f7782f";
export const SEVERE_WARNING_BG_DISABLED = "#f7782f";
export const SEVERE_WARNING_BG_WEAK_DEFAULT = "#23282e";
export const SEVERE_WARNING_BG_WEAK_HOVER = "#23282e";
export const SEVERE_WARNING_BG_WEAK_FOCUS = "#23282e";
export const SEVERE_WARNING_BG_WEAK_ACTIVE = "#23282e";
export const SEVERE_WARNING_BG_WEAK_DISABLED = "#23282e";
export const SEVERE_WARNING_BG_MEDIUM_DEFAULT = "#644637";
export const SEVERE_WARNING_BG_MEDIUM_HOVER = "#644637";
export const SEVERE_WARNING_BG_MEDIUM_FOCUS = "#644637";
export const SEVERE_WARNING_BG_MEDIUM_ACTIVE = "#644637";
export const SEVERE_WARNING_BG_MEDIUM_DISABLED = "#373c42";
export const SEVERE_WARNING_BG_STRONG_DEFAULT = "#462e21";
export const SEVERE_WARNING_ICON_DEFAULT = "#f7782f";
export const SEVERE_WARNING_ICON_HOVER = "#f7782f";
export const SEVERE_WARNING_ICON_FOCUS = "#f7782f";
export const SEVERE_WARNING_ICON_ACTIVE = "#f7782f";
export const SEVERE_WARNING_ICON_DISABLED = "#596069";
export const SEVERE_WARNING_ICON_STRONG_DEFAULT = "#fc8d4c";
export const SEVERE_WARNING_ICON_STRONG_HOVER = "#fc8d4c";
export const SEVERE_WARNING_ICON_STRONG_FOCUS = "#fc8d4c";
export const SEVERE_WARNING_ICON_STRONG_ACTIVE = "#fc8d4c";
export const SEVERE_WARNING_ICON_STRONG_DISABLED = "#fc8d4c";
export const SEVERE_WARNING_ICON_IN_DEFAULT = "#23282e";
export const SEVERE_WARNING_ICON_IN_HOVER = "#23282e";
export const SEVERE_WARNING_ICON_IN_FOCUS = "#23282e";
export const SEVERE_WARNING_ICON_IN_ACTIVE = "#23282e";
export const SEVERE_WARNING_ICON_IN_DISABLED = "#23282e";
export const SEVERE_WARNING_BORDER_DEFAULT = "#f7782f";
export const SEVERE_WARNING_BORDER_HOVER = "#f7782f";
export const SEVERE_WARNING_BORDER_FOCUS = "#f7782f";
export const SEVERE_WARNING_BORDER_ACTIVE = "#f7782f";
export const SEVERE_WARNING_BORDER_DISABLED = "#f7782f";
export const SEVERE_WARNING_BORDER_WEAK_DEFAULT = "#23282e";
export const SEVERE_WARNING_BORDER_WEAK_HOVER = "#23282e";
export const SEVERE_WARNING_BORDER_WEAK_FOCUS = "#23282e";
export const SEVERE_WARNING_BORDER_WEAK_ACTIVE = "#23282e";
export const SEVERE_WARNING_BORDER_WEAK_DISABLED = "#596069";
export const NEGATIVE_BG_DEFAULT = "#fa5762";
export const NEGATIVE_BG_HOVER = "#ff6e72";
export const NEGATIVE_BG_FOCUS = "#ff6e72";
export const NEGATIVE_BG_ACTIVE = "#ff6e72";
export const NEGATIVE_BG_DISABLED = "#8f4048";
export const NEGATIVE_BG_WEAK_DEFAULT = "#23282e";
export const NEGATIVE_BG_WEAK_HOVER = "#994a52";
export const NEGATIVE_BG_WEAK_FOCUS = "#994a52";
export const NEGATIVE_BG_WEAK_ACTIVE = "#994a52";
export const NEGATIVE_BG_WEAK_DISABLED = "#373c42";
export const NEGATIVE_BG_MEDIUM_DEFAULT = "#64363e";
export const NEGATIVE_BG_MEDIUM_HOVER = "#64363e";
export const NEGATIVE_BG_MEDIUM_FOCUS = "#64363e";
export const NEGATIVE_BG_MEDIUM_ACTIVE = "#64363e";
export const NEGATIVE_BG_MEDIUM_DISABLED = "#373c42";
export const NEGATIVE_BG_STRONG_DEFAULT = "#46272a";
export const NEGATIVE_TEXT_DEFAULT = "#fa5762";
export const NEGATIVE_TEXT_HOVER = "#ff878b";
export const NEGATIVE_TEXT_FOCUS = "#ff878b";
export const NEGATIVE_TEXT_ACTIVE = "#ff878b";
export const NEGATIVE_TEXT_DISABLED = "#8f4048";
export const NEGATIVE_TEXT_IN_DEFAULT = "#23282e";
export const NEGATIVE_TEXT_IN_HOVER = "#23282e";
export const NEGATIVE_TEXT_IN_FOCUS = "#23282e";
export const NEGATIVE_TEXT_IN_ACTIVE = "#23282e";
export const NEGATIVE_TEXT_IN_DISABLED = "#23282e";
export const NEGATIVE_ICON_DEFAULT = "#fa5762";
export const NEGATIVE_ICON_HOVER = "#ff878b";
export const NEGATIVE_ICON_FOCUS = "#ff878b";
export const NEGATIVE_ICON_ACTIVE = "#ff878b";
export const NEGATIVE_ICON_DISABLED = "#8f4048";
export const NEGATIVE_ICON_WEAK_DEFAULT = "#ffb2b5";
export const NEGATIVE_ICON_WEAK_HOVER = "#ffb2b5";
export const NEGATIVE_ICON_WEAK_FOCUS = "#ffb2b5";
export const NEGATIVE_ICON_WEAK_ACTIVE = "#ffb2b5";
export const NEGATIVE_ICON_WEAK_DISABLED = "#596069";
export const NEGATIVE_ICON_IN_DEFAULT = "#23282e";
export const NEGATIVE_ICON_IN_HOVER = "#23282e";
export const NEGATIVE_ICON_IN_FOCUS = "#23282e";
export const NEGATIVE_ICON_IN_ACTIVE = "#23282e";
export const NEGATIVE_ICON_IN_DISABLED = "#23282e";
export const NEGATIVE_BORDER_DEFAULT = "#fa5762";
export const NEGATIVE_BORDER_HOVER = "#ff878b";
export const NEGATIVE_BORDER_FOCUS = "#ff878b";
export const NEGATIVE_BORDER_ACTIVE = "#ff878b";
export const NEGATIVE_BORDER_DISABLED = "#914b50";
export const NEGATIVE_BORDER_WEAK_DEFAULT = "#23282e";
export const NEGATIVE_BORDER_WEAK_HOVER = "#23282e";
export const NEGATIVE_BORDER_WEAK_FOCUS = "#23282e";
export const NEGATIVE_BORDER_WEAK_ACTIVE = "#23282e";
export const NEGATIVE_BORDER_WEAK_DISABLED = "#596069";
export const NEGATIVE_BORDER_MEDIUM_DEFAULT = "#eb4651";
export const NEGATIVE_BORDER_MEDIUM_HOVER = "#ff878b";
export const NEGATIVE_BORDER_MEDIUM_FOCUS = "#ff878b";
export const NEGATIVE_BORDER_MEDIUM_ACTIVE = "#ff878b";
export const NEGATIVE_BORDER_MEDIUM_DISABLED = "#8f4048";
export const NEGATIVE_BORDER_STRONG_DEFAULT = "#f7f7f7";
export const NEGATIVE_BORDER_STRONG_HOVER = "#ffffff";
export const NEGATIVE_BORDER_STRONG_FOCUS = "#f7f7f7";
export const NEGATIVE_BORDER_STRONG_ACTIVE = "#f7f7f7";
export const NEGATIVE_BORDER_STRONG_DISABLED = "#f7f7f7";
export const INFO_BG_DEFAULT = "#b587fa";
export const INFO_BG_HOVER = "#b587fa";
export const INFO_BG_FOCUS = "#b587fa";
export const INFO_BG_ACTIVE = "#b587fa";
export const INFO_BG_DISABLED = "#b587fa";
export const INFO_BG_WEAK_DEFAULT = "#23282e";
export const INFO_BG_WEAK_HOVER = "#23282e";
export const INFO_BG_WEAK_FOCUS = "#23282e";
export const INFO_BG_WEAK_ACTIVE = "#23282e";
export const INFO_BG_WEAK_DISABLED = "#23282e";
export const INFO_BG_MEDIUM_DEFAULT = "#534a6d";
export const INFO_BG_MEDIUM_HOVER = "#534a6d";
export const INFO_BG_MEDIUM_FOCUS = "#534a6d";
export const INFO_BG_MEDIUM_ACTIVE = "#534a6d";
export const INFO_BG_MEDIUM_DISABLED = "#373c42";
export const INFO_BG_STRONG_DEFAULT = "#352d49";
export const INFO_ICON_DEFAULT = "#b587fa";
export const INFO_ICON_HOVER = "#b587fa";
export const INFO_ICON_FOCUS = "#b587fa";
export const INFO_ICON_ACTIVE = "#b587fa";
export const INFO_ICON_DISABLED = "#596069";
export const INFO_ICON_WEAK_DEFAULT = "#a974f7";
export const INFO_ICON_WEAK_HOVER = "#a974f7";
export const INFO_ICON_WEAK_FOCUS = "#a974f7";
export const INFO_ICON_WEAK_ACTIVE = "#a974f7";
export const INFO_ICON_WEAK_DISABLED = "#596069";
export const INFO_ICON_IN_DEFAULT = "#23282e";
export const INFO_ICON_IN_HOVER = "#23282e";
export const INFO_ICON_IN_FOCUS = "#23282e";
export const INFO_ICON_IN_ACTIVE = "#23282e";
export const INFO_ICON_IN_DISABLED = "#23282e";
export const INFO_BORDER_DEFAULT = "#b587fa";
export const INFO_BORDER_HOVER = "#b587fa";
export const INFO_BORDER_FOCUS = "#b587fa";
export const INFO_BORDER_ACTIVE = "#b587fa";
export const INFO_BORDER_DISABLED = "#b587fa";
export const INFO_BORDER_WEAK_DEFAULT = "#23282e";
export const INFO_BORDER_WEAK_HOVER = "#23282e";
export const INFO_BORDER_WEAK_FOCUS = "#23282e";
export const INFO_BORDER_WEAK_ACTIVE = "#23282e";
export const INFO_BORDER_WEAK_DISABLED = "#596069";
export const DORMANT_BG_DEFAULT = "#6f7680";
export const DORMANT_BG_DISABLED = "#6f7680";
export const DORMANT_BG_MEDIUM_DEFAULT = "#464c54";
export const DORMANT_BG_MEDIUM_DISABLED = "#373c42";
export const DORMANT_ICON_DEFAULT = "#c1c6cc";
export const DORMANT_ICON_DISABLED = "#596069";
export const DORMANT_ICON_MEDIUM_DEFAULT = "#ffffff";
export const DORMANT_ICON_IN_DEFAULT = "#23282e";
export const DORMANT_ICON_IN_DISABLED = "#23282e";
export const DORMANT_ICON_IN_MEDIUM_DEFAULT = "#23282e";
export const DORMANT_BORDER_DEFAULT = "#a7adb5";
export const DORMANT_BORDER_DISABLED = "#596069";
export const ACCENT_A_DEFAULT = "#9ca6ff";
export const ACCENT_A_HOVER = "#8a95ff";
export const ACCENT_A_WEAK_DEFAULT = "#4d5477";
export const ACCENT_B_DEFAULT = "#e3447c";
export const ACCENT_B_HOVER = "#cf3a7a";
export const ACCENT_B_WEAK_DEFAULT = "#704959";
export const ACCENT_C_DEFAULT = "#fcb3c8";
export const ACCENT_C_HOVER = "#ff87a9";
export const ACCENT_C_WEAK_DEFAULT = "#6f5964";
export const ACCENT_D_DEFAULT = "#9b5ff5";
export const ACCENT_D_HOVER = "#8d4eed";
export const ACCENT_D_WEAK_DEFAULT = "#5b5077";
export const ACCENT_E_DEFAULT = "#9dba4c";
export const ACCENT_E_HOVER = "#89ab2c";
export const ACCENT_E_WEAK_DEFAULT = "#4e5b39";
export const ACCENT_F_DEFAULT = "#d95a1a";
export const ACCENT_F_HOVER = "#c44f14";
export const ACCENT_F_WEAK_DEFAULT = "#6f4b39";
export const ACCENT_G_DEFAULT = "#4ad9d9";
export const ACCENT_G_HOVER = "#17c2c2";
export const ACCENT_G_WEAK_DEFAULT = "#1f5e62";
export const ACCENT_H_DEFAULT = "#028e99";
export const ACCENT_H_HOVER = "#01818c";
export const ACCENT_H_WEAK_DEFAULT = "#45696d";
export const ACCENT_I_DEFAULT = "#f582d8";
export const ACCENT_I_HOVER = "#f26dd1";
export const ACCENT_I_WEAK_DEFAULT = "#6d486a";
export const ACCENT_J_DEFAULT = "#767eb2";
export const ACCENT_J_HOVER = "#6871a3";
export const ACCENT_J_WEAK_DEFAULT = "#505669";
export const ACCENT_K_DEFAULT = "#4cbf7f";
export const ACCENT_K_HOVER = "#36b26e";
export const ACCENT_K_WEAK_DEFAULT = "#315d4a";
export const IN_PROGRESS_BG_MEDIUM_DEFAULT = "#534a6d";
export const IN_PROGRESS_BG_MEDIUM_HOVER = "#534a6d";
export const IN_PROGRESS_BG_MEDIUM_FOCUS = "#534a6d";
export const IN_PROGRESS_BG_MEDIUM_ACTIVE = "#534a6d";
export const IN_PROGRESS_BG_MEDIUM_DISABLED = "#373c42";
export const IN_PROGRESS_ICON_DEFAULT = "#b587fa";
export const IN_PROGRESS_ICON_HOVER = "#b587fa";
export const IN_PROGRESS_ICON_FOCUS = "#b587fa";
export const IN_PROGRESS_ICON_ACTIVE = "#b587fa";
export const IN_PROGRESS_ICON_DISABLED = "#596069";
export const IN_PROGRESS_ICON_IN_DEFAULT = "#23282e";
export const IN_PROGRESS_ICON_IN_HOVER = "#23282e";
export const IN_PROGRESS_ICON_IN_FOCUS = "#23282e";
export const IN_PROGRESS_ICON_IN_ACTIVE = "#23282e";
export const IN_PROGRESS_ICON_IN_DISABLED = "#23282e";
export const IN_PROGRESS_BORDER_DEFAULT = "#b587fa";
export const IN_PROGRESS_BORDER_HOVER = "#b587fa";
export const IN_PROGRESS_BORDER_FOCUS = "#b587fa";
export const IN_PROGRESS_BORDER_ACTIVE = "#b587fa";
export const IN_PROGRESS_BORDER_DISABLED = "#b587fa";
export const DECENT_BG_MEDIUM_DEFAULT = "#1f565a";
export const DECENT_BG_MEDIUM_HOVER = "#1f565a";
export const DECENT_BG_MEDIUM_FOCUS = "#1f565a";
export const DECENT_BG_MEDIUM_ACTIVE = "#1f565a";
export const DECENT_BG_MEDIUM_DISABLED = "#373c42";
export const DECENT_ICON_DEFAULT = "#0bb2b8";
export const DECENT_ICON_HOVER = "#0bb2b8";
export const DECENT_ICON_FOCUS = "#0bb2b8";
export const DECENT_ICON_ACTIVE = "#0bb2b8";
export const DECENT_ICON_DISABLED = "#596069";
export const DECENT_ICON_IN_DEFAULT = "#23282e";
export const DECENT_ICON_IN_HOVER = "#23282e";
export const DECENT_ICON_IN_FOCUS = "#23282e";
export const DECENT_ICON_IN_ACTIVE = "#23282e";
export const DECENT_ICON_IN_DISABLED = "#23282e";
export const DECENT_BORDER_DEFAULT = "#0bb2b8";
export const DECENT_BORDER_HOVER = "#0bb2b8";
export const DECENT_BORDER_FOCUS = "#0bb2b8";
export const DECENT_BORDER_ACTIVE = "#0bb2b8";
export const DECENT_BORDER_DISABLED = "#0bb2b8";
export const EXCELLENT_BG_MEDIUM_DEFAULT = "#3e506a";
export const EXCELLENT_BG_MEDIUM_HOVER = "#3e506a";
export const EXCELLENT_BG_MEDIUM_FOCUS = "#3e506a";
export const EXCELLENT_BG_MEDIUM_ACTIVE = "#3e506a";
export const EXCELLENT_BG_MEDIUM_DISABLED = "#373c42";
export const EXCELLENT_ICON_DEFAULT = "#649ef5";
export const EXCELLENT_ICON_HOVER = "#649ef5";
export const EXCELLENT_ICON_FOCUS = "#649ef5";
export const EXCELLENT_ICON_ACTIVE = "#649ef5";
export const EXCELLENT_ICON_DISABLED = "#596069";
export const EXCELLENT_ICON_IN_DEFAULT = "#23282e";
export const EXCELLENT_ICON_IN_HOVER = "#23282e";
export const EXCELLENT_ICON_IN_FOCUS = "#23282e";
export const EXCELLENT_ICON_IN_ACTIVE = "#23282e";
export const EXCELLENT_ICON_IN_DISABLED = "#23282e";
export const EXCELLENT_BORDER_DEFAULT = "#649ef5";
export const EXCELLENT_BORDER_HOVER = "#649ef5";
export const EXCELLENT_BORDER_FOCUS = "#649ef5";
export const EXCELLENT_BORDER_ACTIVE = "#649ef5";
export const EXCELLENT_BORDER_DISABLED = "#649ef5";
