// @ciscodesignsystems/cds-react-empty-state/empty-state.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var index = require('./components/illustration/index.cjs.js');
var index$1 = require('./components/message/index.cjs.js');
var index$2 = require('./components/title/index.cjs.js');
var emptyState_module = require('./empty-state.module.cjs.js');

const CDSEmptyState = /*#__PURE__*/react.forwardRef((props, ref) => {
  const {
    children,
    className,
    ...rest
  } = props;
  const classNames = cdsComponentUtilities.cx(emptyState_module["default"], 'cds-empty-state', className);
  return jsxRuntime.jsx("div", {
    className: classNames,
    ref: ref,
    ...rest,
    children: children
  });
});
CDSEmptyState.displayName = 'CDSEmptyState';
const CDSEmptyStateNamespace = Object.assign(CDSEmptyState, {
  Illustration: index.CDSEmptyStateIllustration,
  Message: index$1.CDSEmptyStateMessage,
  Title: index$2.CDSEmptyStateTitle
});
CDSEmptyStateNamespace.displayName = 'CDSEmptyState';

exports.CDSEmptyState = CDSEmptyStateNamespace;
exports["default"] = CDSEmptyState;
