// @ciscodesignsystems/cds-react-empty-state/empty-state.esm.js
import { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { CDSEmptyStateIllustration } from './components/illustration/index.esm.js';
import { CDSEmptyStateMessage } from './components/message/index.esm.js';
import { CDSEmptyStateTitle } from './components/title/index.esm.js';
import styles from './empty-state.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const CDSEmptyState = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    className,
    ...rest
  } = props;
  const classNames = cx(styles, 'cds-empty-state', className);
  return /*#__PURE__*/jsx("div", {
    className: classNames,
    ref: ref,
    ...rest,
    children: children
  });
});
CDSEmptyState.displayName = 'CDSEmptyState';
const CDSEmptyStateNamespace = Object.assign(CDSEmptyState, {
  Illustration: CDSEmptyStateIllustration,
  Message: CDSEmptyStateMessage,
  Title: CDSEmptyStateTitle
});
CDSEmptyStateNamespace.displayName = 'CDSEmptyState';

export { CDSEmptyStateNamespace as CDSEmptyState, CDSEmptyState as default };
