// @ciscodesignsystems/cds-react-header/components/tenant-menu/tenant-menu.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var headerMenu = require('../header-menu/header-menu.cjs.js');
var tenantSwitcher = require('../tenant-switcher/tenant-switcher.cjs.js');
var tenantMenu_module = require('./tenant-menu.module.cjs.js');

// TODO @jjc - add title attribute and make ellipsis optional
/**
 * The most-parent element of the alerts widget
 */
const CDSTenantMenuRoot = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    children,
    contentRef,
    switcherType,
    userName,
    tenantName,
    sentiment,
    containerMaxWidth,
    containerFlex,
    contentWidth,
    className
  } = props;
  return jsxRuntime.jsx(headerMenu.CDSHeaderMenuRoot, {
    ref: ref,
    id: "headerTenantMenu",
    containerMaxWidth: containerMaxWidth,
    flex: userName || tenantName ? containerFlex : '',
    trigger: jsxRuntime.jsx(tenantSwitcher.CDSTenantSwitcher, {
      switcherType: switcherType,
      sentiment: sentiment,
      userName: userName,
      tenantName: tenantName,
      width: contentWidth
    }),
    className: className,
    children: jsxRuntime.jsx(headerMenu.CDSHeaderMenuContent, {
      className: cdsComponentUtilities.cx(tenantMenu_module["default"], 'cds-tenant-menu'),
      align: "left",
      ref: contentRef,
      children: children
    })
  });
});
CDSTenantMenuRoot.displayName = 'CDSTenantMenuRoot';
const CDSTenantMenuRow = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    children,
    leftGutter,
    rightGutter,
    truncate,
    className,
    ...rest
  } = props;
  const menuRowContentClasses = cdsComponentUtilities.cx(tenantMenu_module["default"], 'cds-tenant-menu__row__content', {
    'cds-tenant-menu__row__content--truncated': truncate
  });
  return jsxRuntime.jsxs(headerMenu.CDSHeaderMenuRow, {
    ref: ref,
    className: cdsComponentUtilities.cx(tenantMenu_module["default"], 'cds-tenant-menu__row', className),
    ...rest,
    children: [leftGutter ? jsxRuntime.jsx("div", {
      className: cdsComponentUtilities.cx(tenantMenu_module["default"], 'cds-tenant-menu__row__gutter', 'cds-tenant-menu__row__gutter--left'),
      children: leftGutter
    }) : null, jsxRuntime.jsx("div", {
      className: menuRowContentClasses,
      children: children
    }), rightGutter ? jsxRuntime.jsx("div", {
      className: cdsComponentUtilities.cx(tenantMenu_module["default"], 'cds-tenant-menu__row__gutter', 'cds-tenant-menu__row__gutter--right'),
      children: rightGutter
    }) : null]
  });
});
CDSTenantMenuRow.displayName = 'CDSTenantMenuRow';
CDSTenantMenuRow.defaultProps = {
  leftGutter: undefined,
  rightGutter: undefined,
  truncate: false
};
/**
 * Divider that goes between rows of content
 */
const CDSTenantMenuDivider = /*#__PURE__*/React.forwardRef((props, ref) => {
  return jsxRuntime.jsx("hr", {
    className: cdsComponentUtilities.cx(tenantMenu_module["default"], 'cds-tenant-menu__divider'),
    ref: ref,
    ...props
  });
});
CDSTenantMenuDivider.displayName = 'CDSTenantMenuDivider';
/**
 * Content that goes at the bottom of the Alerts Menu
 */
const CDSTenantMenuFooter = /*#__PURE__*/React.forwardRef((props, ref) => {
  const {
    children,
    className,
    ...rest
  } = props;
  return jsxRuntime.jsx(headerMenu.CDSHeaderMenuRow, {
    ref: ref,
    className: cdsComponentUtilities.cx(tenantMenu_module["default"], 'cds-tenant-menu__footer', className),
    ...rest,
    children: children
  });
});
CDSTenantMenuFooter.displayName = 'CDSTenantMenuFooter';
CDSTenantMenuRoot.defaultProps = {
  sentiment: 'base',
  switcherType: 'combo',
  containerMaxWidth: '220px',
  containerFlex: '1',
  contentRef: undefined,
  userName: 'User Name',
  tenantName: 'Tenant Name',
  contentWidth: '120px',
  className: undefined
};

exports.CDSTenantMenuDivider = CDSTenantMenuDivider;
exports.CDSTenantMenuFooter = CDSTenantMenuFooter;
exports.CDSTenantMenuRoot = CDSTenantMenuRoot;
exports.CDSTenantMenuRow = CDSTenantMenuRow;
exports["default"] = CDSTenantMenuRoot;
