// @ciscodesignsystems/cds-react-menu/injected/menu.esm.js
import { forwardRef } from 'react';
import { DropdownMenuPortal, DropdownMenuContent, DropdownMenuItem, DropdownMenuTrigger, DropdownMenuSub, DropdownMenuSubTrigger, DropdownMenuSubContent } from '@radix-ui/react-dropdown-menu';
export { DropdownMenu as CDSMenuRoot } from '@radix-ui/react-dropdown-menu';
import { jsxs, jsx } from 'react/jsx-runtime';

/**
 * MenuContent
 */
const CDSMenuContent = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return /*#__PURE__*/jsxs("div", {
    children: [/*#__PURE__*/jsx("h3", {
      children: "Do not implement - this is for example purposes only"
    }), /*#__PURE__*/jsx(DropdownMenuPortal, {
      children: /*#__PURE__*/jsx(DropdownMenuContent, {
        ref: ref,
        "data-cds-menu-content": true,
        ...rest,
        children: children
      })
    })]
  });
});
CDSMenuContent.displayName = 'CDSMenuContent';

/**
 * MenuItem
 */
const CDSMenuItem = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return /*#__PURE__*/jsx(DropdownMenuItem, {
    ref: ref,
    "data-cds-menu-item": true,
    ...rest,
    children: children
  });
});
CDSMenuItem.displayName = 'CDSMenuItem';

/**
 * MenuTrigger
 */
const CDSMenuTrigger = props => {
  const {
    children,
    ...rest
  } = props;
  return /*#__PURE__*/jsx(DropdownMenuTrigger, {
    asChild: true,
    ...rest,
    children: children
  });
};

/**
 * SubMenu
 */
const CDSSubMenuRoot = ({
  children,
  ...rest
}) => {
  return /*#__PURE__*/jsx(DropdownMenuSub, {
    ...rest,
    children: children
  });
};

/**
 * SubMenuTrigger
 * TODO @bla - add a caret icon prop
 */
const CDSSubMenuTrigger = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return /*#__PURE__*/jsxs(DropdownMenuSubTrigger, {
    ref: ref,
    "data-cds-menu-item": true,
    "data-cds-sub-menu-item": true,
    ...rest,
    children: [children, /*#__PURE__*/jsx(ArrowRightIcon, {})]
  });
});
CDSSubMenuTrigger.displayName = 'CDSSubMenuTrigger';

/**
 * SubMenuContent
 */
const CDSSubMenuContent = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    ...rest
  } = props;
  return /*#__PURE__*/jsx(DropdownMenuPortal, {
    children: /*#__PURE__*/jsx(DropdownMenuSubContent, {
      ref: ref,
      "data-cds-menu-content": true,
      ...rest,
      children: children
    })
  });
});
CDSSubMenuContent.displayName = 'CDSSubMenuContent';
const ArrowRightIcon = () => {
  return /*#__PURE__*/jsxs("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    width: "16",
    height: "16",
    fill: "#000000",
    viewBox: "0 0 256 256",
    children: [/*#__PURE__*/jsx("rect", {
      width: "256",
      height: "256",
      fill: "none"
    }), /*#__PURE__*/jsx("line", {
      x1: "40",
      y1: "128",
      x2: "216",
      y2: "128",
      fill: "none",
      stroke: "#000000",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "16"
    }), /*#__PURE__*/jsx("polyline", {
      points: "144 56 216 128 144 200",
      fill: "none",
      stroke: "#000000",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "16"
    })]
  });
};

export { CDSMenuContent, CDSMenuItem, CDSMenuTrigger, CDSSubMenuContent, CDSSubMenuRoot, CDSSubMenuTrigger };
