// @ciscodesignsystems/cds-react-table/injected/components/table-row/table-row.esm.js
import { forwardRef } from 'react';
import { cx } from '@ciscodesignsystems/cds-component-utilities';
import { TableCell } from '../table-cell/table-cell.esm.js';
import styles from './table-row.module.esm.js';
import { jsx } from 'react/jsx-runtime';

const TableRowComponent = ({
  row,
  pinnedPositions = undefined,
  className,
  children,
  dragComponent = undefined,
  onClick,
  ...rest
}, ref) => {
  const rowClassNames = cx(styles, 'cds-table__row', {
    'cds-table__row--clickable': onClick
  }, className);
  return /*#__PURE__*/jsx("tr", {
    ref: ref,
    className: rowClassNames,
    ...rest,
    onClick: onClick,
    children: row.getVisibleCells().map(cell => {
      const isPinned = cell.column.getIsPinned();
      const pinnedPosition = isPinned !== false ? pinnedPositions?.[isPinned][cell.column.getPinnedIndex()] : undefined;
      return /*#__PURE__*/jsx(TableCell, {
        cell: cell,
        pinnedPosition: pinnedPosition,
        children: cell.column.id === 'row-sort-dnd' ? dragComponent : null
      }, cell.id);
    })
  });
};
const TableRow = /*#__PURE__*/forwardRef(TableRowComponent);

export { TableRow };
