// @ciscodesignsystems/cds-react-table/injected/components/table-row/table-row.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var cdsComponentUtilities = require('@ciscodesignsystems/cds-component-utilities');
var tableCell = require('../table-cell/table-cell.cjs.js');
var tableRow_module = require('./table-row.module.cjs.js');

const TableRowComponent = ({
  row,
  pinnedPositions = undefined,
  className,
  children,
  dragComponent = undefined,
  onClick,
  ...rest
}, ref) => {
  const rowClassNames = cdsComponentUtilities.cx(tableRow_module["default"], 'cds-table__row', {
    'cds-table__row--clickable': onClick
  }, className);
  return jsxRuntime.jsx("tr", {
    ref: ref,
    className: rowClassNames,
    ...rest,
    onClick: onClick,
    children: row.getVisibleCells().map(cell => {
      const isPinned = cell.column.getIsPinned();
      const pinnedPosition = isPinned !== false ? pinnedPositions === null || pinnedPositions === void 0 ? void 0 : pinnedPositions[isPinned][cell.column.getPinnedIndex()] : undefined;
      return jsxRuntime.jsx(tableCell.TableCell, {
        cell: cell,
        pinnedPosition: pinnedPosition,
        children: cell.column.id === 'row-sort-dnd' ? dragComponent : null
      }, cell.id);
    })
  });
};
const TableRow = /*#__PURE__*/react.forwardRef(TableRowComponent);

exports.TableRow = TableRow;
